// @hubspot/api-client/lib/codegen/crm/associations/v4/schema/models/BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum = exports.BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrors = void 0;
class BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrors {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrors = BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrors;
BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrors.discriminator = undefined;
BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrors.mapping = undefined;
BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociationDefinitionConfigurationUpdateResult>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum;
(function (BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum) {
    BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum = exports.BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum || (exports.BatchResponsePublicAssociationDefinitionConfigurationUpdateResultWithErrorsStatusEnum = {}));
