// @hubspot/api-client/lib/codegen/crm/associations/v4/schema/models/BatchResponsePublicAssociationDefinitionUserConfigurationWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum = exports.BatchResponsePublicAssociationDefinitionUserConfigurationWithErrors = void 0;
class BatchResponsePublicAssociationDefinitionUserConfigurationWithErrors {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociationDefinitionUserConfigurationWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePublicAssociationDefinitionUserConfigurationWithErrors = BatchResponsePublicAssociationDefinitionUserConfigurationWithErrors;
BatchResponsePublicAssociationDefinitionUserConfigurationWithErrors.discriminator = undefined;
BatchResponsePublicAssociationDefinitionUserConfigurationWithErrors.mapping = undefined;
BatchResponsePublicAssociationDefinitionUserConfigurationWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociationDefinitionUserConfiguration>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum;
(function (BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum) {
    BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum = exports.BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum || (exports.BatchResponsePublicAssociationDefinitionUserConfigurationWithErrorsStatusEnum = {}));
