// @hubspot/api-client/lib/codegen/crm/associations/v4/schema/models/BatchResponsePublicAssociationDefinitionUserConfiguration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum = exports.BatchResponsePublicAssociationDefinitionUserConfiguration = void 0;
class BatchResponsePublicAssociationDefinitionUserConfiguration {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociationDefinitionUserConfiguration.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePublicAssociationDefinitionUserConfiguration = BatchResponsePublicAssociationDefinitionUserConfiguration;
BatchResponsePublicAssociationDefinitionUserConfiguration.discriminator = undefined;
BatchResponsePublicAssociationDefinitionUserConfiguration.mapping = undefined;
BatchResponsePublicAssociationDefinitionUserConfiguration.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociationDefinitionUserConfiguration>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum",
        "format": ""
    }
];
var BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum;
(function (BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum) {
    BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum["Pending"] = "PENDING";
    BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum = exports.BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum || (exports.BatchResponsePublicAssociationDefinitionUserConfigurationStatusEnum = {}));
