// @hubspot/api-client/lib/codegen/crm/associations/v4/schema/models/BatchResponsePublicAssociationDefinitionConfigurationUpdateResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum = exports.BatchResponsePublicAssociationDefinitionConfigurationUpdateResult = void 0;
class BatchResponsePublicAssociationDefinitionConfigurationUpdateResult {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociationDefinitionConfigurationUpdateResult.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponsePublicAssociationDefinitionConfigurationUpdateResult = BatchResponsePublicAssociationDefinitionConfigurationUpdateResult;
BatchResponsePublicAssociationDefinitionConfigurationUpdateResult.discriminator = undefined;
BatchResponsePublicAssociationDefinitionConfigurationUpdateResult.mapping = undefined;
BatchResponsePublicAssociationDefinitionConfigurationUpdateResult.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociationDefinitionConfigurationUpdateResult>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum",
        "format": ""
    }
];
var BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum;
(function (BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum) {
    BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum["Pending"] = "PENDING";
    BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum["Processing"] = "PROCESSING";
    BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum["Canceled"] = "CANCELED";
    BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum["Complete"] = "COMPLETE";
})(BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum = exports.BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum || (exports.BatchResponsePublicAssociationDefinitionConfigurationUpdateResultStatusEnum = {}));
