// @hubspot/api-client/lib/codegen/crm/associations/v4/schema/models/PublicAssociationDefinitionConfigurationCreateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicAssociationDefinitionConfigurationCreateRequestCategoryEnum = exports.PublicAssociationDefinitionConfigurationCreateRequest = void 0;
class PublicAssociationDefinitionConfigurationCreateRequest {
    static getAttributeTypeMap() {
        return PublicAssociationDefinitionConfigurationCreateRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicAssociationDefinitionConfigurationCreateRequest = PublicAssociationDefinitionConfigurationCreateRequest;
PublicAssociationDefinitionConfigurationCreateRequest.discriminator = undefined;
PublicAssociationDefinitionConfigurationCreateRequest.mapping = undefined;
PublicAssociationDefinitionConfigurationCreateRequest.attributeTypeMap = [
    {
        "name": "typeId",
        "baseName": "typeId",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "PublicAssociationDefinitionConfigurationCreateRequestCategoryEnum",
        "format": ""
    },
    {
        "name": "maxToObjectIds",
        "baseName": "maxToObjectIds",
        "type": "number",
        "format": "int32"
    }
];
var PublicAssociationDefinitionConfigurationCreateRequestCategoryEnum;
(function (PublicAssociationDefinitionConfigurationCreateRequestCategoryEnum) {
    PublicAssociationDefinitionConfigurationCreateRequestCategoryEnum["HubspotDefined"] = "HUBSPOT_DEFINED";
    PublicAssociationDefinitionConfigurationCreateRequestCategoryEnum["UserDefined"] = "USER_DEFINED";
    PublicAssociationDefinitionConfigurationCreateRequestCategoryEnum["IntegratorDefined"] = "INTEGRATOR_DEFINED";
})(PublicAssociationDefinitionConfigurationCreateRequestCategoryEnum = exports.PublicAssociationDefinitionConfigurationCreateRequestCategoryEnum || (exports.PublicAssociationDefinitionConfigurationCreateRequestCategoryEnum = {}));
