// @hubspot/api-client/lib/codegen/crm/associations/v4/schema/models/PublicAssociationDefinitionConfigurationUpdateResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicAssociationDefinitionConfigurationUpdateResultCategoryEnum = exports.PublicAssociationDefinitionConfigurationUpdateResult = void 0;
class PublicAssociationDefinitionConfigurationUpdateResult {
    static getAttributeTypeMap() {
        return PublicAssociationDefinitionConfigurationUpdateResult.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicAssociationDefinitionConfigurationUpdateResult = PublicAssociationDefinitionConfigurationUpdateResult;
PublicAssociationDefinitionConfigurationUpdateResult.discriminator = undefined;
PublicAssociationDefinitionConfigurationUpdateResult.mapping = undefined;
PublicAssociationDefinitionConfigurationUpdateResult.attributeTypeMap = [
    {
        "name": "userEnforcedMaxToObjectIds",
        "baseName": "userEnforcedMaxToObjectIds",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "typeId",
        "baseName": "typeId",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "PublicAssociationDefinitionConfigurationUpdateResultCategoryEnum",
        "format": ""
    }
];
var PublicAssociationDefinitionConfigurationUpdateResultCategoryEnum;
(function (PublicAssociationDefinitionConfigurationUpdateResultCategoryEnum) {
    PublicAssociationDefinitionConfigurationUpdateResultCategoryEnum["HubspotDefined"] = "HUBSPOT_DEFINED";
    PublicAssociationDefinitionConfigurationUpdateResultCategoryEnum["UserDefined"] = "USER_DEFINED";
    PublicAssociationDefinitionConfigurationUpdateResultCategoryEnum["IntegratorDefined"] = "INTEGRATOR_DEFINED";
})(PublicAssociationDefinitionConfigurationUpdateResultCategoryEnum = exports.PublicAssociationDefinitionConfigurationUpdateResultCategoryEnum || (exports.PublicAssociationDefinitionConfigurationUpdateResultCategoryEnum = {}));
