// @hubspot/api-client/lib/codegen/crm/associations/v4/schema/models/PublicAssociationDefinitionUserConfiguration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicAssociationDefinitionUserConfigurationCategoryEnum = exports.PublicAssociationDefinitionUserConfiguration = void 0;
class PublicAssociationDefinitionUserConfiguration {
    static getAttributeTypeMap() {
        return PublicAssociationDefinitionUserConfiguration.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicAssociationDefinitionUserConfiguration = PublicAssociationDefinitionUserConfiguration;
PublicAssociationDefinitionUserConfiguration.discriminator = undefined;
PublicAssociationDefinitionUserConfiguration.mapping = undefined;
PublicAssociationDefinitionUserConfiguration.attributeTypeMap = [
    {
        "name": "userEnforcedMaxToObjectIds",
        "baseName": "userEnforcedMaxToObjectIds",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "typeId",
        "baseName": "typeId",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "PublicAssociationDefinitionUserConfigurationCategoryEnum",
        "format": ""
    }
];
var PublicAssociationDefinitionUserConfigurationCategoryEnum;
(function (PublicAssociationDefinitionUserConfigurationCategoryEnum) {
    PublicAssociationDefinitionUserConfigurationCategoryEnum["HubspotDefined"] = "HUBSPOT_DEFINED";
    PublicAssociationDefinitionUserConfigurationCategoryEnum["UserDefined"] = "USER_DEFINED";
    PublicAssociationDefinitionUserConfigurationCategoryEnum["IntegratorDefined"] = "INTEGRATOR_DEFINED";
})(PublicAssociationDefinitionUserConfigurationCategoryEnum = exports.PublicAssociationDefinitionUserConfigurationCategoryEnum || (exports.PublicAssociationDefinitionUserConfigurationCategoryEnum = {}));
