// @hubspot/api-client/lib/codegen/crm/associations/v4/schema/models/PublicAssociationDefinitionConfigurationUpdateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicAssociationDefinitionConfigurationUpdateRequestCategoryEnum = exports.PublicAssociationDefinitionConfigurationUpdateRequest = void 0;
class PublicAssociationDefinitionConfigurationUpdateRequest {
    static getAttributeTypeMap() {
        return PublicAssociationDefinitionConfigurationUpdateRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicAssociationDefinitionConfigurationUpdateRequest = PublicAssociationDefinitionConfigurationUpdateRequest;
PublicAssociationDefinitionConfigurationUpdateRequest.discriminator = undefined;
PublicAssociationDefinitionConfigurationUpdateRequest.mapping = undefined;
PublicAssociationDefinitionConfigurationUpdateRequest.attributeTypeMap = [
    {
        "name": "typeId",
        "baseName": "typeId",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "PublicAssociationDefinitionConfigurationUpdateRequestCategoryEnum",
        "format": ""
    },
    {
        "name": "maxToObjectIds",
        "baseName": "maxToObjectIds",
        "type": "number",
        "format": "int32"
    }
];
var PublicAssociationDefinitionConfigurationUpdateRequestCategoryEnum;
(function (PublicAssociationDefinitionConfigurationUpdateRequestCategoryEnum) {
    PublicAssociationDefinitionConfigurationUpdateRequestCategoryEnum["HubspotDefined"] = "HUBSPOT_DEFINED";
    PublicAssociationDefinitionConfigurationUpdateRequestCategoryEnum["UserDefined"] = "USER_DEFINED";
    PublicAssociationDefinitionConfigurationUpdateRequestCategoryEnum["IntegratorDefined"] = "INTEGRATOR_DEFINED";
})(PublicAssociationDefinitionConfigurationUpdateRequestCategoryEnum = exports.PublicAssociationDefinitionConfigurationUpdateRequestCategoryEnum || (exports.PublicAssociationDefinitionConfigurationUpdateRequestCategoryEnum = {}));
