// @hubspot/api-client/lib/codegen/crm/exports/models/ActionResponseWithSingleResultURI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionResponseWithSingleResultURIStatusEnum = exports.ActionResponseWithSingleResultURI = void 0;
class ActionResponseWithSingleResultURI {
    static getAttributeTypeMap() {
        return ActionResponseWithSingleResultURI.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ActionResponseWithSingleResultURI = ActionResponseWithSingleResultURI;
ActionResponseWithSingleResultURI.discriminator = undefined;
ActionResponseWithSingleResultURI.mapping = undefined;
ActionResponseWithSingleResultURI.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "result",
        "type": "string",
        "format": ""
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "ActionResponseWithSingleResultURIStatusEnum",
        "format": ""
    }
];
var ActionResponseWithSingleResultURIStatusEnum;
(function (ActionResponseWithSingleResultURIStatusEnum) {
    ActionResponseWithSingleResultURIStatusEnum["Pending"] = "PENDING";
    ActionResponseWithSingleResultURIStatusEnum["Processing"] = "PROCESSING";
    ActionResponseWithSingleResultURIStatusEnum["Canceled"] = "CANCELED";
    ActionResponseWithSingleResultURIStatusEnum["Complete"] = "COMPLETE";
})(ActionResponseWithSingleResultURIStatusEnum = exports.ActionResponseWithSingleResultURIStatusEnum || (exports.ActionResponseWithSingleResultURIStatusEnum = {}));
