// @hubspot/api-client/lib/codegen/crm/exports/models/PublicExportListRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicExportListRequestExportInternalValuesOptionsEnum = exports.PublicExportListRequestLanguageEnum = exports.PublicExportListRequestFormatEnum = exports.PublicExportListRequestExportTypeEnum = exports.PublicExportListRequest = void 0;
class PublicExportListRequest {
    static getAttributeTypeMap() {
        return PublicExportListRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicExportListRequest = PublicExportListRequest;
PublicExportListRequest.discriminator = undefined;
PublicExportListRequest.mapping = undefined;
PublicExportListRequest.attributeTypeMap = [
    {
        "name": "exportType",
        "baseName": "exportType",
        "type": "PublicExportListRequestExportTypeEnum",
        "format": ""
    },
    {
        "name": "format",
        "baseName": "format",
        "type": "PublicExportListRequestFormatEnum",
        "format": ""
    },
    {
        "name": "exportName",
        "baseName": "exportName",
        "type": "string",
        "format": ""
    },
    {
        "name": "objectProperties",
        "baseName": "objectProperties",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "associatedObjectType",
        "baseName": "associatedObjectType",
        "type": "string",
        "format": ""
    },
    {
        "name": "objectType",
        "baseName": "objectType",
        "type": "string",
        "format": ""
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "PublicExportListRequestLanguageEnum",
        "format": ""
    },
    {
        "name": "exportInternalValuesOptions",
        "baseName": "exportInternalValuesOptions",
        "type": "Array<PublicExportListRequestExportInternalValuesOptionsEnum>",
        "format": ""
    },
    {
        "name": "overrideAssociatedObjectsPerDefinitionPerRowLimit",
        "baseName": "overrideAssociatedObjectsPerDefinitionPerRowLimit",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "listId",
        "baseName": "listId",
        "type": "string",
        "format": ""
    }
];
var PublicExportListRequestExportTypeEnum;
(function (PublicExportListRequestExportTypeEnum) {
    PublicExportListRequestExportTypeEnum["List"] = "LIST";
})(PublicExportListRequestExportTypeEnum = exports.PublicExportListRequestExportTypeEnum || (exports.PublicExportListRequestExportTypeEnum = {}));
var PublicExportListRequestFormatEnum;
(function (PublicExportListRequestFormatEnum) {
    PublicExportListRequestFormatEnum["Xls"] = "XLS";
    PublicExportListRequestFormatEnum["Xlsx"] = "XLSX";
    PublicExportListRequestFormatEnum["Csv"] = "CSV";
})(PublicExportListRequestFormatEnum = exports.PublicExportListRequestFormatEnum || (exports.PublicExportListRequestFormatEnum = {}));
var PublicExportListRequestLanguageEnum;
(function (PublicExportListRequestLanguageEnum) {
    PublicExportListRequestLanguageEnum["En"] = "EN";
    PublicExportListRequestLanguageEnum["De"] = "DE";
    PublicExportListRequestLanguageEnum["Es"] = "ES";
    PublicExportListRequestLanguageEnum["Fr"] = "FR";
    PublicExportListRequestLanguageEnum["Ja"] = "JA";
    PublicExportListRequestLanguageEnum["Nl"] = "NL";
    PublicExportListRequestLanguageEnum["PtBr"] = "PT_BR";
    PublicExportListRequestLanguageEnum["It"] = "IT";
    PublicExportListRequestLanguageEnum["Pl"] = "PL";
    PublicExportListRequestLanguageEnum["Sv"] = "SV";
    PublicExportListRequestLanguageEnum["Fi"] = "FI";
    PublicExportListRequestLanguageEnum["ZhTw"] = "ZH_TW";
    PublicExportListRequestLanguageEnum["DaDk"] = "DA_DK";
    PublicExportListRequestLanguageEnum["No"] = "NO";
})(PublicExportListRequestLanguageEnum = exports.PublicExportListRequestLanguageEnum || (exports.PublicExportListRequestLanguageEnum = {}));
var PublicExportListRequestExportInternalValuesOptionsEnum;
(function (PublicExportListRequestExportInternalValuesOptionsEnum) {
    PublicExportListRequestExportInternalValuesOptionsEnum["Names"] = "NAMES";
    PublicExportListRequestExportInternalValuesOptionsEnum["Values"] = "VALUES";
})(PublicExportListRequestExportInternalValuesOptionsEnum = exports.PublicExportListRequestExportInternalValuesOptionsEnum || (exports.PublicExportListRequestExportInternalValuesOptionsEnum = {}));
