// @hubspot/api-client/lib/codegen/crm/exports/models/PublicExportViewRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicExportViewRequestExportInternalValuesOptionsEnum = exports.PublicExportViewRequestLanguageEnum = exports.PublicExportViewRequestFormatEnum = exports.PublicExportViewRequestExportTypeEnum = exports.PublicExportViewRequest = void 0;
class PublicExportViewRequest {
    static getAttributeTypeMap() {
        return PublicExportViewRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicExportViewRequest = PublicExportViewRequest;
PublicExportViewRequest.discriminator = undefined;
PublicExportViewRequest.mapping = undefined;
PublicExportViewRequest.attributeTypeMap = [
    {
        "name": "exportType",
        "baseName": "exportType",
        "type": "PublicExportViewRequestExportTypeEnum",
        "format": ""
    },
    {
        "name": "format",
        "baseName": "format",
        "type": "PublicExportViewRequestFormatEnum",
        "format": ""
    },
    {
        "name": "exportName",
        "baseName": "exportName",
        "type": "string",
        "format": ""
    },
    {
        "name": "objectProperties",
        "baseName": "objectProperties",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "associatedObjectType",
        "baseName": "associatedObjectType",
        "type": "string",
        "format": ""
    },
    {
        "name": "objectType",
        "baseName": "objectType",
        "type": "string",
        "format": ""
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "PublicExportViewRequestLanguageEnum",
        "format": ""
    },
    {
        "name": "exportInternalValuesOptions",
        "baseName": "exportInternalValuesOptions",
        "type": "Array<PublicExportViewRequestExportInternalValuesOptionsEnum>",
        "format": ""
    },
    {
        "name": "overrideAssociatedObjectsPerDefinitionPerRowLimit",
        "baseName": "overrideAssociatedObjectsPerDefinitionPerRowLimit",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "publicCrmSearchRequest",
        "baseName": "publicCrmSearchRequest",
        "type": "PublicCrmSearchRequest",
        "format": ""
    }
];
var PublicExportViewRequestExportTypeEnum;
(function (PublicExportViewRequestExportTypeEnum) {
    PublicExportViewRequestExportTypeEnum["View"] = "VIEW";
})(PublicExportViewRequestExportTypeEnum = exports.PublicExportViewRequestExportTypeEnum || (exports.PublicExportViewRequestExportTypeEnum = {}));
var PublicExportViewRequestFormatEnum;
(function (PublicExportViewRequestFormatEnum) {
    PublicExportViewRequestFormatEnum["Xls"] = "XLS";
    PublicExportViewRequestFormatEnum["Xlsx"] = "XLSX";
    PublicExportViewRequestFormatEnum["Csv"] = "CSV";
})(PublicExportViewRequestFormatEnum = exports.PublicExportViewRequestFormatEnum || (exports.PublicExportViewRequestFormatEnum = {}));
var PublicExportViewRequestLanguageEnum;
(function (PublicExportViewRequestLanguageEnum) {
    PublicExportViewRequestLanguageEnum["En"] = "EN";
    PublicExportViewRequestLanguageEnum["De"] = "DE";
    PublicExportViewRequestLanguageEnum["Es"] = "ES";
    PublicExportViewRequestLanguageEnum["Fr"] = "FR";
    PublicExportViewRequestLanguageEnum["Ja"] = "JA";
    PublicExportViewRequestLanguageEnum["Nl"] = "NL";
    PublicExportViewRequestLanguageEnum["PtBr"] = "PT_BR";
    PublicExportViewRequestLanguageEnum["It"] = "IT";
    PublicExportViewRequestLanguageEnum["Pl"] = "PL";
    PublicExportViewRequestLanguageEnum["Sv"] = "SV";
    PublicExportViewRequestLanguageEnum["Fi"] = "FI";
    PublicExportViewRequestLanguageEnum["ZhTw"] = "ZH_TW";
    PublicExportViewRequestLanguageEnum["DaDk"] = "DA_DK";
    PublicExportViewRequestLanguageEnum["No"] = "NO";
})(PublicExportViewRequestLanguageEnum = exports.PublicExportViewRequestLanguageEnum || (exports.PublicExportViewRequestLanguageEnum = {}));
var PublicExportViewRequestExportInternalValuesOptionsEnum;
(function (PublicExportViewRequestExportInternalValuesOptionsEnum) {
    PublicExportViewRequestExportInternalValuesOptionsEnum["Names"] = "NAMES";
    PublicExportViewRequestExportInternalValuesOptionsEnum["Values"] = "VALUES";
})(PublicExportViewRequestExportInternalValuesOptionsEnum = exports.PublicExportViewRequestExportInternalValuesOptionsEnum || (exports.PublicExportViewRequestExportInternalValuesOptionsEnum = {}));
