// @hubspot/api-client/lib/codegen/crm/objects/deal_splits/models/BatchResponseDealToDealSplits.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseDealToDealSplitsStatusEnum = exports.BatchResponseDealToDealSplits = void 0;
class BatchResponseDealToDealSplits {
    static getAttributeTypeMap() {
        return BatchResponseDealToDealSplits.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseDealToDealSplits = BatchResponseDealToDealSplits;
BatchResponseDealToDealSplits.discriminator = undefined;
BatchResponseDealToDealSplits.mapping = undefined;
BatchResponseDealToDealSplits.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<DealToDealSplits>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseDealToDealSplitsStatusEnum",
        "format": ""
    }
];
var BatchResponseDealToDealSplitsStatusEnum;
(function (BatchResponseDealToDealSplitsStatusEnum) {
    BatchResponseDealToDealSplitsStatusEnum["Pending"] = "PENDING";
    BatchResponseDealToDealSplitsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseDealToDealSplitsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseDealToDealSplitsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseDealToDealSplitsStatusEnum = exports.BatchResponseDealToDealSplitsStatusEnum || (exports.BatchResponseDealToDealSplitsStatusEnum = {}));
