// @hubspot/api-client/lib/codegen/crm/objects/deal_splits/models/BatchResponseDealToDealSplitsWithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseDealToDealSplitsWithErrorsStatusEnum = exports.BatchResponseDealToDealSplitsWithErrors = void 0;
class BatchResponseDealToDealSplitsWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseDealToDealSplitsWithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseDealToDealSplitsWithErrors = BatchResponseDealToDealSplitsWithErrors;
BatchResponseDealToDealSplitsWithErrors.discriminator = undefined;
BatchResponseDealToDealSplitsWithErrors.mapping = undefined;
BatchResponseDealToDealSplitsWithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<DealToDealSplits>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseDealToDealSplitsWithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseDealToDealSplitsWithErrorsStatusEnum;
(function (BatchResponseDealToDealSplitsWithErrorsStatusEnum) {
    BatchResponseDealToDealSplitsWithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseDealToDealSplitsWithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseDealToDealSplitsWithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseDealToDealSplitsWithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseDealToDealSplitsWithErrorsStatusEnum = exports.BatchResponseDealToDealSplitsWithErrorsStatusEnum || (exports.BatchResponseDealToDealSplitsWithErrorsStatusEnum = {}));
