// @hubspot/api-client/lib/codegen/marketing/events/models/BatchResponseMarketingEventPublicDefaultResponseV2WithErrors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum = exports.BatchResponseMarketingEventPublicDefaultResponseV2WithErrors = void 0;
class BatchResponseMarketingEventPublicDefaultResponseV2WithErrors {
    static getAttributeTypeMap() {
        return BatchResponseMarketingEventPublicDefaultResponseV2WithErrors.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseMarketingEventPublicDefaultResponseV2WithErrors = BatchResponseMarketingEventPublicDefaultResponseV2WithErrors;
BatchResponseMarketingEventPublicDefaultResponseV2WithErrors.discriminator = undefined;
BatchResponseMarketingEventPublicDefaultResponseV2WithErrors.mapping = undefined;
BatchResponseMarketingEventPublicDefaultResponseV2WithErrors.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<MarketingEventPublicDefaultResponseV2>",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum",
        "format": ""
    }
];
var BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum;
(function (BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum) {
    BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum["Pending"] = "PENDING";
    BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum["Processing"] = "PROCESSING";
    BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum["Canceled"] = "CANCELED";
    BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum["Complete"] = "COMPLETE";
})(BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum = exports.BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum || (exports.BatchResponseMarketingEventPublicDefaultResponseV2WithErrorsStatusEnum = {}));
