// @hubspot/api-client/lib/codegen/marketing/events/models/BatchResponseMarketingEventPublicDefaultResponseV2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum = exports.BatchResponseMarketingEventPublicDefaultResponseV2 = void 0;
class BatchResponseMarketingEventPublicDefaultResponseV2 {
    static getAttributeTypeMap() {
        return BatchResponseMarketingEventPublicDefaultResponseV2.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BatchResponseMarketingEventPublicDefaultResponseV2 = BatchResponseMarketingEventPublicDefaultResponseV2;
BatchResponseMarketingEventPublicDefaultResponseV2.discriminator = undefined;
BatchResponseMarketingEventPublicDefaultResponseV2.mapping = undefined;
BatchResponseMarketingEventPublicDefaultResponseV2.attributeTypeMap = [
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<MarketingEventPublicDefaultResponseV2>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum",
        "format": ""
    }
];
var BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum;
(function (BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum) {
    BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum["Pending"] = "PENDING";
    BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum["Processing"] = "PROCESSING";
    BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum["Canceled"] = "CANCELED";
    BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum["Complete"] = "COMPLETE";
})(BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum = exports.BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum || (exports.BatchResponseMarketingEventPublicDefaultResponseV2StatusEnum = {}));
