// @hubspot/api-client/lib/codegen/cms/audit_logs/types/PromiseAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromiseAuditLogsApi = void 0;
const middleware_1 = require("../middleware");
const ObservableAPI_1 = require("./ObservableAPI");
class PromiseAuditLogsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableAuditLogsApi(configuration, requestFactory, responseProcessor);
    }
    getPageWithHttpInfo(userId, eventType, objectType, objectId, after, before, limit, sort, _options) {
        var _a;
        let observableOptions;
        if (_options) {
            observableOptions = {
                baseServer: _options.baseServer,
                httpApi: _options.httpApi,
                middleware: (_a = _options.middleware) === null || _a === void 0 ? void 0 : _a.map(m => new middleware_1.PromiseMiddlewareWrapper(m)),
                middlewareMergeStrategy: _options.middlewareMergeStrategy,
                authMethods: _options.authMethods
            };
        }
        const result = this.api.getPageWithHttpInfo(userId, eventType, objectType, objectId, after, before, limit, sort, observableOptions);
        return result.toPromise();
    }
    getPage(userId, eventType, objectType, objectId, after, before, limit, sort, _options) {
        var _a;
        let observableOptions;
        if (_options) {
            observableOptions = {
                baseServer: _options.baseServer,
                httpApi: _options.httpApi,
                middleware: (_a = _options.middleware) === null || _a === void 0 ? void 0 : _a.map(m => new middleware_1.PromiseMiddlewareWrapper(m)),
                middlewareMergeStrategy: _options.middlewareMergeStrategy,
                authMethods: _options.authMethods
            };
        }
        const result = this.api.getPage(userId, eventType, objectType, objectId, after, before, limit, sort, observableOptions);
        return result.toPromise();
    }
}
exports.PromiseAuditLogsApi = PromiseAuditLogsApi;
