// @hubspot/api-client/lib/codegen/marketing/emails/models/EmailCreateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailCreateRequestSubcategoryEnum = exports.EmailCreateRequestStateEnum = exports.EmailCreateRequestLanguageEnum = exports.EmailCreateRequest = void 0;
class EmailCreateRequest {
    static getAttributeTypeMap() {
        return EmailCreateRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.EmailCreateRequest = EmailCreateRequest;
EmailCreateRequest.discriminator = undefined;
EmailCreateRequest.mapping = undefined;
EmailCreateRequest.attributeTypeMap = [
    {
        "name": "feedbackSurveyId",
        "baseName": "feedbackSurveyId",
        "type": "string",
        "format": ""
    },
    {
        "name": "rssData",
        "baseName": "rssData",
        "type": "PublicRssEmailDetails",
        "format": ""
    },
    {
        "name": "subject",
        "baseName": "subject",
        "type": "string",
        "format": ""
    },
    {
        "name": "testing",
        "baseName": "testing",
        "type": "PublicEmailTestingDetails",
        "format": ""
    },
    {
        "name": "publishDate",
        "baseName": "publishDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "EmailCreateRequestLanguageEnum",
        "format": ""
    },
    {
        "name": "businessUnitId",
        "baseName": "businessUnitId",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "content",
        "baseName": "content",
        "type": "PublicEmailContent",
        "format": ""
    },
    {
        "name": "webversion",
        "baseName": "webversion",
        "type": "PublicWebversionDetails",
        "format": ""
    },
    {
        "name": "archived",
        "baseName": "archived",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "subscriptionDetails",
        "baseName": "subscriptionDetails",
        "type": "PublicEmailSubscriptionDetails",
        "format": ""
    },
    {
        "name": "activeDomain",
        "baseName": "activeDomain",
        "type": "string",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "campaign",
        "baseName": "campaign",
        "type": "string",
        "format": ""
    },
    {
        "name": "_from",
        "baseName": "from",
        "type": "PublicEmailFromDetails",
        "format": ""
    },
    {
        "name": "jitterSendTime",
        "baseName": "jitterSendTime",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "state",
        "baseName": "state",
        "type": "EmailCreateRequestStateEnum",
        "format": ""
    },
    {
        "name": "to",
        "baseName": "to",
        "type": "PublicEmailToDetails",
        "format": ""
    },
    {
        "name": "subcategory",
        "baseName": "subcategory",
        "type": "EmailCreateRequestSubcategoryEnum",
        "format": ""
    },
    {
        "name": "sendOnPublish",
        "baseName": "sendOnPublish",
        "type": "boolean",
        "format": ""
    }
];
var EmailCreateRequestLanguageEnum;
(function (EmailCreateRequestLanguageEnum) {
    EmailCreateRequestLanguageEnum["Af"] = "af";
    EmailCreateRequestLanguageEnum["AfNa"] = "af-na";
    EmailCreateRequestLanguageEnum["AfZa"] = "af-za";
    EmailCreateRequestLanguageEnum["Agq"] = "agq";
    EmailCreateRequestLanguageEnum["AgqCm"] = "agq-cm";
    EmailCreateRequestLanguageEnum["Ak"] = "ak";
    EmailCreateRequestLanguageEnum["AkGh"] = "ak-gh";
    EmailCreateRequestLanguageEnum["Am"] = "am";
    EmailCreateRequestLanguageEnum["AmEt"] = "am-et";
    EmailCreateRequestLanguageEnum["Ann"] = "ann";
    EmailCreateRequestLanguageEnum["AnnNg"] = "ann-ng";
    EmailCreateRequestLanguageEnum["Ar"] = "ar";
    EmailCreateRequestLanguageEnum["Ar001"] = "ar-001";
    EmailCreateRequestLanguageEnum["ArAe"] = "ar-ae";
    EmailCreateRequestLanguageEnum["ArBh"] = "ar-bh";
    EmailCreateRequestLanguageEnum["ArDj"] = "ar-dj";
    EmailCreateRequestLanguageEnum["ArDz"] = "ar-dz";
    EmailCreateRequestLanguageEnum["ArEg"] = "ar-eg";
    EmailCreateRequestLanguageEnum["ArEh"] = "ar-eh";
    EmailCreateRequestLanguageEnum["ArEr"] = "ar-er";
    EmailCreateRequestLanguageEnum["ArIl"] = "ar-il";
    EmailCreateRequestLanguageEnum["ArIq"] = "ar-iq";
    EmailCreateRequestLanguageEnum["ArJo"] = "ar-jo";
    EmailCreateRequestLanguageEnum["ArKm"] = "ar-km";
    EmailCreateRequestLanguageEnum["ArKw"] = "ar-kw";
    EmailCreateRequestLanguageEnum["ArLb"] = "ar-lb";
    EmailCreateRequestLanguageEnum["ArLy"] = "ar-ly";
    EmailCreateRequestLanguageEnum["ArMa"] = "ar-ma";
    EmailCreateRequestLanguageEnum["ArMr"] = "ar-mr";
    EmailCreateRequestLanguageEnum["ArOm"] = "ar-om";
    EmailCreateRequestLanguageEnum["ArPs"] = "ar-ps";
    EmailCreateRequestLanguageEnum["ArQa"] = "ar-qa";
    EmailCreateRequestLanguageEnum["ArSa"] = "ar-sa";
    EmailCreateRequestLanguageEnum["ArSd"] = "ar-sd";
    EmailCreateRequestLanguageEnum["ArSo"] = "ar-so";
    EmailCreateRequestLanguageEnum["ArSs"] = "ar-ss";
    EmailCreateRequestLanguageEnum["ArSy"] = "ar-sy";
    EmailCreateRequestLanguageEnum["ArTd"] = "ar-td";
    EmailCreateRequestLanguageEnum["ArTn"] = "ar-tn";
    EmailCreateRequestLanguageEnum["ArYe"] = "ar-ye";
    EmailCreateRequestLanguageEnum["As"] = "as";
    EmailCreateRequestLanguageEnum["Asa"] = "asa";
    EmailCreateRequestLanguageEnum["AsaTz"] = "asa-tz";
    EmailCreateRequestLanguageEnum["Ast"] = "ast";
    EmailCreateRequestLanguageEnum["AstEs"] = "ast-es";
    EmailCreateRequestLanguageEnum["AsIn"] = "as-in";
    EmailCreateRequestLanguageEnum["Az"] = "az";
    EmailCreateRequestLanguageEnum["AzAz"] = "az-az";
    EmailCreateRequestLanguageEnum["Bas"] = "bas";
    EmailCreateRequestLanguageEnum["BasCm"] = "bas-cm";
    EmailCreateRequestLanguageEnum["Be"] = "be";
    EmailCreateRequestLanguageEnum["Bem"] = "bem";
    EmailCreateRequestLanguageEnum["BemZm"] = "bem-zm";
    EmailCreateRequestLanguageEnum["Bez"] = "bez";
    EmailCreateRequestLanguageEnum["BezTz"] = "bez-tz";
    EmailCreateRequestLanguageEnum["BeBy"] = "be-by";
    EmailCreateRequestLanguageEnum["Bg"] = "bg";
    EmailCreateRequestLanguageEnum["Bgc"] = "bgc";
    EmailCreateRequestLanguageEnum["BgcIn"] = "bgc-in";
    EmailCreateRequestLanguageEnum["BgBg"] = "bg-bg";
    EmailCreateRequestLanguageEnum["Bho"] = "bho";
    EmailCreateRequestLanguageEnum["BhoIn"] = "bho-in";
    EmailCreateRequestLanguageEnum["Bm"] = "bm";
    EmailCreateRequestLanguageEnum["BmMl"] = "bm-ml";
    EmailCreateRequestLanguageEnum["Bn"] = "bn";
    EmailCreateRequestLanguageEnum["BnBd"] = "bn-bd";
    EmailCreateRequestLanguageEnum["BnIn"] = "bn-in";
    EmailCreateRequestLanguageEnum["Bo"] = "bo";
    EmailCreateRequestLanguageEnum["BoCn"] = "bo-cn";
    EmailCreateRequestLanguageEnum["BoIn"] = "bo-in";
    EmailCreateRequestLanguageEnum["Br"] = "br";
    EmailCreateRequestLanguageEnum["Brx"] = "brx";
    EmailCreateRequestLanguageEnum["BrxIn"] = "brx-in";
    EmailCreateRequestLanguageEnum["BrFr"] = "br-fr";
    EmailCreateRequestLanguageEnum["Bs"] = "bs";
    EmailCreateRequestLanguageEnum["BsBa"] = "bs-ba";
    EmailCreateRequestLanguageEnum["Ca"] = "ca";
    EmailCreateRequestLanguageEnum["CaAd"] = "ca-ad";
    EmailCreateRequestLanguageEnum["CaEs"] = "ca-es";
    EmailCreateRequestLanguageEnum["CaFr"] = "ca-fr";
    EmailCreateRequestLanguageEnum["CaIt"] = "ca-it";
    EmailCreateRequestLanguageEnum["Ccp"] = "ccp";
    EmailCreateRequestLanguageEnum["CcpBd"] = "ccp-bd";
    EmailCreateRequestLanguageEnum["CcpIn"] = "ccp-in";
    EmailCreateRequestLanguageEnum["Ce"] = "ce";
    EmailCreateRequestLanguageEnum["Ceb"] = "ceb";
    EmailCreateRequestLanguageEnum["CebPh"] = "ceb-ph";
    EmailCreateRequestLanguageEnum["CeRu"] = "ce-ru";
    EmailCreateRequestLanguageEnum["Cgg"] = "cgg";
    EmailCreateRequestLanguageEnum["CggUg"] = "cgg-ug";
    EmailCreateRequestLanguageEnum["Chr"] = "chr";
    EmailCreateRequestLanguageEnum["ChrUs"] = "chr-us";
    EmailCreateRequestLanguageEnum["Ckb"] = "ckb";
    EmailCreateRequestLanguageEnum["CkbIq"] = "ckb-iq";
    EmailCreateRequestLanguageEnum["CkbIr"] = "ckb-ir";
    EmailCreateRequestLanguageEnum["Cs"] = "cs";
    EmailCreateRequestLanguageEnum["CsCz"] = "cs-cz";
    EmailCreateRequestLanguageEnum["Cu"] = "cu";
    EmailCreateRequestLanguageEnum["CuRu"] = "cu-ru";
    EmailCreateRequestLanguageEnum["Cv"] = "cv";
    EmailCreateRequestLanguageEnum["CvRu"] = "cv-ru";
    EmailCreateRequestLanguageEnum["Cy"] = "cy";
    EmailCreateRequestLanguageEnum["CyGb"] = "cy-gb";
    EmailCreateRequestLanguageEnum["Da"] = "da";
    EmailCreateRequestLanguageEnum["Dav"] = "dav";
    EmailCreateRequestLanguageEnum["DavKe"] = "dav-ke";
    EmailCreateRequestLanguageEnum["DaDk"] = "da-dk";
    EmailCreateRequestLanguageEnum["DaGl"] = "da-gl";
    EmailCreateRequestLanguageEnum["De"] = "de";
    EmailCreateRequestLanguageEnum["DeAt"] = "de-at";
    EmailCreateRequestLanguageEnum["DeBe"] = "de-be";
    EmailCreateRequestLanguageEnum["DeCh"] = "de-ch";
    EmailCreateRequestLanguageEnum["DeDe"] = "de-de";
    EmailCreateRequestLanguageEnum["DeGr"] = "de-gr";
    EmailCreateRequestLanguageEnum["DeIt"] = "de-it";
    EmailCreateRequestLanguageEnum["DeLi"] = "de-li";
    EmailCreateRequestLanguageEnum["DeLu"] = "de-lu";
    EmailCreateRequestLanguageEnum["Dje"] = "dje";
    EmailCreateRequestLanguageEnum["DjeNe"] = "dje-ne";
    EmailCreateRequestLanguageEnum["Doi"] = "doi";
    EmailCreateRequestLanguageEnum["DoiIn"] = "doi-in";
    EmailCreateRequestLanguageEnum["Dsb"] = "dsb";
    EmailCreateRequestLanguageEnum["DsbDe"] = "dsb-de";
    EmailCreateRequestLanguageEnum["Dua"] = "dua";
    EmailCreateRequestLanguageEnum["DuaCm"] = "dua-cm";
    EmailCreateRequestLanguageEnum["Dyo"] = "dyo";
    EmailCreateRequestLanguageEnum["DyoSn"] = "dyo-sn";
    EmailCreateRequestLanguageEnum["Dz"] = "dz";
    EmailCreateRequestLanguageEnum["DzBt"] = "dz-bt";
    EmailCreateRequestLanguageEnum["Ebu"] = "ebu";
    EmailCreateRequestLanguageEnum["EbuKe"] = "ebu-ke";
    EmailCreateRequestLanguageEnum["Ee"] = "ee";
    EmailCreateRequestLanguageEnum["EeGh"] = "ee-gh";
    EmailCreateRequestLanguageEnum["EeTg"] = "ee-tg";
    EmailCreateRequestLanguageEnum["El"] = "el";
    EmailCreateRequestLanguageEnum["ElCy"] = "el-cy";
    EmailCreateRequestLanguageEnum["ElGr"] = "el-gr";
    EmailCreateRequestLanguageEnum["En"] = "en";
    EmailCreateRequestLanguageEnum["En001"] = "en-001";
    EmailCreateRequestLanguageEnum["En150"] = "en-150";
    EmailCreateRequestLanguageEnum["EnAe"] = "en-ae";
    EmailCreateRequestLanguageEnum["EnAg"] = "en-ag";
    EmailCreateRequestLanguageEnum["EnAi"] = "en-ai";
    EmailCreateRequestLanguageEnum["EnAs"] = "en-as";
    EmailCreateRequestLanguageEnum["EnAt"] = "en-at";
    EmailCreateRequestLanguageEnum["EnAu"] = "en-au";
    EmailCreateRequestLanguageEnum["EnBb"] = "en-bb";
    EmailCreateRequestLanguageEnum["EnBe"] = "en-be";
    EmailCreateRequestLanguageEnum["EnBi"] = "en-bi";
    EmailCreateRequestLanguageEnum["EnBm"] = "en-bm";
    EmailCreateRequestLanguageEnum["EnBs"] = "en-bs";
    EmailCreateRequestLanguageEnum["EnBw"] = "en-bw";
    EmailCreateRequestLanguageEnum["EnBz"] = "en-bz";
    EmailCreateRequestLanguageEnum["EnCa"] = "en-ca";
    EmailCreateRequestLanguageEnum["EnCc"] = "en-cc";
    EmailCreateRequestLanguageEnum["EnCh"] = "en-ch";
    EmailCreateRequestLanguageEnum["EnCk"] = "en-ck";
    EmailCreateRequestLanguageEnum["EnCm"] = "en-cm";
    EmailCreateRequestLanguageEnum["EnCn"] = "en-cn";
    EmailCreateRequestLanguageEnum["EnCx"] = "en-cx";
    EmailCreateRequestLanguageEnum["EnCy"] = "en-cy";
    EmailCreateRequestLanguageEnum["EnDe"] = "en-de";
    EmailCreateRequestLanguageEnum["EnDg"] = "en-dg";
    EmailCreateRequestLanguageEnum["EnDk"] = "en-dk";
    EmailCreateRequestLanguageEnum["EnDm"] = "en-dm";
    EmailCreateRequestLanguageEnum["EnEe"] = "en-ee";
    EmailCreateRequestLanguageEnum["EnEr"] = "en-er";
    EmailCreateRequestLanguageEnum["EnEs"] = "en-es";
    EmailCreateRequestLanguageEnum["EnFi"] = "en-fi";
    EmailCreateRequestLanguageEnum["EnFj"] = "en-fj";
    EmailCreateRequestLanguageEnum["EnFk"] = "en-fk";
    EmailCreateRequestLanguageEnum["EnFm"] = "en-fm";
    EmailCreateRequestLanguageEnum["EnFr"] = "en-fr";
    EmailCreateRequestLanguageEnum["EnGb"] = "en-gb";
    EmailCreateRequestLanguageEnum["EnGd"] = "en-gd";
    EmailCreateRequestLanguageEnum["EnGg"] = "en-gg";
    EmailCreateRequestLanguageEnum["EnGh"] = "en-gh";
    EmailCreateRequestLanguageEnum["EnGi"] = "en-gi";
    EmailCreateRequestLanguageEnum["EnGm"] = "en-gm";
    EmailCreateRequestLanguageEnum["EnGu"] = "en-gu";
    EmailCreateRequestLanguageEnum["EnGy"] = "en-gy";
    EmailCreateRequestLanguageEnum["EnHk"] = "en-hk";
    EmailCreateRequestLanguageEnum["EnIe"] = "en-ie";
    EmailCreateRequestLanguageEnum["EnIl"] = "en-il";
    EmailCreateRequestLanguageEnum["EnIm"] = "en-im";
    EmailCreateRequestLanguageEnum["EnIn"] = "en-in";
    EmailCreateRequestLanguageEnum["EnIo"] = "en-io";
    EmailCreateRequestLanguageEnum["EnJe"] = "en-je";
    EmailCreateRequestLanguageEnum["EnJm"] = "en-jm";
    EmailCreateRequestLanguageEnum["EnKe"] = "en-ke";
    EmailCreateRequestLanguageEnum["EnKi"] = "en-ki";
    EmailCreateRequestLanguageEnum["EnKn"] = "en-kn";
    EmailCreateRequestLanguageEnum["EnKy"] = "en-ky";
    EmailCreateRequestLanguageEnum["EnLc"] = "en-lc";
    EmailCreateRequestLanguageEnum["EnLr"] = "en-lr";
    EmailCreateRequestLanguageEnum["EnLs"] = "en-ls";
    EmailCreateRequestLanguageEnum["EnLu"] = "en-lu";
    EmailCreateRequestLanguageEnum["EnMg"] = "en-mg";
    EmailCreateRequestLanguageEnum["EnMh"] = "en-mh";
    EmailCreateRequestLanguageEnum["EnMo"] = "en-mo";
    EmailCreateRequestLanguageEnum["EnMp"] = "en-mp";
    EmailCreateRequestLanguageEnum["EnMs"] = "en-ms";
    EmailCreateRequestLanguageEnum["EnMt"] = "en-mt";
    EmailCreateRequestLanguageEnum["EnMu"] = "en-mu";
    EmailCreateRequestLanguageEnum["EnMv"] = "en-mv";
    EmailCreateRequestLanguageEnum["EnMw"] = "en-mw";
    EmailCreateRequestLanguageEnum["EnMx"] = "en-mx";
    EmailCreateRequestLanguageEnum["EnMy"] = "en-my";
    EmailCreateRequestLanguageEnum["EnNa"] = "en-na";
    EmailCreateRequestLanguageEnum["EnNf"] = "en-nf";
    EmailCreateRequestLanguageEnum["EnNg"] = "en-ng";
    EmailCreateRequestLanguageEnum["EnNl"] = "en-nl";
    EmailCreateRequestLanguageEnum["EnNr"] = "en-nr";
    EmailCreateRequestLanguageEnum["EnNu"] = "en-nu";
    EmailCreateRequestLanguageEnum["EnNz"] = "en-nz";
    EmailCreateRequestLanguageEnum["EnPg"] = "en-pg";
    EmailCreateRequestLanguageEnum["EnPh"] = "en-ph";
    EmailCreateRequestLanguageEnum["EnPk"] = "en-pk";
    EmailCreateRequestLanguageEnum["EnPn"] = "en-pn";
    EmailCreateRequestLanguageEnum["EnPr"] = "en-pr";
    EmailCreateRequestLanguageEnum["EnPw"] = "en-pw";
    EmailCreateRequestLanguageEnum["EnRw"] = "en-rw";
    EmailCreateRequestLanguageEnum["EnSb"] = "en-sb";
    EmailCreateRequestLanguageEnum["EnSc"] = "en-sc";
    EmailCreateRequestLanguageEnum["EnSd"] = "en-sd";
    EmailCreateRequestLanguageEnum["EnSe"] = "en-se";
    EmailCreateRequestLanguageEnum["EnSg"] = "en-sg";
    EmailCreateRequestLanguageEnum["EnSh"] = "en-sh";
    EmailCreateRequestLanguageEnum["EnSi"] = "en-si";
    EmailCreateRequestLanguageEnum["EnSl"] = "en-sl";
    EmailCreateRequestLanguageEnum["EnSs"] = "en-ss";
    EmailCreateRequestLanguageEnum["EnSx"] = "en-sx";
    EmailCreateRequestLanguageEnum["EnSz"] = "en-sz";
    EmailCreateRequestLanguageEnum["EnTc"] = "en-tc";
    EmailCreateRequestLanguageEnum["EnTk"] = "en-tk";
    EmailCreateRequestLanguageEnum["EnTn"] = "en-tn";
    EmailCreateRequestLanguageEnum["EnTo"] = "en-to";
    EmailCreateRequestLanguageEnum["EnTt"] = "en-tt";
    EmailCreateRequestLanguageEnum["EnTv"] = "en-tv";
    EmailCreateRequestLanguageEnum["EnTz"] = "en-tz";
    EmailCreateRequestLanguageEnum["EnUg"] = "en-ug";
    EmailCreateRequestLanguageEnum["EnUm"] = "en-um";
    EmailCreateRequestLanguageEnum["EnUs"] = "en-us";
    EmailCreateRequestLanguageEnum["EnVc"] = "en-vc";
    EmailCreateRequestLanguageEnum["EnVg"] = "en-vg";
    EmailCreateRequestLanguageEnum["EnVi"] = "en-vi";
    EmailCreateRequestLanguageEnum["EnVu"] = "en-vu";
    EmailCreateRequestLanguageEnum["EnWs"] = "en-ws";
    EmailCreateRequestLanguageEnum["EnZa"] = "en-za";
    EmailCreateRequestLanguageEnum["EnZm"] = "en-zm";
    EmailCreateRequestLanguageEnum["EnZw"] = "en-zw";
    EmailCreateRequestLanguageEnum["Eo"] = "eo";
    EmailCreateRequestLanguageEnum["Eo001"] = "eo-001";
    EmailCreateRequestLanguageEnum["Es"] = "es";
    EmailCreateRequestLanguageEnum["Es419"] = "es-419";
    EmailCreateRequestLanguageEnum["EsAr"] = "es-ar";
    EmailCreateRequestLanguageEnum["EsBo"] = "es-bo";
    EmailCreateRequestLanguageEnum["EsBr"] = "es-br";
    EmailCreateRequestLanguageEnum["EsBz"] = "es-bz";
    EmailCreateRequestLanguageEnum["EsCl"] = "es-cl";
    EmailCreateRequestLanguageEnum["EsCo"] = "es-co";
    EmailCreateRequestLanguageEnum["EsCr"] = "es-cr";
    EmailCreateRequestLanguageEnum["EsCu"] = "es-cu";
    EmailCreateRequestLanguageEnum["EsDo"] = "es-do";
    EmailCreateRequestLanguageEnum["EsEa"] = "es-ea";
    EmailCreateRequestLanguageEnum["EsEc"] = "es-ec";
    EmailCreateRequestLanguageEnum["EsEs"] = "es-es";
    EmailCreateRequestLanguageEnum["EsGq"] = "es-gq";
    EmailCreateRequestLanguageEnum["EsGt"] = "es-gt";
    EmailCreateRequestLanguageEnum["EsHn"] = "es-hn";
    EmailCreateRequestLanguageEnum["EsIc"] = "es-ic";
    EmailCreateRequestLanguageEnum["EsMx"] = "es-mx";
    EmailCreateRequestLanguageEnum["EsNi"] = "es-ni";
    EmailCreateRequestLanguageEnum["EsPa"] = "es-pa";
    EmailCreateRequestLanguageEnum["EsPe"] = "es-pe";
    EmailCreateRequestLanguageEnum["EsPh"] = "es-ph";
    EmailCreateRequestLanguageEnum["EsPr"] = "es-pr";
    EmailCreateRequestLanguageEnum["EsPy"] = "es-py";
    EmailCreateRequestLanguageEnum["EsSv"] = "es-sv";
    EmailCreateRequestLanguageEnum["EsUs"] = "es-us";
    EmailCreateRequestLanguageEnum["EsUy"] = "es-uy";
    EmailCreateRequestLanguageEnum["EsVe"] = "es-ve";
    EmailCreateRequestLanguageEnum["Et"] = "et";
    EmailCreateRequestLanguageEnum["EtEe"] = "et-ee";
    EmailCreateRequestLanguageEnum["Eu"] = "eu";
    EmailCreateRequestLanguageEnum["EuEs"] = "eu-es";
    EmailCreateRequestLanguageEnum["Ewo"] = "ewo";
    EmailCreateRequestLanguageEnum["EwoCm"] = "ewo-cm";
    EmailCreateRequestLanguageEnum["Fa"] = "fa";
    EmailCreateRequestLanguageEnum["FaAf"] = "fa-af";
    EmailCreateRequestLanguageEnum["FaIr"] = "fa-ir";
    EmailCreateRequestLanguageEnum["Ff"] = "ff";
    EmailCreateRequestLanguageEnum["FfBf"] = "ff-bf";
    EmailCreateRequestLanguageEnum["FfCm"] = "ff-cm";
    EmailCreateRequestLanguageEnum["FfGh"] = "ff-gh";
    EmailCreateRequestLanguageEnum["FfGm"] = "ff-gm";
    EmailCreateRequestLanguageEnum["FfGn"] = "ff-gn";
    EmailCreateRequestLanguageEnum["FfGw"] = "ff-gw";
    EmailCreateRequestLanguageEnum["FfLr"] = "ff-lr";
    EmailCreateRequestLanguageEnum["FfMr"] = "ff-mr";
    EmailCreateRequestLanguageEnum["FfNe"] = "ff-ne";
    EmailCreateRequestLanguageEnum["FfNg"] = "ff-ng";
    EmailCreateRequestLanguageEnum["FfSl"] = "ff-sl";
    EmailCreateRequestLanguageEnum["FfSn"] = "ff-sn";
    EmailCreateRequestLanguageEnum["Fi"] = "fi";
    EmailCreateRequestLanguageEnum["Fil"] = "fil";
    EmailCreateRequestLanguageEnum["FilPh"] = "fil-ph";
    EmailCreateRequestLanguageEnum["FiFi"] = "fi-fi";
    EmailCreateRequestLanguageEnum["Fo"] = "fo";
    EmailCreateRequestLanguageEnum["FoDk"] = "fo-dk";
    EmailCreateRequestLanguageEnum["FoFo"] = "fo-fo";
    EmailCreateRequestLanguageEnum["Fr"] = "fr";
    EmailCreateRequestLanguageEnum["Frr"] = "frr";
    EmailCreateRequestLanguageEnum["FrrDe"] = "frr-de";
    EmailCreateRequestLanguageEnum["FrBe"] = "fr-be";
    EmailCreateRequestLanguageEnum["FrBf"] = "fr-bf";
    EmailCreateRequestLanguageEnum["FrBi"] = "fr-bi";
    EmailCreateRequestLanguageEnum["FrBj"] = "fr-bj";
    EmailCreateRequestLanguageEnum["FrBl"] = "fr-bl";
    EmailCreateRequestLanguageEnum["FrCa"] = "fr-ca";
    EmailCreateRequestLanguageEnum["FrCd"] = "fr-cd";
    EmailCreateRequestLanguageEnum["FrCf"] = "fr-cf";
    EmailCreateRequestLanguageEnum["FrCg"] = "fr-cg";
    EmailCreateRequestLanguageEnum["FrCh"] = "fr-ch";
    EmailCreateRequestLanguageEnum["FrCi"] = "fr-ci";
    EmailCreateRequestLanguageEnum["FrCm"] = "fr-cm";
    EmailCreateRequestLanguageEnum["FrDj"] = "fr-dj";
    EmailCreateRequestLanguageEnum["FrDz"] = "fr-dz";
    EmailCreateRequestLanguageEnum["FrFr"] = "fr-fr";
    EmailCreateRequestLanguageEnum["FrGa"] = "fr-ga";
    EmailCreateRequestLanguageEnum["FrGf"] = "fr-gf";
    EmailCreateRequestLanguageEnum["FrGn"] = "fr-gn";
    EmailCreateRequestLanguageEnum["FrGp"] = "fr-gp";
    EmailCreateRequestLanguageEnum["FrGq"] = "fr-gq";
    EmailCreateRequestLanguageEnum["FrHt"] = "fr-ht";
    EmailCreateRequestLanguageEnum["FrKm"] = "fr-km";
    EmailCreateRequestLanguageEnum["FrLu"] = "fr-lu";
    EmailCreateRequestLanguageEnum["FrMa"] = "fr-ma";
    EmailCreateRequestLanguageEnum["FrMc"] = "fr-mc";
    EmailCreateRequestLanguageEnum["FrMf"] = "fr-mf";
    EmailCreateRequestLanguageEnum["FrMg"] = "fr-mg";
    EmailCreateRequestLanguageEnum["FrMl"] = "fr-ml";
    EmailCreateRequestLanguageEnum["FrMq"] = "fr-mq";
    EmailCreateRequestLanguageEnum["FrMr"] = "fr-mr";
    EmailCreateRequestLanguageEnum["FrMu"] = "fr-mu";
    EmailCreateRequestLanguageEnum["FrNc"] = "fr-nc";
    EmailCreateRequestLanguageEnum["FrNe"] = "fr-ne";
    EmailCreateRequestLanguageEnum["FrPf"] = "fr-pf";
    EmailCreateRequestLanguageEnum["FrPm"] = "fr-pm";
    EmailCreateRequestLanguageEnum["FrRe"] = "fr-re";
    EmailCreateRequestLanguageEnum["FrRw"] = "fr-rw";
    EmailCreateRequestLanguageEnum["FrSc"] = "fr-sc";
    EmailCreateRequestLanguageEnum["FrSn"] = "fr-sn";
    EmailCreateRequestLanguageEnum["FrSy"] = "fr-sy";
    EmailCreateRequestLanguageEnum["FrTd"] = "fr-td";
    EmailCreateRequestLanguageEnum["FrTg"] = "fr-tg";
    EmailCreateRequestLanguageEnum["FrTn"] = "fr-tn";
    EmailCreateRequestLanguageEnum["FrVu"] = "fr-vu";
    EmailCreateRequestLanguageEnum["FrWf"] = "fr-wf";
    EmailCreateRequestLanguageEnum["FrYt"] = "fr-yt";
    EmailCreateRequestLanguageEnum["Fur"] = "fur";
    EmailCreateRequestLanguageEnum["FurIt"] = "fur-it";
    EmailCreateRequestLanguageEnum["Fy"] = "fy";
    EmailCreateRequestLanguageEnum["FyNl"] = "fy-nl";
    EmailCreateRequestLanguageEnum["Ga"] = "ga";
    EmailCreateRequestLanguageEnum["GaGb"] = "ga-gb";
    EmailCreateRequestLanguageEnum["GaIe"] = "ga-ie";
    EmailCreateRequestLanguageEnum["Gd"] = "gd";
    EmailCreateRequestLanguageEnum["GdGb"] = "gd-gb";
    EmailCreateRequestLanguageEnum["Gl"] = "gl";
    EmailCreateRequestLanguageEnum["GlEs"] = "gl-es";
    EmailCreateRequestLanguageEnum["Gsw"] = "gsw";
    EmailCreateRequestLanguageEnum["GswCh"] = "gsw-ch";
    EmailCreateRequestLanguageEnum["GswFr"] = "gsw-fr";
    EmailCreateRequestLanguageEnum["GswLi"] = "gsw-li";
    EmailCreateRequestLanguageEnum["Gu"] = "gu";
    EmailCreateRequestLanguageEnum["Guz"] = "guz";
    EmailCreateRequestLanguageEnum["GuzKe"] = "guz-ke";
    EmailCreateRequestLanguageEnum["GuIn"] = "gu-in";
    EmailCreateRequestLanguageEnum["Gv"] = "gv";
    EmailCreateRequestLanguageEnum["GvIm"] = "gv-im";
    EmailCreateRequestLanguageEnum["Ha"] = "ha";
    EmailCreateRequestLanguageEnum["Haw"] = "haw";
    EmailCreateRequestLanguageEnum["HawUs"] = "haw-us";
    EmailCreateRequestLanguageEnum["HaGh"] = "ha-gh";
    EmailCreateRequestLanguageEnum["HaNe"] = "ha-ne";
    EmailCreateRequestLanguageEnum["HaNg"] = "ha-ng";
    EmailCreateRequestLanguageEnum["He"] = "he";
    EmailCreateRequestLanguageEnum["HeIl"] = "he-il";
    EmailCreateRequestLanguageEnum["Hi"] = "hi";
    EmailCreateRequestLanguageEnum["HiIn"] = "hi-in";
    EmailCreateRequestLanguageEnum["Hr"] = "hr";
    EmailCreateRequestLanguageEnum["HrBa"] = "hr-ba";
    EmailCreateRequestLanguageEnum["HrHr"] = "hr-hr";
    EmailCreateRequestLanguageEnum["Hsb"] = "hsb";
    EmailCreateRequestLanguageEnum["HsbDe"] = "hsb-de";
    EmailCreateRequestLanguageEnum["Hu"] = "hu";
    EmailCreateRequestLanguageEnum["HuHu"] = "hu-hu";
    EmailCreateRequestLanguageEnum["Hy"] = "hy";
    EmailCreateRequestLanguageEnum["HyAm"] = "hy-am";
    EmailCreateRequestLanguageEnum["Ia"] = "ia";
    EmailCreateRequestLanguageEnum["Ia001"] = "ia-001";
    EmailCreateRequestLanguageEnum["Id"] = "id";
    EmailCreateRequestLanguageEnum["Ig"] = "ig";
    EmailCreateRequestLanguageEnum["IgNg"] = "ig-ng";
    EmailCreateRequestLanguageEnum["Ii"] = "ii";
    EmailCreateRequestLanguageEnum["IiCn"] = "ii-cn";
    EmailCreateRequestLanguageEnum["IdId"] = "id-id";
    EmailCreateRequestLanguageEnum["Is"] = "is";
    EmailCreateRequestLanguageEnum["IsIs"] = "is-is";
    EmailCreateRequestLanguageEnum["It"] = "it";
    EmailCreateRequestLanguageEnum["ItCh"] = "it-ch";
    EmailCreateRequestLanguageEnum["ItIt"] = "it-it";
    EmailCreateRequestLanguageEnum["ItSm"] = "it-sm";
    EmailCreateRequestLanguageEnum["ItVa"] = "it-va";
    EmailCreateRequestLanguageEnum["Ja"] = "ja";
    EmailCreateRequestLanguageEnum["JaJp"] = "ja-jp";
    EmailCreateRequestLanguageEnum["Jgo"] = "jgo";
    EmailCreateRequestLanguageEnum["JgoCm"] = "jgo-cm";
    EmailCreateRequestLanguageEnum["Yi"] = "yi";
    EmailCreateRequestLanguageEnum["Yi001"] = "yi-001";
    EmailCreateRequestLanguageEnum["Jmc"] = "jmc";
    EmailCreateRequestLanguageEnum["JmcTz"] = "jmc-tz";
    EmailCreateRequestLanguageEnum["Jv"] = "jv";
    EmailCreateRequestLanguageEnum["JvId"] = "jv-id";
    EmailCreateRequestLanguageEnum["Ka"] = "ka";
    EmailCreateRequestLanguageEnum["Kab"] = "kab";
    EmailCreateRequestLanguageEnum["KabDz"] = "kab-dz";
    EmailCreateRequestLanguageEnum["Kam"] = "kam";
    EmailCreateRequestLanguageEnum["KamKe"] = "kam-ke";
    EmailCreateRequestLanguageEnum["KaGe"] = "ka-ge";
    EmailCreateRequestLanguageEnum["Kde"] = "kde";
    EmailCreateRequestLanguageEnum["KdeTz"] = "kde-tz";
    EmailCreateRequestLanguageEnum["Kea"] = "kea";
    EmailCreateRequestLanguageEnum["KeaCv"] = "kea-cv";
    EmailCreateRequestLanguageEnum["Kgp"] = "kgp";
    EmailCreateRequestLanguageEnum["KgpBr"] = "kgp-br";
    EmailCreateRequestLanguageEnum["Khq"] = "khq";
    EmailCreateRequestLanguageEnum["KhqMl"] = "khq-ml";
    EmailCreateRequestLanguageEnum["Ki"] = "ki";
    EmailCreateRequestLanguageEnum["KiKe"] = "ki-ke";
    EmailCreateRequestLanguageEnum["Kk"] = "kk";
    EmailCreateRequestLanguageEnum["Kkj"] = "kkj";
    EmailCreateRequestLanguageEnum["KkjCm"] = "kkj-cm";
    EmailCreateRequestLanguageEnum["KkKz"] = "kk-kz";
    EmailCreateRequestLanguageEnum["Kl"] = "kl";
    EmailCreateRequestLanguageEnum["Kln"] = "kln";
    EmailCreateRequestLanguageEnum["KlnKe"] = "kln-ke";
    EmailCreateRequestLanguageEnum["KlGl"] = "kl-gl";
    EmailCreateRequestLanguageEnum["Km"] = "km";
    EmailCreateRequestLanguageEnum["KmKh"] = "km-kh";
    EmailCreateRequestLanguageEnum["Kn"] = "kn";
    EmailCreateRequestLanguageEnum["KnIn"] = "kn-in";
    EmailCreateRequestLanguageEnum["Ko"] = "ko";
    EmailCreateRequestLanguageEnum["Kok"] = "kok";
    EmailCreateRequestLanguageEnum["KokIn"] = "kok-in";
    EmailCreateRequestLanguageEnum["KoKp"] = "ko-kp";
    EmailCreateRequestLanguageEnum["KoKr"] = "ko-kr";
    EmailCreateRequestLanguageEnum["Ks"] = "ks";
    EmailCreateRequestLanguageEnum["Ksb"] = "ksb";
    EmailCreateRequestLanguageEnum["KsbTz"] = "ksb-tz";
    EmailCreateRequestLanguageEnum["Ksf"] = "ksf";
    EmailCreateRequestLanguageEnum["KsfCm"] = "ksf-cm";
    EmailCreateRequestLanguageEnum["Ksh"] = "ksh";
    EmailCreateRequestLanguageEnum["KshDe"] = "ksh-de";
    EmailCreateRequestLanguageEnum["KsIn"] = "ks-in";
    EmailCreateRequestLanguageEnum["Ku"] = "ku";
    EmailCreateRequestLanguageEnum["KuTr"] = "ku-tr";
    EmailCreateRequestLanguageEnum["Kw"] = "kw";
    EmailCreateRequestLanguageEnum["KwGb"] = "kw-gb";
    EmailCreateRequestLanguageEnum["Ky"] = "ky";
    EmailCreateRequestLanguageEnum["KyKg"] = "ky-kg";
    EmailCreateRequestLanguageEnum["Lag"] = "lag";
    EmailCreateRequestLanguageEnum["LagTz"] = "lag-tz";
    EmailCreateRequestLanguageEnum["Lb"] = "lb";
    EmailCreateRequestLanguageEnum["LbLu"] = "lb-lu";
    EmailCreateRequestLanguageEnum["Lg"] = "lg";
    EmailCreateRequestLanguageEnum["LgUg"] = "lg-ug";
    EmailCreateRequestLanguageEnum["Lkt"] = "lkt";
    EmailCreateRequestLanguageEnum["LktUs"] = "lkt-us";
    EmailCreateRequestLanguageEnum["Ln"] = "ln";
    EmailCreateRequestLanguageEnum["LnAo"] = "ln-ao";
    EmailCreateRequestLanguageEnum["LnCd"] = "ln-cd";
    EmailCreateRequestLanguageEnum["LnCf"] = "ln-cf";
    EmailCreateRequestLanguageEnum["LnCg"] = "ln-cg";
    EmailCreateRequestLanguageEnum["Lo"] = "lo";
    EmailCreateRequestLanguageEnum["LoLa"] = "lo-la";
    EmailCreateRequestLanguageEnum["Lrc"] = "lrc";
    EmailCreateRequestLanguageEnum["LrcIq"] = "lrc-iq";
    EmailCreateRequestLanguageEnum["LrcIr"] = "lrc-ir";
    EmailCreateRequestLanguageEnum["Lt"] = "lt";
    EmailCreateRequestLanguageEnum["LtLt"] = "lt-lt";
    EmailCreateRequestLanguageEnum["Lu"] = "lu";
    EmailCreateRequestLanguageEnum["Luo"] = "luo";
    EmailCreateRequestLanguageEnum["LuoKe"] = "luo-ke";
    EmailCreateRequestLanguageEnum["Luy"] = "luy";
    EmailCreateRequestLanguageEnum["LuyKe"] = "luy-ke";
    EmailCreateRequestLanguageEnum["LuCd"] = "lu-cd";
    EmailCreateRequestLanguageEnum["Lv"] = "lv";
    EmailCreateRequestLanguageEnum["LvLv"] = "lv-lv";
    EmailCreateRequestLanguageEnum["Mai"] = "mai";
    EmailCreateRequestLanguageEnum["MaiIn"] = "mai-in";
    EmailCreateRequestLanguageEnum["Mas"] = "mas";
    EmailCreateRequestLanguageEnum["MasKe"] = "mas-ke";
    EmailCreateRequestLanguageEnum["MasTz"] = "mas-tz";
    EmailCreateRequestLanguageEnum["Mdf"] = "mdf";
    EmailCreateRequestLanguageEnum["MdfRu"] = "mdf-ru";
    EmailCreateRequestLanguageEnum["Mer"] = "mer";
    EmailCreateRequestLanguageEnum["MerKe"] = "mer-ke";
    EmailCreateRequestLanguageEnum["Mfe"] = "mfe";
    EmailCreateRequestLanguageEnum["MfeMu"] = "mfe-mu";
    EmailCreateRequestLanguageEnum["Mg"] = "mg";
    EmailCreateRequestLanguageEnum["Mgh"] = "mgh";
    EmailCreateRequestLanguageEnum["MghMz"] = "mgh-mz";
    EmailCreateRequestLanguageEnum["Mgo"] = "mgo";
    EmailCreateRequestLanguageEnum["MgoCm"] = "mgo-cm";
    EmailCreateRequestLanguageEnum["MgMg"] = "mg-mg";
    EmailCreateRequestLanguageEnum["Mi"] = "mi";
    EmailCreateRequestLanguageEnum["MiNz"] = "mi-nz";
    EmailCreateRequestLanguageEnum["Mk"] = "mk";
    EmailCreateRequestLanguageEnum["MkMk"] = "mk-mk";
    EmailCreateRequestLanguageEnum["Ml"] = "ml";
    EmailCreateRequestLanguageEnum["MlIn"] = "ml-in";
    EmailCreateRequestLanguageEnum["Mn"] = "mn";
    EmailCreateRequestLanguageEnum["Mni"] = "mni";
    EmailCreateRequestLanguageEnum["MniIn"] = "mni-in";
    EmailCreateRequestLanguageEnum["MnMn"] = "mn-mn";
    EmailCreateRequestLanguageEnum["Mr"] = "mr";
    EmailCreateRequestLanguageEnum["MrIn"] = "mr-in";
    EmailCreateRequestLanguageEnum["Ms"] = "ms";
    EmailCreateRequestLanguageEnum["MsBn"] = "ms-bn";
    EmailCreateRequestLanguageEnum["MsId"] = "ms-id";
    EmailCreateRequestLanguageEnum["MsMy"] = "ms-my";
    EmailCreateRequestLanguageEnum["MsSg"] = "ms-sg";
    EmailCreateRequestLanguageEnum["Mt"] = "mt";
    EmailCreateRequestLanguageEnum["MtMt"] = "mt-mt";
    EmailCreateRequestLanguageEnum["Mua"] = "mua";
    EmailCreateRequestLanguageEnum["MuaCm"] = "mua-cm";
    EmailCreateRequestLanguageEnum["My"] = "my";
    EmailCreateRequestLanguageEnum["MyMm"] = "my-mm";
    EmailCreateRequestLanguageEnum["Mzn"] = "mzn";
    EmailCreateRequestLanguageEnum["MznIr"] = "mzn-ir";
    EmailCreateRequestLanguageEnum["Naq"] = "naq";
    EmailCreateRequestLanguageEnum["NaqNa"] = "naq-na";
    EmailCreateRequestLanguageEnum["Nb"] = "nb";
    EmailCreateRequestLanguageEnum["NbNo"] = "nb-no";
    EmailCreateRequestLanguageEnum["NbSj"] = "nb-sj";
    EmailCreateRequestLanguageEnum["Nd"] = "nd";
    EmailCreateRequestLanguageEnum["Nds"] = "nds";
    EmailCreateRequestLanguageEnum["NdsDe"] = "nds-de";
    EmailCreateRequestLanguageEnum["NdsNl"] = "nds-nl";
    EmailCreateRequestLanguageEnum["NdZw"] = "nd-zw";
    EmailCreateRequestLanguageEnum["Ne"] = "ne";
    EmailCreateRequestLanguageEnum["NeIn"] = "ne-in";
    EmailCreateRequestLanguageEnum["NeNp"] = "ne-np";
    EmailCreateRequestLanguageEnum["Nl"] = "nl";
    EmailCreateRequestLanguageEnum["NlAw"] = "nl-aw";
    EmailCreateRequestLanguageEnum["NlBe"] = "nl-be";
    EmailCreateRequestLanguageEnum["NlBq"] = "nl-bq";
    EmailCreateRequestLanguageEnum["NlCh"] = "nl-ch";
    EmailCreateRequestLanguageEnum["NlCw"] = "nl-cw";
    EmailCreateRequestLanguageEnum["NlLu"] = "nl-lu";
    EmailCreateRequestLanguageEnum["NlNl"] = "nl-nl";
    EmailCreateRequestLanguageEnum["NlSr"] = "nl-sr";
    EmailCreateRequestLanguageEnum["NlSx"] = "nl-sx";
    EmailCreateRequestLanguageEnum["Nmg"] = "nmg";
    EmailCreateRequestLanguageEnum["NmgCm"] = "nmg-cm";
    EmailCreateRequestLanguageEnum["Nn"] = "nn";
    EmailCreateRequestLanguageEnum["Nnh"] = "nnh";
    EmailCreateRequestLanguageEnum["NnhCm"] = "nnh-cm";
    EmailCreateRequestLanguageEnum["NnNo"] = "nn-no";
    EmailCreateRequestLanguageEnum["No"] = "no";
    EmailCreateRequestLanguageEnum["NoNo"] = "no-no";
    EmailCreateRequestLanguageEnum["Nus"] = "nus";
    EmailCreateRequestLanguageEnum["NusSs"] = "nus-ss";
    EmailCreateRequestLanguageEnum["Nyn"] = "nyn";
    EmailCreateRequestLanguageEnum["NynUg"] = "nyn-ug";
    EmailCreateRequestLanguageEnum["Oc"] = "oc";
    EmailCreateRequestLanguageEnum["OcEs"] = "oc-es";
    EmailCreateRequestLanguageEnum["OcFr"] = "oc-fr";
    EmailCreateRequestLanguageEnum["Om"] = "om";
    EmailCreateRequestLanguageEnum["OmEt"] = "om-et";
    EmailCreateRequestLanguageEnum["OmKe"] = "om-ke";
    EmailCreateRequestLanguageEnum["Or"] = "or";
    EmailCreateRequestLanguageEnum["OrIn"] = "or-in";
    EmailCreateRequestLanguageEnum["Os"] = "os";
    EmailCreateRequestLanguageEnum["OsGe"] = "os-ge";
    EmailCreateRequestLanguageEnum["OsRu"] = "os-ru";
    EmailCreateRequestLanguageEnum["Pa"] = "pa";
    EmailCreateRequestLanguageEnum["PaIn"] = "pa-in";
    EmailCreateRequestLanguageEnum["PaPk"] = "pa-pk";
    EmailCreateRequestLanguageEnum["Pcm"] = "pcm";
    EmailCreateRequestLanguageEnum["PcmNg"] = "pcm-ng";
    EmailCreateRequestLanguageEnum["Pis"] = "pis";
    EmailCreateRequestLanguageEnum["PisSb"] = "pis-sb";
    EmailCreateRequestLanguageEnum["Pl"] = "pl";
    EmailCreateRequestLanguageEnum["PlPl"] = "pl-pl";
    EmailCreateRequestLanguageEnum["Prg"] = "prg";
    EmailCreateRequestLanguageEnum["Prg001"] = "prg-001";
    EmailCreateRequestLanguageEnum["Ps"] = "ps";
    EmailCreateRequestLanguageEnum["PsAf"] = "ps-af";
    EmailCreateRequestLanguageEnum["PsPk"] = "ps-pk";
    EmailCreateRequestLanguageEnum["Pt"] = "pt";
    EmailCreateRequestLanguageEnum["PtAo"] = "pt-ao";
    EmailCreateRequestLanguageEnum["PtBr"] = "pt-br";
    EmailCreateRequestLanguageEnum["PtCh"] = "pt-ch";
    EmailCreateRequestLanguageEnum["PtCv"] = "pt-cv";
    EmailCreateRequestLanguageEnum["PtGq"] = "pt-gq";
    EmailCreateRequestLanguageEnum["PtGw"] = "pt-gw";
    EmailCreateRequestLanguageEnum["PtLu"] = "pt-lu";
    EmailCreateRequestLanguageEnum["PtMo"] = "pt-mo";
    EmailCreateRequestLanguageEnum["PtMz"] = "pt-mz";
    EmailCreateRequestLanguageEnum["PtPt"] = "pt-pt";
    EmailCreateRequestLanguageEnum["PtSt"] = "pt-st";
    EmailCreateRequestLanguageEnum["PtTl"] = "pt-tl";
    EmailCreateRequestLanguageEnum["Qu"] = "qu";
    EmailCreateRequestLanguageEnum["QuBo"] = "qu-bo";
    EmailCreateRequestLanguageEnum["QuEc"] = "qu-ec";
    EmailCreateRequestLanguageEnum["QuPe"] = "qu-pe";
    EmailCreateRequestLanguageEnum["Raj"] = "raj";
    EmailCreateRequestLanguageEnum["RajIn"] = "raj-in";
    EmailCreateRequestLanguageEnum["Rm"] = "rm";
    EmailCreateRequestLanguageEnum["RmCh"] = "rm-ch";
    EmailCreateRequestLanguageEnum["Rn"] = "rn";
    EmailCreateRequestLanguageEnum["RnBi"] = "rn-bi";
    EmailCreateRequestLanguageEnum["Ro"] = "ro";
    EmailCreateRequestLanguageEnum["Rof"] = "rof";
    EmailCreateRequestLanguageEnum["RofTz"] = "rof-tz";
    EmailCreateRequestLanguageEnum["RoMd"] = "ro-md";
    EmailCreateRequestLanguageEnum["RoRo"] = "ro-ro";
    EmailCreateRequestLanguageEnum["Ru"] = "ru";
    EmailCreateRequestLanguageEnum["RuBy"] = "ru-by";
    EmailCreateRequestLanguageEnum["RuKg"] = "ru-kg";
    EmailCreateRequestLanguageEnum["RuKz"] = "ru-kz";
    EmailCreateRequestLanguageEnum["RuMd"] = "ru-md";
    EmailCreateRequestLanguageEnum["RuRu"] = "ru-ru";
    EmailCreateRequestLanguageEnum["RuUa"] = "ru-ua";
    EmailCreateRequestLanguageEnum["Rw"] = "rw";
    EmailCreateRequestLanguageEnum["Rwk"] = "rwk";
    EmailCreateRequestLanguageEnum["RwkTz"] = "rwk-tz";
    EmailCreateRequestLanguageEnum["RwRw"] = "rw-rw";
    EmailCreateRequestLanguageEnum["Sa"] = "sa";
    EmailCreateRequestLanguageEnum["Sah"] = "sah";
    EmailCreateRequestLanguageEnum["SahRu"] = "sah-ru";
    EmailCreateRequestLanguageEnum["Saq"] = "saq";
    EmailCreateRequestLanguageEnum["SaqKe"] = "saq-ke";
    EmailCreateRequestLanguageEnum["Sat"] = "sat";
    EmailCreateRequestLanguageEnum["SatIn"] = "sat-in";
    EmailCreateRequestLanguageEnum["SaIn"] = "sa-in";
    EmailCreateRequestLanguageEnum["Sbp"] = "sbp";
    EmailCreateRequestLanguageEnum["SbpTz"] = "sbp-tz";
    EmailCreateRequestLanguageEnum["Sc"] = "sc";
    EmailCreateRequestLanguageEnum["ScIt"] = "sc-it";
    EmailCreateRequestLanguageEnum["Sd"] = "sd";
    EmailCreateRequestLanguageEnum["SdIn"] = "sd-in";
    EmailCreateRequestLanguageEnum["SdPk"] = "sd-pk";
    EmailCreateRequestLanguageEnum["Se"] = "se";
    EmailCreateRequestLanguageEnum["Seh"] = "seh";
    EmailCreateRequestLanguageEnum["SehMz"] = "seh-mz";
    EmailCreateRequestLanguageEnum["Ses"] = "ses";
    EmailCreateRequestLanguageEnum["SesMl"] = "ses-ml";
    EmailCreateRequestLanguageEnum["SeFi"] = "se-fi";
    EmailCreateRequestLanguageEnum["SeNo"] = "se-no";
    EmailCreateRequestLanguageEnum["SeSe"] = "se-se";
    EmailCreateRequestLanguageEnum["Sg"] = "sg";
    EmailCreateRequestLanguageEnum["SgCf"] = "sg-cf";
    EmailCreateRequestLanguageEnum["Shi"] = "shi";
    EmailCreateRequestLanguageEnum["ShiMa"] = "shi-ma";
    EmailCreateRequestLanguageEnum["Si"] = "si";
    EmailCreateRequestLanguageEnum["SiLk"] = "si-lk";
    EmailCreateRequestLanguageEnum["Sk"] = "sk";
    EmailCreateRequestLanguageEnum["SkSk"] = "sk-sk";
    EmailCreateRequestLanguageEnum["Sl"] = "sl";
    EmailCreateRequestLanguageEnum["SlSi"] = "sl-si";
    EmailCreateRequestLanguageEnum["Smn"] = "smn";
    EmailCreateRequestLanguageEnum["SmnFi"] = "smn-fi";
    EmailCreateRequestLanguageEnum["Sms"] = "sms";
    EmailCreateRequestLanguageEnum["SmsFi"] = "sms-fi";
    EmailCreateRequestLanguageEnum["Sn"] = "sn";
    EmailCreateRequestLanguageEnum["SnZw"] = "sn-zw";
    EmailCreateRequestLanguageEnum["So"] = "so";
    EmailCreateRequestLanguageEnum["SoDj"] = "so-dj";
    EmailCreateRequestLanguageEnum["SoEt"] = "so-et";
    EmailCreateRequestLanguageEnum["SoKe"] = "so-ke";
    EmailCreateRequestLanguageEnum["SoSo"] = "so-so";
    EmailCreateRequestLanguageEnum["Sq"] = "sq";
    EmailCreateRequestLanguageEnum["SqAl"] = "sq-al";
    EmailCreateRequestLanguageEnum["SqMk"] = "sq-mk";
    EmailCreateRequestLanguageEnum["SqXk"] = "sq-xk";
    EmailCreateRequestLanguageEnum["Sr"] = "sr";
    EmailCreateRequestLanguageEnum["SrBa"] = "sr-ba";
    EmailCreateRequestLanguageEnum["SrCs"] = "sr-cs";
    EmailCreateRequestLanguageEnum["SrMe"] = "sr-me";
    EmailCreateRequestLanguageEnum["SrRs"] = "sr-rs";
    EmailCreateRequestLanguageEnum["SrXk"] = "sr-xk";
    EmailCreateRequestLanguageEnum["Su"] = "su";
    EmailCreateRequestLanguageEnum["SuId"] = "su-id";
    EmailCreateRequestLanguageEnum["Sv"] = "sv";
    EmailCreateRequestLanguageEnum["SvAx"] = "sv-ax";
    EmailCreateRequestLanguageEnum["SvFi"] = "sv-fi";
    EmailCreateRequestLanguageEnum["SvSe"] = "sv-se";
    EmailCreateRequestLanguageEnum["Sw"] = "sw";
    EmailCreateRequestLanguageEnum["SwCd"] = "sw-cd";
    EmailCreateRequestLanguageEnum["SwKe"] = "sw-ke";
    EmailCreateRequestLanguageEnum["SwTz"] = "sw-tz";
    EmailCreateRequestLanguageEnum["SwUg"] = "sw-ug";
    EmailCreateRequestLanguageEnum["Sy"] = "sy";
    EmailCreateRequestLanguageEnum["Ta"] = "ta";
    EmailCreateRequestLanguageEnum["TaIn"] = "ta-in";
    EmailCreateRequestLanguageEnum["TaLk"] = "ta-lk";
    EmailCreateRequestLanguageEnum["TaMy"] = "ta-my";
    EmailCreateRequestLanguageEnum["TaSg"] = "ta-sg";
    EmailCreateRequestLanguageEnum["Te"] = "te";
    EmailCreateRequestLanguageEnum["Teo"] = "teo";
    EmailCreateRequestLanguageEnum["TeoKe"] = "teo-ke";
    EmailCreateRequestLanguageEnum["TeoUg"] = "teo-ug";
    EmailCreateRequestLanguageEnum["TeIn"] = "te-in";
    EmailCreateRequestLanguageEnum["Tg"] = "tg";
    EmailCreateRequestLanguageEnum["TgTj"] = "tg-tj";
    EmailCreateRequestLanguageEnum["Th"] = "th";
    EmailCreateRequestLanguageEnum["ThTh"] = "th-th";
    EmailCreateRequestLanguageEnum["Ti"] = "ti";
    EmailCreateRequestLanguageEnum["TiEr"] = "ti-er";
    EmailCreateRequestLanguageEnum["TiEt"] = "ti-et";
    EmailCreateRequestLanguageEnum["Tk"] = "tk";
    EmailCreateRequestLanguageEnum["TkTm"] = "tk-tm";
    EmailCreateRequestLanguageEnum["Tl"] = "tl";
    EmailCreateRequestLanguageEnum["To"] = "to";
    EmailCreateRequestLanguageEnum["Tok"] = "tok";
    EmailCreateRequestLanguageEnum["Tok001"] = "tok-001";
    EmailCreateRequestLanguageEnum["ToTo"] = "to-to";
    EmailCreateRequestLanguageEnum["Tr"] = "tr";
    EmailCreateRequestLanguageEnum["TrCy"] = "tr-cy";
    EmailCreateRequestLanguageEnum["TrTr"] = "tr-tr";
    EmailCreateRequestLanguageEnum["Tt"] = "tt";
    EmailCreateRequestLanguageEnum["TtRu"] = "tt-ru";
    EmailCreateRequestLanguageEnum["Twq"] = "twq";
    EmailCreateRequestLanguageEnum["TwqNe"] = "twq-ne";
    EmailCreateRequestLanguageEnum["Tzm"] = "tzm";
    EmailCreateRequestLanguageEnum["TzmMa"] = "tzm-ma";
    EmailCreateRequestLanguageEnum["Ug"] = "ug";
    EmailCreateRequestLanguageEnum["UgCn"] = "ug-cn";
    EmailCreateRequestLanguageEnum["Uk"] = "uk";
    EmailCreateRequestLanguageEnum["UkUa"] = "uk-ua";
    EmailCreateRequestLanguageEnum["Ur"] = "ur";
    EmailCreateRequestLanguageEnum["UrIn"] = "ur-in";
    EmailCreateRequestLanguageEnum["UrPk"] = "ur-pk";
    EmailCreateRequestLanguageEnum["Uz"] = "uz";
    EmailCreateRequestLanguageEnum["UzAf"] = "uz-af";
    EmailCreateRequestLanguageEnum["UzUz"] = "uz-uz";
    EmailCreateRequestLanguageEnum["Vai"] = "vai";
    EmailCreateRequestLanguageEnum["VaiLr"] = "vai-lr";
    EmailCreateRequestLanguageEnum["Vi"] = "vi";
    EmailCreateRequestLanguageEnum["ViVn"] = "vi-vn";
    EmailCreateRequestLanguageEnum["Vo"] = "vo";
    EmailCreateRequestLanguageEnum["Vo001"] = "vo-001";
    EmailCreateRequestLanguageEnum["Vun"] = "vun";
    EmailCreateRequestLanguageEnum["VunTz"] = "vun-tz";
    EmailCreateRequestLanguageEnum["Wae"] = "wae";
    EmailCreateRequestLanguageEnum["WaeCh"] = "wae-ch";
    EmailCreateRequestLanguageEnum["Wo"] = "wo";
    EmailCreateRequestLanguageEnum["WoSn"] = "wo-sn";
    EmailCreateRequestLanguageEnum["Xh"] = "xh";
    EmailCreateRequestLanguageEnum["XhZa"] = "xh-za";
    EmailCreateRequestLanguageEnum["Xog"] = "xog";
    EmailCreateRequestLanguageEnum["XogUg"] = "xog-ug";
    EmailCreateRequestLanguageEnum["Yav"] = "yav";
    EmailCreateRequestLanguageEnum["YavCm"] = "yav-cm";
    EmailCreateRequestLanguageEnum["Yo"] = "yo";
    EmailCreateRequestLanguageEnum["YoBj"] = "yo-bj";
    EmailCreateRequestLanguageEnum["YoNg"] = "yo-ng";
    EmailCreateRequestLanguageEnum["Yrl"] = "yrl";
    EmailCreateRequestLanguageEnum["YrlBr"] = "yrl-br";
    EmailCreateRequestLanguageEnum["YrlCo"] = "yrl-co";
    EmailCreateRequestLanguageEnum["YrlVe"] = "yrl-ve";
    EmailCreateRequestLanguageEnum["Yue"] = "yue";
    EmailCreateRequestLanguageEnum["YueCn"] = "yue-cn";
    EmailCreateRequestLanguageEnum["YueHk"] = "yue-hk";
    EmailCreateRequestLanguageEnum["Zgh"] = "zgh";
    EmailCreateRequestLanguageEnum["ZghMa"] = "zgh-ma";
    EmailCreateRequestLanguageEnum["Zh"] = "zh";
    EmailCreateRequestLanguageEnum["ZhCn"] = "zh-cn";
    EmailCreateRequestLanguageEnum["ZhHans"] = "zh-hans";
    EmailCreateRequestLanguageEnum["ZhHant"] = "zh-hant";
    EmailCreateRequestLanguageEnum["ZhHk"] = "zh-hk";
    EmailCreateRequestLanguageEnum["ZhMo"] = "zh-mo";
    EmailCreateRequestLanguageEnum["ZhSg"] = "zh-sg";
    EmailCreateRequestLanguageEnum["ZhTw"] = "zh-tw";
    EmailCreateRequestLanguageEnum["Zu"] = "zu";
    EmailCreateRequestLanguageEnum["ZuZa"] = "zu-za";
})(EmailCreateRequestLanguageEnum = exports.EmailCreateRequestLanguageEnum || (exports.EmailCreateRequestLanguageEnum = {}));
var EmailCreateRequestStateEnum;
(function (EmailCreateRequestStateEnum) {
    EmailCreateRequestStateEnum["Automated"] = "AUTOMATED";
    EmailCreateRequestStateEnum["AutomatedDraft"] = "AUTOMATED_DRAFT";
    EmailCreateRequestStateEnum["AutomatedSending"] = "AUTOMATED_SENDING";
    EmailCreateRequestStateEnum["AutomatedForForm"] = "AUTOMATED_FOR_FORM";
    EmailCreateRequestStateEnum["AutomatedForFormBuffer"] = "AUTOMATED_FOR_FORM_BUFFER";
    EmailCreateRequestStateEnum["AutomatedForFormDraft"] = "AUTOMATED_FOR_FORM_DRAFT";
    EmailCreateRequestStateEnum["AutomatedForFormLegacy"] = "AUTOMATED_FOR_FORM_LEGACY";
    EmailCreateRequestStateEnum["BlogEmailDraft"] = "BLOG_EMAIL_DRAFT";
    EmailCreateRequestStateEnum["BlogEmailPublished"] = "BLOG_EMAIL_PUBLISHED";
    EmailCreateRequestStateEnum["Draft"] = "DRAFT";
    EmailCreateRequestStateEnum["DraftAb"] = "DRAFT_AB";
    EmailCreateRequestStateEnum["DraftAbVariant"] = "DRAFT_AB_VARIANT";
    EmailCreateRequestStateEnum["Error"] = "ERROR";
    EmailCreateRequestStateEnum["LoserAbVariant"] = "LOSER_AB_VARIANT";
    EmailCreateRequestStateEnum["PageStub"] = "PAGE_STUB";
    EmailCreateRequestStateEnum["PreProcessing"] = "PRE_PROCESSING";
    EmailCreateRequestStateEnum["Processing"] = "PROCESSING";
    EmailCreateRequestStateEnum["Published"] = "PUBLISHED";
    EmailCreateRequestStateEnum["PublishedAb"] = "PUBLISHED_AB";
    EmailCreateRequestStateEnum["PublishedAbVariant"] = "PUBLISHED_AB_VARIANT";
    EmailCreateRequestStateEnum["PublishedOrScheduled"] = "PUBLISHED_OR_SCHEDULED";
    EmailCreateRequestStateEnum["RssToEmailDraft"] = "RSS_TO_EMAIL_DRAFT";
    EmailCreateRequestStateEnum["RssToEmailPublished"] = "RSS_TO_EMAIL_PUBLISHED";
    EmailCreateRequestStateEnum["Scheduled"] = "SCHEDULED";
    EmailCreateRequestStateEnum["ScheduledAb"] = "SCHEDULED_AB";
    EmailCreateRequestStateEnum["ScheduledOrPublished"] = "SCHEDULED_OR_PUBLISHED";
    EmailCreateRequestStateEnum["AutomatedAb"] = "AUTOMATED_AB";
    EmailCreateRequestStateEnum["AutomatedAbVariant"] = "AUTOMATED_AB_VARIANT";
    EmailCreateRequestStateEnum["AutomatedDraftAb"] = "AUTOMATED_DRAFT_AB";
    EmailCreateRequestStateEnum["AutomatedDraftAbvariant"] = "AUTOMATED_DRAFT_ABVARIANT";
    EmailCreateRequestStateEnum["AutomatedLoserAbvariant"] = "AUTOMATED_LOSER_ABVARIANT";
})(EmailCreateRequestStateEnum = exports.EmailCreateRequestStateEnum || (exports.EmailCreateRequestStateEnum = {}));
var EmailCreateRequestSubcategoryEnum;
(function (EmailCreateRequestSubcategoryEnum) {
    EmailCreateRequestSubcategoryEnum["AbMaster"] = "ab_master";
    EmailCreateRequestSubcategoryEnum["AbVariant"] = "ab_variant";
    EmailCreateRequestSubcategoryEnum["AbLoserVariant"] = "ab_loser_variant";
    EmailCreateRequestSubcategoryEnum["PageStub"] = "page_stub";
    EmailCreateRequestSubcategoryEnum["LandingPage"] = "landing_page";
    EmailCreateRequestSubcategoryEnum["SitePage"] = "site_page";
    EmailCreateRequestSubcategoryEnum["LegacyPage"] = "legacy_page";
    EmailCreateRequestSubcategoryEnum["AbMasterSitePage"] = "ab_master_site_page";
    EmailCreateRequestSubcategoryEnum["AbVariantSitePage"] = "ab_variant_site_page";
    EmailCreateRequestSubcategoryEnum["AbLoserVariantSitePage"] = "ab_loser_variant_site_page";
    EmailCreateRequestSubcategoryEnum["PerformableLandingPage"] = "performable_landing_page";
    EmailCreateRequestSubcategoryEnum["PerformableLandingPageCutover"] = "performable_landing_page_cutover";
    EmailCreateRequestSubcategoryEnum["StagedPage"] = "staged_page";
    EmailCreateRequestSubcategoryEnum["Automated"] = "automated";
    EmailCreateRequestSubcategoryEnum["AutomatedForDeal"] = "automated_for_deal";
    EmailCreateRequestSubcategoryEnum["AutomatedForForm"] = "automated_for_form";
    EmailCreateRequestSubcategoryEnum["AutomatedForFormLegacy"] = "automated_for_form_legacy";
    EmailCreateRequestSubcategoryEnum["AutomatedForFormBuffer"] = "automated_for_form_buffer";
    EmailCreateRequestSubcategoryEnum["AutomatedForFormDraft"] = "automated_for_form_draft";
    EmailCreateRequestSubcategoryEnum["RssToEmail"] = "rss_to_email";
    EmailCreateRequestSubcategoryEnum["RssToEmailChild"] = "rss_to_email_child";
    EmailCreateRequestSubcategoryEnum["BlogEmail"] = "blog_email";
    EmailCreateRequestSubcategoryEnum["BlogEmailChild"] = "blog_email_child";
    EmailCreateRequestSubcategoryEnum["OptinEmail"] = "optin_email";
    EmailCreateRequestSubcategoryEnum["OptinFollowupEmail"] = "optin_followup_email";
    EmailCreateRequestSubcategoryEnum["Batch"] = "batch";
    EmailCreateRequestSubcategoryEnum["ResubscribeEmail"] = "resubscribe_email";
    EmailCreateRequestSubcategoryEnum["UnsubscribeConfirmationEmail"] = "unsubscribe_confirmation_email";
    EmailCreateRequestSubcategoryEnum["ResubscribeConfirmationEmail"] = "resubscribe_confirmation_email";
    EmailCreateRequestSubcategoryEnum["SingleSendApi"] = "single_send_api";
    EmailCreateRequestSubcategoryEnum["MarketingSingleSendApi"] = "marketing_single_send_api";
    EmailCreateRequestSubcategoryEnum["SmtpToken"] = "smtp_token";
    EmailCreateRequestSubcategoryEnum["Localtime"] = "localtime";
    EmailCreateRequestSubcategoryEnum["AutomatedForTicket"] = "automated_for_ticket";
    EmailCreateRequestSubcategoryEnum["AutomatedForLeadflow"] = "automated_for_leadflow";
    EmailCreateRequestSubcategoryEnum["AutomatedForFeedbackCes"] = "automated_for_feedback_ces";
    EmailCreateRequestSubcategoryEnum["AutomatedForFeedbackNps"] = "automated_for_feedback_nps";
    EmailCreateRequestSubcategoryEnum["AutomatedForFeedbackCustom"] = "automated_for_feedback_custom";
    EmailCreateRequestSubcategoryEnum["MembershipRegistration"] = "membership_registration";
    EmailCreateRequestSubcategoryEnum["MembershipPasswordSaved"] = "membership_password_saved";
    EmailCreateRequestSubcategoryEnum["MembershipPasswordReset"] = "membership_password_reset";
    EmailCreateRequestSubcategoryEnum["MembershipOtpLogin"] = "membership_otp_login";
    EmailCreateRequestSubcategoryEnum["MembershipPasswordlessAuth"] = "membership_passwordless_auth";
    EmailCreateRequestSubcategoryEnum["MembershipEmailVerification"] = "membership_email_verification";
    EmailCreateRequestSubcategoryEnum["MembershipRegistrationFollowUp"] = "membership_registration_follow_up";
    EmailCreateRequestSubcategoryEnum["MembershipVerification"] = "membership_verification";
    EmailCreateRequestSubcategoryEnum["MembershipFollowUp"] = "membership_follow_up";
    EmailCreateRequestSubcategoryEnum["TicketClosedKickbackEmail"] = "ticket_closed_kickback_email";
    EmailCreateRequestSubcategoryEnum["TicketOpenedKickbackEmail"] = "ticket_opened_kickback_email";
    EmailCreateRequestSubcategoryEnum["AutomatedForCustomSurvey"] = "automated_for_custom_survey";
    EmailCreateRequestSubcategoryEnum["DiscardableStub"] = "discardable_stub";
    EmailCreateRequestSubcategoryEnum["NormalBlogPost"] = "normal_blog_post";
    EmailCreateRequestSubcategoryEnum["LegacyBlogPost"] = "legacy_blog_post";
    EmailCreateRequestSubcategoryEnum["ImportedBlogPost"] = "imported_blog_post";
    EmailCreateRequestSubcategoryEnum["AutomatedAbMaster"] = "automated_ab_master";
    EmailCreateRequestSubcategoryEnum["AutomatedAbVariant"] = "automated_ab_variant";
    EmailCreateRequestSubcategoryEnum["WebInteractive"] = "web_interactive";
    EmailCreateRequestSubcategoryEnum["PortalContent"] = "portal_content";
    EmailCreateRequestSubcategoryEnum["PageInstanceLayout"] = "page_instance_layout";
    EmailCreateRequestSubcategoryEnum["KbArticleInstanceLayout"] = "kb_article_instance_layout";
    EmailCreateRequestSubcategoryEnum["KbListing"] = "kb_listing";
    EmailCreateRequestSubcategoryEnum["KbSearchResults"] = "kb_search_results";
    EmailCreateRequestSubcategoryEnum["KbSupportForm"] = "kb_support_form";
    EmailCreateRequestSubcategoryEnum["Kb404Page"] = "kb_404_page";
    EmailCreateRequestSubcategoryEnum["CaseStudy"] = "case_study";
    EmailCreateRequestSubcategoryEnum["CaseStudyListing"] = "case_study_listing";
    EmailCreateRequestSubcategoryEnum["CaseStudyInstanceLayout"] = "case_study_instance_layout";
    EmailCreateRequestSubcategoryEnum["ScpStaticPage"] = "scp_static_page";
    EmailCreateRequestSubcategoryEnum["ScpInstanceLayoutPage"] = "scp_instance_layout_page";
    EmailCreateRequestSubcategoryEnum["PodcastInstanceLayout"] = "podcast_instance_layout";
    EmailCreateRequestSubcategoryEnum["PodcastListing"] = "podcast_listing";
    EmailCreateRequestSubcategoryEnum["BlogArticleInstanceLayout"] = "blog_article_instance_layout";
    EmailCreateRequestSubcategoryEnum["BlogArticleListing"] = "blog_article_listing";
    EmailCreateRequestSubcategoryEnum["BlogAuthorDetail"] = "blog_author_detail";
    EmailCreateRequestSubcategoryEnum["Unknown"] = "UNKNOWN";
})(EmailCreateRequestSubcategoryEnum = exports.EmailCreateRequestSubcategoryEnum || (exports.EmailCreateRequestSubcategoryEnum = {}));
