// @hubspot/api-client/lib/codegen/marketing/emails/models/EmailUpdateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailUpdateRequestSubcategoryEnum = exports.EmailUpdateRequestStateEnum = exports.EmailUpdateRequestLanguageEnum = exports.EmailUpdateRequest = void 0;
class EmailUpdateRequest {
    static getAttributeTypeMap() {
        return EmailUpdateRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.EmailUpdateRequest = EmailUpdateRequest;
EmailUpdateRequest.discriminator = undefined;
EmailUpdateRequest.mapping = undefined;
EmailUpdateRequest.attributeTypeMap = [
    {
        "name": "rssData",
        "baseName": "rssData",
        "type": "PublicRssEmailDetails",
        "format": ""
    },
    {
        "name": "subject",
        "baseName": "subject",
        "type": "string",
        "format": ""
    },
    {
        "name": "testing",
        "baseName": "testing",
        "type": "PublicEmailTestingDetails",
        "format": ""
    },
    {
        "name": "publishDate",
        "baseName": "publishDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "EmailUpdateRequestLanguageEnum",
        "format": ""
    },
    {
        "name": "businessUnitId",
        "baseName": "businessUnitId",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "content",
        "baseName": "content",
        "type": "PublicEmailContent",
        "format": ""
    },
    {
        "name": "webversion",
        "baseName": "webversion",
        "type": "PublicWebversionDetails",
        "format": ""
    },
    {
        "name": "archived",
        "baseName": "archived",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "subscriptionDetails",
        "baseName": "subscriptionDetails",
        "type": "PublicEmailSubscriptionDetails",
        "format": ""
    },
    {
        "name": "activeDomain",
        "baseName": "activeDomain",
        "type": "string",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "campaign",
        "baseName": "campaign",
        "type": "string",
        "format": ""
    },
    {
        "name": "_from",
        "baseName": "from",
        "type": "PublicEmailFromDetails",
        "format": ""
    },
    {
        "name": "jitterSendTime",
        "baseName": "jitterSendTime",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "state",
        "baseName": "state",
        "type": "EmailUpdateRequestStateEnum",
        "format": ""
    },
    {
        "name": "to",
        "baseName": "to",
        "type": "PublicEmailToDetails",
        "format": ""
    },
    {
        "name": "subcategory",
        "baseName": "subcategory",
        "type": "EmailUpdateRequestSubcategoryEnum",
        "format": ""
    },
    {
        "name": "sendOnPublish",
        "baseName": "sendOnPublish",
        "type": "boolean",
        "format": ""
    }
];
var EmailUpdateRequestLanguageEnum;
(function (EmailUpdateRequestLanguageEnum) {
    EmailUpdateRequestLanguageEnum["Af"] = "af";
    EmailUpdateRequestLanguageEnum["AfNa"] = "af-na";
    EmailUpdateRequestLanguageEnum["AfZa"] = "af-za";
    EmailUpdateRequestLanguageEnum["Agq"] = "agq";
    EmailUpdateRequestLanguageEnum["AgqCm"] = "agq-cm";
    EmailUpdateRequestLanguageEnum["Ak"] = "ak";
    EmailUpdateRequestLanguageEnum["AkGh"] = "ak-gh";
    EmailUpdateRequestLanguageEnum["Am"] = "am";
    EmailUpdateRequestLanguageEnum["AmEt"] = "am-et";
    EmailUpdateRequestLanguageEnum["Ann"] = "ann";
    EmailUpdateRequestLanguageEnum["AnnNg"] = "ann-ng";
    EmailUpdateRequestLanguageEnum["Ar"] = "ar";
    EmailUpdateRequestLanguageEnum["Ar001"] = "ar-001";
    EmailUpdateRequestLanguageEnum["ArAe"] = "ar-ae";
    EmailUpdateRequestLanguageEnum["ArBh"] = "ar-bh";
    EmailUpdateRequestLanguageEnum["ArDj"] = "ar-dj";
    EmailUpdateRequestLanguageEnum["ArDz"] = "ar-dz";
    EmailUpdateRequestLanguageEnum["ArEg"] = "ar-eg";
    EmailUpdateRequestLanguageEnum["ArEh"] = "ar-eh";
    EmailUpdateRequestLanguageEnum["ArEr"] = "ar-er";
    EmailUpdateRequestLanguageEnum["ArIl"] = "ar-il";
    EmailUpdateRequestLanguageEnum["ArIq"] = "ar-iq";
    EmailUpdateRequestLanguageEnum["ArJo"] = "ar-jo";
    EmailUpdateRequestLanguageEnum["ArKm"] = "ar-km";
    EmailUpdateRequestLanguageEnum["ArKw"] = "ar-kw";
    EmailUpdateRequestLanguageEnum["ArLb"] = "ar-lb";
    EmailUpdateRequestLanguageEnum["ArLy"] = "ar-ly";
    EmailUpdateRequestLanguageEnum["ArMa"] = "ar-ma";
    EmailUpdateRequestLanguageEnum["ArMr"] = "ar-mr";
    EmailUpdateRequestLanguageEnum["ArOm"] = "ar-om";
    EmailUpdateRequestLanguageEnum["ArPs"] = "ar-ps";
    EmailUpdateRequestLanguageEnum["ArQa"] = "ar-qa";
    EmailUpdateRequestLanguageEnum["ArSa"] = "ar-sa";
    EmailUpdateRequestLanguageEnum["ArSd"] = "ar-sd";
    EmailUpdateRequestLanguageEnum["ArSo"] = "ar-so";
    EmailUpdateRequestLanguageEnum["ArSs"] = "ar-ss";
    EmailUpdateRequestLanguageEnum["ArSy"] = "ar-sy";
    EmailUpdateRequestLanguageEnum["ArTd"] = "ar-td";
    EmailUpdateRequestLanguageEnum["ArTn"] = "ar-tn";
    EmailUpdateRequestLanguageEnum["ArYe"] = "ar-ye";
    EmailUpdateRequestLanguageEnum["As"] = "as";
    EmailUpdateRequestLanguageEnum["Asa"] = "asa";
    EmailUpdateRequestLanguageEnum["AsaTz"] = "asa-tz";
    EmailUpdateRequestLanguageEnum["Ast"] = "ast";
    EmailUpdateRequestLanguageEnum["AstEs"] = "ast-es";
    EmailUpdateRequestLanguageEnum["AsIn"] = "as-in";
    EmailUpdateRequestLanguageEnum["Az"] = "az";
    EmailUpdateRequestLanguageEnum["AzAz"] = "az-az";
    EmailUpdateRequestLanguageEnum["Bas"] = "bas";
    EmailUpdateRequestLanguageEnum["BasCm"] = "bas-cm";
    EmailUpdateRequestLanguageEnum["Be"] = "be";
    EmailUpdateRequestLanguageEnum["Bem"] = "bem";
    EmailUpdateRequestLanguageEnum["BemZm"] = "bem-zm";
    EmailUpdateRequestLanguageEnum["Bez"] = "bez";
    EmailUpdateRequestLanguageEnum["BezTz"] = "bez-tz";
    EmailUpdateRequestLanguageEnum["BeBy"] = "be-by";
    EmailUpdateRequestLanguageEnum["Bg"] = "bg";
    EmailUpdateRequestLanguageEnum["Bgc"] = "bgc";
    EmailUpdateRequestLanguageEnum["BgcIn"] = "bgc-in";
    EmailUpdateRequestLanguageEnum["BgBg"] = "bg-bg";
    EmailUpdateRequestLanguageEnum["Bho"] = "bho";
    EmailUpdateRequestLanguageEnum["BhoIn"] = "bho-in";
    EmailUpdateRequestLanguageEnum["Bm"] = "bm";
    EmailUpdateRequestLanguageEnum["BmMl"] = "bm-ml";
    EmailUpdateRequestLanguageEnum["Bn"] = "bn";
    EmailUpdateRequestLanguageEnum["BnBd"] = "bn-bd";
    EmailUpdateRequestLanguageEnum["BnIn"] = "bn-in";
    EmailUpdateRequestLanguageEnum["Bo"] = "bo";
    EmailUpdateRequestLanguageEnum["BoCn"] = "bo-cn";
    EmailUpdateRequestLanguageEnum["BoIn"] = "bo-in";
    EmailUpdateRequestLanguageEnum["Br"] = "br";
    EmailUpdateRequestLanguageEnum["Brx"] = "brx";
    EmailUpdateRequestLanguageEnum["BrxIn"] = "brx-in";
    EmailUpdateRequestLanguageEnum["BrFr"] = "br-fr";
    EmailUpdateRequestLanguageEnum["Bs"] = "bs";
    EmailUpdateRequestLanguageEnum["BsBa"] = "bs-ba";
    EmailUpdateRequestLanguageEnum["Ca"] = "ca";
    EmailUpdateRequestLanguageEnum["CaAd"] = "ca-ad";
    EmailUpdateRequestLanguageEnum["CaEs"] = "ca-es";
    EmailUpdateRequestLanguageEnum["CaFr"] = "ca-fr";
    EmailUpdateRequestLanguageEnum["CaIt"] = "ca-it";
    EmailUpdateRequestLanguageEnum["Ccp"] = "ccp";
    EmailUpdateRequestLanguageEnum["CcpBd"] = "ccp-bd";
    EmailUpdateRequestLanguageEnum["CcpIn"] = "ccp-in";
    EmailUpdateRequestLanguageEnum["Ce"] = "ce";
    EmailUpdateRequestLanguageEnum["Ceb"] = "ceb";
    EmailUpdateRequestLanguageEnum["CebPh"] = "ceb-ph";
    EmailUpdateRequestLanguageEnum["CeRu"] = "ce-ru";
    EmailUpdateRequestLanguageEnum["Cgg"] = "cgg";
    EmailUpdateRequestLanguageEnum["CggUg"] = "cgg-ug";
    EmailUpdateRequestLanguageEnum["Chr"] = "chr";
    EmailUpdateRequestLanguageEnum["ChrUs"] = "chr-us";
    EmailUpdateRequestLanguageEnum["Ckb"] = "ckb";
    EmailUpdateRequestLanguageEnum["CkbIq"] = "ckb-iq";
    EmailUpdateRequestLanguageEnum["CkbIr"] = "ckb-ir";
    EmailUpdateRequestLanguageEnum["Cs"] = "cs";
    EmailUpdateRequestLanguageEnum["CsCz"] = "cs-cz";
    EmailUpdateRequestLanguageEnum["Cu"] = "cu";
    EmailUpdateRequestLanguageEnum["CuRu"] = "cu-ru";
    EmailUpdateRequestLanguageEnum["Cv"] = "cv";
    EmailUpdateRequestLanguageEnum["CvRu"] = "cv-ru";
    EmailUpdateRequestLanguageEnum["Cy"] = "cy";
    EmailUpdateRequestLanguageEnum["CyGb"] = "cy-gb";
    EmailUpdateRequestLanguageEnum["Da"] = "da";
    EmailUpdateRequestLanguageEnum["Dav"] = "dav";
    EmailUpdateRequestLanguageEnum["DavKe"] = "dav-ke";
    EmailUpdateRequestLanguageEnum["DaDk"] = "da-dk";
    EmailUpdateRequestLanguageEnum["DaGl"] = "da-gl";
    EmailUpdateRequestLanguageEnum["De"] = "de";
    EmailUpdateRequestLanguageEnum["DeAt"] = "de-at";
    EmailUpdateRequestLanguageEnum["DeBe"] = "de-be";
    EmailUpdateRequestLanguageEnum["DeCh"] = "de-ch";
    EmailUpdateRequestLanguageEnum["DeDe"] = "de-de";
    EmailUpdateRequestLanguageEnum["DeGr"] = "de-gr";
    EmailUpdateRequestLanguageEnum["DeIt"] = "de-it";
    EmailUpdateRequestLanguageEnum["DeLi"] = "de-li";
    EmailUpdateRequestLanguageEnum["DeLu"] = "de-lu";
    EmailUpdateRequestLanguageEnum["Dje"] = "dje";
    EmailUpdateRequestLanguageEnum["DjeNe"] = "dje-ne";
    EmailUpdateRequestLanguageEnum["Doi"] = "doi";
    EmailUpdateRequestLanguageEnum["DoiIn"] = "doi-in";
    EmailUpdateRequestLanguageEnum["Dsb"] = "dsb";
    EmailUpdateRequestLanguageEnum["DsbDe"] = "dsb-de";
    EmailUpdateRequestLanguageEnum["Dua"] = "dua";
    EmailUpdateRequestLanguageEnum["DuaCm"] = "dua-cm";
    EmailUpdateRequestLanguageEnum["Dyo"] = "dyo";
    EmailUpdateRequestLanguageEnum["DyoSn"] = "dyo-sn";
    EmailUpdateRequestLanguageEnum["Dz"] = "dz";
    EmailUpdateRequestLanguageEnum["DzBt"] = "dz-bt";
    EmailUpdateRequestLanguageEnum["Ebu"] = "ebu";
    EmailUpdateRequestLanguageEnum["EbuKe"] = "ebu-ke";
    EmailUpdateRequestLanguageEnum["Ee"] = "ee";
    EmailUpdateRequestLanguageEnum["EeGh"] = "ee-gh";
    EmailUpdateRequestLanguageEnum["EeTg"] = "ee-tg";
    EmailUpdateRequestLanguageEnum["El"] = "el";
    EmailUpdateRequestLanguageEnum["ElCy"] = "el-cy";
    EmailUpdateRequestLanguageEnum["ElGr"] = "el-gr";
    EmailUpdateRequestLanguageEnum["En"] = "en";
    EmailUpdateRequestLanguageEnum["En001"] = "en-001";
    EmailUpdateRequestLanguageEnum["En150"] = "en-150";
    EmailUpdateRequestLanguageEnum["EnAe"] = "en-ae";
    EmailUpdateRequestLanguageEnum["EnAg"] = "en-ag";
    EmailUpdateRequestLanguageEnum["EnAi"] = "en-ai";
    EmailUpdateRequestLanguageEnum["EnAs"] = "en-as";
    EmailUpdateRequestLanguageEnum["EnAt"] = "en-at";
    EmailUpdateRequestLanguageEnum["EnAu"] = "en-au";
    EmailUpdateRequestLanguageEnum["EnBb"] = "en-bb";
    EmailUpdateRequestLanguageEnum["EnBe"] = "en-be";
    EmailUpdateRequestLanguageEnum["EnBi"] = "en-bi";
    EmailUpdateRequestLanguageEnum["EnBm"] = "en-bm";
    EmailUpdateRequestLanguageEnum["EnBs"] = "en-bs";
    EmailUpdateRequestLanguageEnum["EnBw"] = "en-bw";
    EmailUpdateRequestLanguageEnum["EnBz"] = "en-bz";
    EmailUpdateRequestLanguageEnum["EnCa"] = "en-ca";
    EmailUpdateRequestLanguageEnum["EnCc"] = "en-cc";
    EmailUpdateRequestLanguageEnum["EnCh"] = "en-ch";
    EmailUpdateRequestLanguageEnum["EnCk"] = "en-ck";
    EmailUpdateRequestLanguageEnum["EnCm"] = "en-cm";
    EmailUpdateRequestLanguageEnum["EnCn"] = "en-cn";
    EmailUpdateRequestLanguageEnum["EnCx"] = "en-cx";
    EmailUpdateRequestLanguageEnum["EnCy"] = "en-cy";
    EmailUpdateRequestLanguageEnum["EnDe"] = "en-de";
    EmailUpdateRequestLanguageEnum["EnDg"] = "en-dg";
    EmailUpdateRequestLanguageEnum["EnDk"] = "en-dk";
    EmailUpdateRequestLanguageEnum["EnDm"] = "en-dm";
    EmailUpdateRequestLanguageEnum["EnEe"] = "en-ee";
    EmailUpdateRequestLanguageEnum["EnEr"] = "en-er";
    EmailUpdateRequestLanguageEnum["EnEs"] = "en-es";
    EmailUpdateRequestLanguageEnum["EnFi"] = "en-fi";
    EmailUpdateRequestLanguageEnum["EnFj"] = "en-fj";
    EmailUpdateRequestLanguageEnum["EnFk"] = "en-fk";
    EmailUpdateRequestLanguageEnum["EnFm"] = "en-fm";
    EmailUpdateRequestLanguageEnum["EnFr"] = "en-fr";
    EmailUpdateRequestLanguageEnum["EnGb"] = "en-gb";
    EmailUpdateRequestLanguageEnum["EnGd"] = "en-gd";
    EmailUpdateRequestLanguageEnum["EnGg"] = "en-gg";
    EmailUpdateRequestLanguageEnum["EnGh"] = "en-gh";
    EmailUpdateRequestLanguageEnum["EnGi"] = "en-gi";
    EmailUpdateRequestLanguageEnum["EnGm"] = "en-gm";
    EmailUpdateRequestLanguageEnum["EnGu"] = "en-gu";
    EmailUpdateRequestLanguageEnum["EnGy"] = "en-gy";
    EmailUpdateRequestLanguageEnum["EnHk"] = "en-hk";
    EmailUpdateRequestLanguageEnum["EnIe"] = "en-ie";
    EmailUpdateRequestLanguageEnum["EnIl"] = "en-il";
    EmailUpdateRequestLanguageEnum["EnIm"] = "en-im";
    EmailUpdateRequestLanguageEnum["EnIn"] = "en-in";
    EmailUpdateRequestLanguageEnum["EnIo"] = "en-io";
    EmailUpdateRequestLanguageEnum["EnJe"] = "en-je";
    EmailUpdateRequestLanguageEnum["EnJm"] = "en-jm";
    EmailUpdateRequestLanguageEnum["EnKe"] = "en-ke";
    EmailUpdateRequestLanguageEnum["EnKi"] = "en-ki";
    EmailUpdateRequestLanguageEnum["EnKn"] = "en-kn";
    EmailUpdateRequestLanguageEnum["EnKy"] = "en-ky";
    EmailUpdateRequestLanguageEnum["EnLc"] = "en-lc";
    EmailUpdateRequestLanguageEnum["EnLr"] = "en-lr";
    EmailUpdateRequestLanguageEnum["EnLs"] = "en-ls";
    EmailUpdateRequestLanguageEnum["EnLu"] = "en-lu";
    EmailUpdateRequestLanguageEnum["EnMg"] = "en-mg";
    EmailUpdateRequestLanguageEnum["EnMh"] = "en-mh";
    EmailUpdateRequestLanguageEnum["EnMo"] = "en-mo";
    EmailUpdateRequestLanguageEnum["EnMp"] = "en-mp";
    EmailUpdateRequestLanguageEnum["EnMs"] = "en-ms";
    EmailUpdateRequestLanguageEnum["EnMt"] = "en-mt";
    EmailUpdateRequestLanguageEnum["EnMu"] = "en-mu";
    EmailUpdateRequestLanguageEnum["EnMv"] = "en-mv";
    EmailUpdateRequestLanguageEnum["EnMw"] = "en-mw";
    EmailUpdateRequestLanguageEnum["EnMx"] = "en-mx";
    EmailUpdateRequestLanguageEnum["EnMy"] = "en-my";
    EmailUpdateRequestLanguageEnum["EnNa"] = "en-na";
    EmailUpdateRequestLanguageEnum["EnNf"] = "en-nf";
    EmailUpdateRequestLanguageEnum["EnNg"] = "en-ng";
    EmailUpdateRequestLanguageEnum["EnNl"] = "en-nl";
    EmailUpdateRequestLanguageEnum["EnNr"] = "en-nr";
    EmailUpdateRequestLanguageEnum["EnNu"] = "en-nu";
    EmailUpdateRequestLanguageEnum["EnNz"] = "en-nz";
    EmailUpdateRequestLanguageEnum["EnPg"] = "en-pg";
    EmailUpdateRequestLanguageEnum["EnPh"] = "en-ph";
    EmailUpdateRequestLanguageEnum["EnPk"] = "en-pk";
    EmailUpdateRequestLanguageEnum["EnPn"] = "en-pn";
    EmailUpdateRequestLanguageEnum["EnPr"] = "en-pr";
    EmailUpdateRequestLanguageEnum["EnPw"] = "en-pw";
    EmailUpdateRequestLanguageEnum["EnRw"] = "en-rw";
    EmailUpdateRequestLanguageEnum["EnSb"] = "en-sb";
    EmailUpdateRequestLanguageEnum["EnSc"] = "en-sc";
    EmailUpdateRequestLanguageEnum["EnSd"] = "en-sd";
    EmailUpdateRequestLanguageEnum["EnSe"] = "en-se";
    EmailUpdateRequestLanguageEnum["EnSg"] = "en-sg";
    EmailUpdateRequestLanguageEnum["EnSh"] = "en-sh";
    EmailUpdateRequestLanguageEnum["EnSi"] = "en-si";
    EmailUpdateRequestLanguageEnum["EnSl"] = "en-sl";
    EmailUpdateRequestLanguageEnum["EnSs"] = "en-ss";
    EmailUpdateRequestLanguageEnum["EnSx"] = "en-sx";
    EmailUpdateRequestLanguageEnum["EnSz"] = "en-sz";
    EmailUpdateRequestLanguageEnum["EnTc"] = "en-tc";
    EmailUpdateRequestLanguageEnum["EnTk"] = "en-tk";
    EmailUpdateRequestLanguageEnum["EnTn"] = "en-tn";
    EmailUpdateRequestLanguageEnum["EnTo"] = "en-to";
    EmailUpdateRequestLanguageEnum["EnTt"] = "en-tt";
    EmailUpdateRequestLanguageEnum["EnTv"] = "en-tv";
    EmailUpdateRequestLanguageEnum["EnTz"] = "en-tz";
    EmailUpdateRequestLanguageEnum["EnUg"] = "en-ug";
    EmailUpdateRequestLanguageEnum["EnUm"] = "en-um";
    EmailUpdateRequestLanguageEnum["EnUs"] = "en-us";
    EmailUpdateRequestLanguageEnum["EnVc"] = "en-vc";
    EmailUpdateRequestLanguageEnum["EnVg"] = "en-vg";
    EmailUpdateRequestLanguageEnum["EnVi"] = "en-vi";
    EmailUpdateRequestLanguageEnum["EnVu"] = "en-vu";
    EmailUpdateRequestLanguageEnum["EnWs"] = "en-ws";
    EmailUpdateRequestLanguageEnum["EnZa"] = "en-za";
    EmailUpdateRequestLanguageEnum["EnZm"] = "en-zm";
    EmailUpdateRequestLanguageEnum["EnZw"] = "en-zw";
    EmailUpdateRequestLanguageEnum["Eo"] = "eo";
    EmailUpdateRequestLanguageEnum["Eo001"] = "eo-001";
    EmailUpdateRequestLanguageEnum["Es"] = "es";
    EmailUpdateRequestLanguageEnum["Es419"] = "es-419";
    EmailUpdateRequestLanguageEnum["EsAr"] = "es-ar";
    EmailUpdateRequestLanguageEnum["EsBo"] = "es-bo";
    EmailUpdateRequestLanguageEnum["EsBr"] = "es-br";
    EmailUpdateRequestLanguageEnum["EsBz"] = "es-bz";
    EmailUpdateRequestLanguageEnum["EsCl"] = "es-cl";
    EmailUpdateRequestLanguageEnum["EsCo"] = "es-co";
    EmailUpdateRequestLanguageEnum["EsCr"] = "es-cr";
    EmailUpdateRequestLanguageEnum["EsCu"] = "es-cu";
    EmailUpdateRequestLanguageEnum["EsDo"] = "es-do";
    EmailUpdateRequestLanguageEnum["EsEa"] = "es-ea";
    EmailUpdateRequestLanguageEnum["EsEc"] = "es-ec";
    EmailUpdateRequestLanguageEnum["EsEs"] = "es-es";
    EmailUpdateRequestLanguageEnum["EsGq"] = "es-gq";
    EmailUpdateRequestLanguageEnum["EsGt"] = "es-gt";
    EmailUpdateRequestLanguageEnum["EsHn"] = "es-hn";
    EmailUpdateRequestLanguageEnum["EsIc"] = "es-ic";
    EmailUpdateRequestLanguageEnum["EsMx"] = "es-mx";
    EmailUpdateRequestLanguageEnum["EsNi"] = "es-ni";
    EmailUpdateRequestLanguageEnum["EsPa"] = "es-pa";
    EmailUpdateRequestLanguageEnum["EsPe"] = "es-pe";
    EmailUpdateRequestLanguageEnum["EsPh"] = "es-ph";
    EmailUpdateRequestLanguageEnum["EsPr"] = "es-pr";
    EmailUpdateRequestLanguageEnum["EsPy"] = "es-py";
    EmailUpdateRequestLanguageEnum["EsSv"] = "es-sv";
    EmailUpdateRequestLanguageEnum["EsUs"] = "es-us";
    EmailUpdateRequestLanguageEnum["EsUy"] = "es-uy";
    EmailUpdateRequestLanguageEnum["EsVe"] = "es-ve";
    EmailUpdateRequestLanguageEnum["Et"] = "et";
    EmailUpdateRequestLanguageEnum["EtEe"] = "et-ee";
    EmailUpdateRequestLanguageEnum["Eu"] = "eu";
    EmailUpdateRequestLanguageEnum["EuEs"] = "eu-es";
    EmailUpdateRequestLanguageEnum["Ewo"] = "ewo";
    EmailUpdateRequestLanguageEnum["EwoCm"] = "ewo-cm";
    EmailUpdateRequestLanguageEnum["Fa"] = "fa";
    EmailUpdateRequestLanguageEnum["FaAf"] = "fa-af";
    EmailUpdateRequestLanguageEnum["FaIr"] = "fa-ir";
    EmailUpdateRequestLanguageEnum["Ff"] = "ff";
    EmailUpdateRequestLanguageEnum["FfBf"] = "ff-bf";
    EmailUpdateRequestLanguageEnum["FfCm"] = "ff-cm";
    EmailUpdateRequestLanguageEnum["FfGh"] = "ff-gh";
    EmailUpdateRequestLanguageEnum["FfGm"] = "ff-gm";
    EmailUpdateRequestLanguageEnum["FfGn"] = "ff-gn";
    EmailUpdateRequestLanguageEnum["FfGw"] = "ff-gw";
    EmailUpdateRequestLanguageEnum["FfLr"] = "ff-lr";
    EmailUpdateRequestLanguageEnum["FfMr"] = "ff-mr";
    EmailUpdateRequestLanguageEnum["FfNe"] = "ff-ne";
    EmailUpdateRequestLanguageEnum["FfNg"] = "ff-ng";
    EmailUpdateRequestLanguageEnum["FfSl"] = "ff-sl";
    EmailUpdateRequestLanguageEnum["FfSn"] = "ff-sn";
    EmailUpdateRequestLanguageEnum["Fi"] = "fi";
    EmailUpdateRequestLanguageEnum["Fil"] = "fil";
    EmailUpdateRequestLanguageEnum["FilPh"] = "fil-ph";
    EmailUpdateRequestLanguageEnum["FiFi"] = "fi-fi";
    EmailUpdateRequestLanguageEnum["Fo"] = "fo";
    EmailUpdateRequestLanguageEnum["FoDk"] = "fo-dk";
    EmailUpdateRequestLanguageEnum["FoFo"] = "fo-fo";
    EmailUpdateRequestLanguageEnum["Fr"] = "fr";
    EmailUpdateRequestLanguageEnum["Frr"] = "frr";
    EmailUpdateRequestLanguageEnum["FrrDe"] = "frr-de";
    EmailUpdateRequestLanguageEnum["FrBe"] = "fr-be";
    EmailUpdateRequestLanguageEnum["FrBf"] = "fr-bf";
    EmailUpdateRequestLanguageEnum["FrBi"] = "fr-bi";
    EmailUpdateRequestLanguageEnum["FrBj"] = "fr-bj";
    EmailUpdateRequestLanguageEnum["FrBl"] = "fr-bl";
    EmailUpdateRequestLanguageEnum["FrCa"] = "fr-ca";
    EmailUpdateRequestLanguageEnum["FrCd"] = "fr-cd";
    EmailUpdateRequestLanguageEnum["FrCf"] = "fr-cf";
    EmailUpdateRequestLanguageEnum["FrCg"] = "fr-cg";
    EmailUpdateRequestLanguageEnum["FrCh"] = "fr-ch";
    EmailUpdateRequestLanguageEnum["FrCi"] = "fr-ci";
    EmailUpdateRequestLanguageEnum["FrCm"] = "fr-cm";
    EmailUpdateRequestLanguageEnum["FrDj"] = "fr-dj";
    EmailUpdateRequestLanguageEnum["FrDz"] = "fr-dz";
    EmailUpdateRequestLanguageEnum["FrFr"] = "fr-fr";
    EmailUpdateRequestLanguageEnum["FrGa"] = "fr-ga";
    EmailUpdateRequestLanguageEnum["FrGf"] = "fr-gf";
    EmailUpdateRequestLanguageEnum["FrGn"] = "fr-gn";
    EmailUpdateRequestLanguageEnum["FrGp"] = "fr-gp";
    EmailUpdateRequestLanguageEnum["FrGq"] = "fr-gq";
    EmailUpdateRequestLanguageEnum["FrHt"] = "fr-ht";
    EmailUpdateRequestLanguageEnum["FrKm"] = "fr-km";
    EmailUpdateRequestLanguageEnum["FrLu"] = "fr-lu";
    EmailUpdateRequestLanguageEnum["FrMa"] = "fr-ma";
    EmailUpdateRequestLanguageEnum["FrMc"] = "fr-mc";
    EmailUpdateRequestLanguageEnum["FrMf"] = "fr-mf";
    EmailUpdateRequestLanguageEnum["FrMg"] = "fr-mg";
    EmailUpdateRequestLanguageEnum["FrMl"] = "fr-ml";
    EmailUpdateRequestLanguageEnum["FrMq"] = "fr-mq";
    EmailUpdateRequestLanguageEnum["FrMr"] = "fr-mr";
    EmailUpdateRequestLanguageEnum["FrMu"] = "fr-mu";
    EmailUpdateRequestLanguageEnum["FrNc"] = "fr-nc";
    EmailUpdateRequestLanguageEnum["FrNe"] = "fr-ne";
    EmailUpdateRequestLanguageEnum["FrPf"] = "fr-pf";
    EmailUpdateRequestLanguageEnum["FrPm"] = "fr-pm";
    EmailUpdateRequestLanguageEnum["FrRe"] = "fr-re";
    EmailUpdateRequestLanguageEnum["FrRw"] = "fr-rw";
    EmailUpdateRequestLanguageEnum["FrSc"] = "fr-sc";
    EmailUpdateRequestLanguageEnum["FrSn"] = "fr-sn";
    EmailUpdateRequestLanguageEnum["FrSy"] = "fr-sy";
    EmailUpdateRequestLanguageEnum["FrTd"] = "fr-td";
    EmailUpdateRequestLanguageEnum["FrTg"] = "fr-tg";
    EmailUpdateRequestLanguageEnum["FrTn"] = "fr-tn";
    EmailUpdateRequestLanguageEnum["FrVu"] = "fr-vu";
    EmailUpdateRequestLanguageEnum["FrWf"] = "fr-wf";
    EmailUpdateRequestLanguageEnum["FrYt"] = "fr-yt";
    EmailUpdateRequestLanguageEnum["Fur"] = "fur";
    EmailUpdateRequestLanguageEnum["FurIt"] = "fur-it";
    EmailUpdateRequestLanguageEnum["Fy"] = "fy";
    EmailUpdateRequestLanguageEnum["FyNl"] = "fy-nl";
    EmailUpdateRequestLanguageEnum["Ga"] = "ga";
    EmailUpdateRequestLanguageEnum["GaGb"] = "ga-gb";
    EmailUpdateRequestLanguageEnum["GaIe"] = "ga-ie";
    EmailUpdateRequestLanguageEnum["Gd"] = "gd";
    EmailUpdateRequestLanguageEnum["GdGb"] = "gd-gb";
    EmailUpdateRequestLanguageEnum["Gl"] = "gl";
    EmailUpdateRequestLanguageEnum["GlEs"] = "gl-es";
    EmailUpdateRequestLanguageEnum["Gsw"] = "gsw";
    EmailUpdateRequestLanguageEnum["GswCh"] = "gsw-ch";
    EmailUpdateRequestLanguageEnum["GswFr"] = "gsw-fr";
    EmailUpdateRequestLanguageEnum["GswLi"] = "gsw-li";
    EmailUpdateRequestLanguageEnum["Gu"] = "gu";
    EmailUpdateRequestLanguageEnum["Guz"] = "guz";
    EmailUpdateRequestLanguageEnum["GuzKe"] = "guz-ke";
    EmailUpdateRequestLanguageEnum["GuIn"] = "gu-in";
    EmailUpdateRequestLanguageEnum["Gv"] = "gv";
    EmailUpdateRequestLanguageEnum["GvIm"] = "gv-im";
    EmailUpdateRequestLanguageEnum["Ha"] = "ha";
    EmailUpdateRequestLanguageEnum["Haw"] = "haw";
    EmailUpdateRequestLanguageEnum["HawUs"] = "haw-us";
    EmailUpdateRequestLanguageEnum["HaGh"] = "ha-gh";
    EmailUpdateRequestLanguageEnum["HaNe"] = "ha-ne";
    EmailUpdateRequestLanguageEnum["HaNg"] = "ha-ng";
    EmailUpdateRequestLanguageEnum["He"] = "he";
    EmailUpdateRequestLanguageEnum["HeIl"] = "he-il";
    EmailUpdateRequestLanguageEnum["Hi"] = "hi";
    EmailUpdateRequestLanguageEnum["HiIn"] = "hi-in";
    EmailUpdateRequestLanguageEnum["Hr"] = "hr";
    EmailUpdateRequestLanguageEnum["HrBa"] = "hr-ba";
    EmailUpdateRequestLanguageEnum["HrHr"] = "hr-hr";
    EmailUpdateRequestLanguageEnum["Hsb"] = "hsb";
    EmailUpdateRequestLanguageEnum["HsbDe"] = "hsb-de";
    EmailUpdateRequestLanguageEnum["Hu"] = "hu";
    EmailUpdateRequestLanguageEnum["HuHu"] = "hu-hu";
    EmailUpdateRequestLanguageEnum["Hy"] = "hy";
    EmailUpdateRequestLanguageEnum["HyAm"] = "hy-am";
    EmailUpdateRequestLanguageEnum["Ia"] = "ia";
    EmailUpdateRequestLanguageEnum["Ia001"] = "ia-001";
    EmailUpdateRequestLanguageEnum["Id"] = "id";
    EmailUpdateRequestLanguageEnum["Ig"] = "ig";
    EmailUpdateRequestLanguageEnum["IgNg"] = "ig-ng";
    EmailUpdateRequestLanguageEnum["Ii"] = "ii";
    EmailUpdateRequestLanguageEnum["IiCn"] = "ii-cn";
    EmailUpdateRequestLanguageEnum["IdId"] = "id-id";
    EmailUpdateRequestLanguageEnum["Is"] = "is";
    EmailUpdateRequestLanguageEnum["IsIs"] = "is-is";
    EmailUpdateRequestLanguageEnum["It"] = "it";
    EmailUpdateRequestLanguageEnum["ItCh"] = "it-ch";
    EmailUpdateRequestLanguageEnum["ItIt"] = "it-it";
    EmailUpdateRequestLanguageEnum["ItSm"] = "it-sm";
    EmailUpdateRequestLanguageEnum["ItVa"] = "it-va";
    EmailUpdateRequestLanguageEnum["Ja"] = "ja";
    EmailUpdateRequestLanguageEnum["JaJp"] = "ja-jp";
    EmailUpdateRequestLanguageEnum["Jgo"] = "jgo";
    EmailUpdateRequestLanguageEnum["JgoCm"] = "jgo-cm";
    EmailUpdateRequestLanguageEnum["Yi"] = "yi";
    EmailUpdateRequestLanguageEnum["Yi001"] = "yi-001";
    EmailUpdateRequestLanguageEnum["Jmc"] = "jmc";
    EmailUpdateRequestLanguageEnum["JmcTz"] = "jmc-tz";
    EmailUpdateRequestLanguageEnum["Jv"] = "jv";
    EmailUpdateRequestLanguageEnum["JvId"] = "jv-id";
    EmailUpdateRequestLanguageEnum["Ka"] = "ka";
    EmailUpdateRequestLanguageEnum["Kab"] = "kab";
    EmailUpdateRequestLanguageEnum["KabDz"] = "kab-dz";
    EmailUpdateRequestLanguageEnum["Kam"] = "kam";
    EmailUpdateRequestLanguageEnum["KamKe"] = "kam-ke";
    EmailUpdateRequestLanguageEnum["KaGe"] = "ka-ge";
    EmailUpdateRequestLanguageEnum["Kde"] = "kde";
    EmailUpdateRequestLanguageEnum["KdeTz"] = "kde-tz";
    EmailUpdateRequestLanguageEnum["Kea"] = "kea";
    EmailUpdateRequestLanguageEnum["KeaCv"] = "kea-cv";
    EmailUpdateRequestLanguageEnum["Kgp"] = "kgp";
    EmailUpdateRequestLanguageEnum["KgpBr"] = "kgp-br";
    EmailUpdateRequestLanguageEnum["Khq"] = "khq";
    EmailUpdateRequestLanguageEnum["KhqMl"] = "khq-ml";
    EmailUpdateRequestLanguageEnum["Ki"] = "ki";
    EmailUpdateRequestLanguageEnum["KiKe"] = "ki-ke";
    EmailUpdateRequestLanguageEnum["Kk"] = "kk";
    EmailUpdateRequestLanguageEnum["Kkj"] = "kkj";
    EmailUpdateRequestLanguageEnum["KkjCm"] = "kkj-cm";
    EmailUpdateRequestLanguageEnum["KkKz"] = "kk-kz";
    EmailUpdateRequestLanguageEnum["Kl"] = "kl";
    EmailUpdateRequestLanguageEnum["Kln"] = "kln";
    EmailUpdateRequestLanguageEnum["KlnKe"] = "kln-ke";
    EmailUpdateRequestLanguageEnum["KlGl"] = "kl-gl";
    EmailUpdateRequestLanguageEnum["Km"] = "km";
    EmailUpdateRequestLanguageEnum["KmKh"] = "km-kh";
    EmailUpdateRequestLanguageEnum["Kn"] = "kn";
    EmailUpdateRequestLanguageEnum["KnIn"] = "kn-in";
    EmailUpdateRequestLanguageEnum["Ko"] = "ko";
    EmailUpdateRequestLanguageEnum["Kok"] = "kok";
    EmailUpdateRequestLanguageEnum["KokIn"] = "kok-in";
    EmailUpdateRequestLanguageEnum["KoKp"] = "ko-kp";
    EmailUpdateRequestLanguageEnum["KoKr"] = "ko-kr";
    EmailUpdateRequestLanguageEnum["Ks"] = "ks";
    EmailUpdateRequestLanguageEnum["Ksb"] = "ksb";
    EmailUpdateRequestLanguageEnum["KsbTz"] = "ksb-tz";
    EmailUpdateRequestLanguageEnum["Ksf"] = "ksf";
    EmailUpdateRequestLanguageEnum["KsfCm"] = "ksf-cm";
    EmailUpdateRequestLanguageEnum["Ksh"] = "ksh";
    EmailUpdateRequestLanguageEnum["KshDe"] = "ksh-de";
    EmailUpdateRequestLanguageEnum["KsIn"] = "ks-in";
    EmailUpdateRequestLanguageEnum["Ku"] = "ku";
    EmailUpdateRequestLanguageEnum["KuTr"] = "ku-tr";
    EmailUpdateRequestLanguageEnum["Kw"] = "kw";
    EmailUpdateRequestLanguageEnum["KwGb"] = "kw-gb";
    EmailUpdateRequestLanguageEnum["Ky"] = "ky";
    EmailUpdateRequestLanguageEnum["KyKg"] = "ky-kg";
    EmailUpdateRequestLanguageEnum["Lag"] = "lag";
    EmailUpdateRequestLanguageEnum["LagTz"] = "lag-tz";
    EmailUpdateRequestLanguageEnum["Lb"] = "lb";
    EmailUpdateRequestLanguageEnum["LbLu"] = "lb-lu";
    EmailUpdateRequestLanguageEnum["Lg"] = "lg";
    EmailUpdateRequestLanguageEnum["LgUg"] = "lg-ug";
    EmailUpdateRequestLanguageEnum["Lkt"] = "lkt";
    EmailUpdateRequestLanguageEnum["LktUs"] = "lkt-us";
    EmailUpdateRequestLanguageEnum["Ln"] = "ln";
    EmailUpdateRequestLanguageEnum["LnAo"] = "ln-ao";
    EmailUpdateRequestLanguageEnum["LnCd"] = "ln-cd";
    EmailUpdateRequestLanguageEnum["LnCf"] = "ln-cf";
    EmailUpdateRequestLanguageEnum["LnCg"] = "ln-cg";
    EmailUpdateRequestLanguageEnum["Lo"] = "lo";
    EmailUpdateRequestLanguageEnum["LoLa"] = "lo-la";
    EmailUpdateRequestLanguageEnum["Lrc"] = "lrc";
    EmailUpdateRequestLanguageEnum["LrcIq"] = "lrc-iq";
    EmailUpdateRequestLanguageEnum["LrcIr"] = "lrc-ir";
    EmailUpdateRequestLanguageEnum["Lt"] = "lt";
    EmailUpdateRequestLanguageEnum["LtLt"] = "lt-lt";
    EmailUpdateRequestLanguageEnum["Lu"] = "lu";
    EmailUpdateRequestLanguageEnum["Luo"] = "luo";
    EmailUpdateRequestLanguageEnum["LuoKe"] = "luo-ke";
    EmailUpdateRequestLanguageEnum["Luy"] = "luy";
    EmailUpdateRequestLanguageEnum["LuyKe"] = "luy-ke";
    EmailUpdateRequestLanguageEnum["LuCd"] = "lu-cd";
    EmailUpdateRequestLanguageEnum["Lv"] = "lv";
    EmailUpdateRequestLanguageEnum["LvLv"] = "lv-lv";
    EmailUpdateRequestLanguageEnum["Mai"] = "mai";
    EmailUpdateRequestLanguageEnum["MaiIn"] = "mai-in";
    EmailUpdateRequestLanguageEnum["Mas"] = "mas";
    EmailUpdateRequestLanguageEnum["MasKe"] = "mas-ke";
    EmailUpdateRequestLanguageEnum["MasTz"] = "mas-tz";
    EmailUpdateRequestLanguageEnum["Mdf"] = "mdf";
    EmailUpdateRequestLanguageEnum["MdfRu"] = "mdf-ru";
    EmailUpdateRequestLanguageEnum["Mer"] = "mer";
    EmailUpdateRequestLanguageEnum["MerKe"] = "mer-ke";
    EmailUpdateRequestLanguageEnum["Mfe"] = "mfe";
    EmailUpdateRequestLanguageEnum["MfeMu"] = "mfe-mu";
    EmailUpdateRequestLanguageEnum["Mg"] = "mg";
    EmailUpdateRequestLanguageEnum["Mgh"] = "mgh";
    EmailUpdateRequestLanguageEnum["MghMz"] = "mgh-mz";
    EmailUpdateRequestLanguageEnum["Mgo"] = "mgo";
    EmailUpdateRequestLanguageEnum["MgoCm"] = "mgo-cm";
    EmailUpdateRequestLanguageEnum["MgMg"] = "mg-mg";
    EmailUpdateRequestLanguageEnum["Mi"] = "mi";
    EmailUpdateRequestLanguageEnum["MiNz"] = "mi-nz";
    EmailUpdateRequestLanguageEnum["Mk"] = "mk";
    EmailUpdateRequestLanguageEnum["MkMk"] = "mk-mk";
    EmailUpdateRequestLanguageEnum["Ml"] = "ml";
    EmailUpdateRequestLanguageEnum["MlIn"] = "ml-in";
    EmailUpdateRequestLanguageEnum["Mn"] = "mn";
    EmailUpdateRequestLanguageEnum["Mni"] = "mni";
    EmailUpdateRequestLanguageEnum["MniIn"] = "mni-in";
    EmailUpdateRequestLanguageEnum["MnMn"] = "mn-mn";
    EmailUpdateRequestLanguageEnum["Mr"] = "mr";
    EmailUpdateRequestLanguageEnum["MrIn"] = "mr-in";
    EmailUpdateRequestLanguageEnum["Ms"] = "ms";
    EmailUpdateRequestLanguageEnum["MsBn"] = "ms-bn";
    EmailUpdateRequestLanguageEnum["MsId"] = "ms-id";
    EmailUpdateRequestLanguageEnum["MsMy"] = "ms-my";
    EmailUpdateRequestLanguageEnum["MsSg"] = "ms-sg";
    EmailUpdateRequestLanguageEnum["Mt"] = "mt";
    EmailUpdateRequestLanguageEnum["MtMt"] = "mt-mt";
    EmailUpdateRequestLanguageEnum["Mua"] = "mua";
    EmailUpdateRequestLanguageEnum["MuaCm"] = "mua-cm";
    EmailUpdateRequestLanguageEnum["My"] = "my";
    EmailUpdateRequestLanguageEnum["MyMm"] = "my-mm";
    EmailUpdateRequestLanguageEnum["Mzn"] = "mzn";
    EmailUpdateRequestLanguageEnum["MznIr"] = "mzn-ir";
    EmailUpdateRequestLanguageEnum["Naq"] = "naq";
    EmailUpdateRequestLanguageEnum["NaqNa"] = "naq-na";
    EmailUpdateRequestLanguageEnum["Nb"] = "nb";
    EmailUpdateRequestLanguageEnum["NbNo"] = "nb-no";
    EmailUpdateRequestLanguageEnum["NbSj"] = "nb-sj";
    EmailUpdateRequestLanguageEnum["Nd"] = "nd";
    EmailUpdateRequestLanguageEnum["Nds"] = "nds";
    EmailUpdateRequestLanguageEnum["NdsDe"] = "nds-de";
    EmailUpdateRequestLanguageEnum["NdsNl"] = "nds-nl";
    EmailUpdateRequestLanguageEnum["NdZw"] = "nd-zw";
    EmailUpdateRequestLanguageEnum["Ne"] = "ne";
    EmailUpdateRequestLanguageEnum["NeIn"] = "ne-in";
    EmailUpdateRequestLanguageEnum["NeNp"] = "ne-np";
    EmailUpdateRequestLanguageEnum["Nl"] = "nl";
    EmailUpdateRequestLanguageEnum["NlAw"] = "nl-aw";
    EmailUpdateRequestLanguageEnum["NlBe"] = "nl-be";
    EmailUpdateRequestLanguageEnum["NlBq"] = "nl-bq";
    EmailUpdateRequestLanguageEnum["NlCh"] = "nl-ch";
    EmailUpdateRequestLanguageEnum["NlCw"] = "nl-cw";
    EmailUpdateRequestLanguageEnum["NlLu"] = "nl-lu";
    EmailUpdateRequestLanguageEnum["NlNl"] = "nl-nl";
    EmailUpdateRequestLanguageEnum["NlSr"] = "nl-sr";
    EmailUpdateRequestLanguageEnum["NlSx"] = "nl-sx";
    EmailUpdateRequestLanguageEnum["Nmg"] = "nmg";
    EmailUpdateRequestLanguageEnum["NmgCm"] = "nmg-cm";
    EmailUpdateRequestLanguageEnum["Nn"] = "nn";
    EmailUpdateRequestLanguageEnum["Nnh"] = "nnh";
    EmailUpdateRequestLanguageEnum["NnhCm"] = "nnh-cm";
    EmailUpdateRequestLanguageEnum["NnNo"] = "nn-no";
    EmailUpdateRequestLanguageEnum["No"] = "no";
    EmailUpdateRequestLanguageEnum["NoNo"] = "no-no";
    EmailUpdateRequestLanguageEnum["Nus"] = "nus";
    EmailUpdateRequestLanguageEnum["NusSs"] = "nus-ss";
    EmailUpdateRequestLanguageEnum["Nyn"] = "nyn";
    EmailUpdateRequestLanguageEnum["NynUg"] = "nyn-ug";
    EmailUpdateRequestLanguageEnum["Oc"] = "oc";
    EmailUpdateRequestLanguageEnum["OcEs"] = "oc-es";
    EmailUpdateRequestLanguageEnum["OcFr"] = "oc-fr";
    EmailUpdateRequestLanguageEnum["Om"] = "om";
    EmailUpdateRequestLanguageEnum["OmEt"] = "om-et";
    EmailUpdateRequestLanguageEnum["OmKe"] = "om-ke";
    EmailUpdateRequestLanguageEnum["Or"] = "or";
    EmailUpdateRequestLanguageEnum["OrIn"] = "or-in";
    EmailUpdateRequestLanguageEnum["Os"] = "os";
    EmailUpdateRequestLanguageEnum["OsGe"] = "os-ge";
    EmailUpdateRequestLanguageEnum["OsRu"] = "os-ru";
    EmailUpdateRequestLanguageEnum["Pa"] = "pa";
    EmailUpdateRequestLanguageEnum["PaIn"] = "pa-in";
    EmailUpdateRequestLanguageEnum["PaPk"] = "pa-pk";
    EmailUpdateRequestLanguageEnum["Pcm"] = "pcm";
    EmailUpdateRequestLanguageEnum["PcmNg"] = "pcm-ng";
    EmailUpdateRequestLanguageEnum["Pis"] = "pis";
    EmailUpdateRequestLanguageEnum["PisSb"] = "pis-sb";
    EmailUpdateRequestLanguageEnum["Pl"] = "pl";
    EmailUpdateRequestLanguageEnum["PlPl"] = "pl-pl";
    EmailUpdateRequestLanguageEnum["Prg"] = "prg";
    EmailUpdateRequestLanguageEnum["Prg001"] = "prg-001";
    EmailUpdateRequestLanguageEnum["Ps"] = "ps";
    EmailUpdateRequestLanguageEnum["PsAf"] = "ps-af";
    EmailUpdateRequestLanguageEnum["PsPk"] = "ps-pk";
    EmailUpdateRequestLanguageEnum["Pt"] = "pt";
    EmailUpdateRequestLanguageEnum["PtAo"] = "pt-ao";
    EmailUpdateRequestLanguageEnum["PtBr"] = "pt-br";
    EmailUpdateRequestLanguageEnum["PtCh"] = "pt-ch";
    EmailUpdateRequestLanguageEnum["PtCv"] = "pt-cv";
    EmailUpdateRequestLanguageEnum["PtGq"] = "pt-gq";
    EmailUpdateRequestLanguageEnum["PtGw"] = "pt-gw";
    EmailUpdateRequestLanguageEnum["PtLu"] = "pt-lu";
    EmailUpdateRequestLanguageEnum["PtMo"] = "pt-mo";
    EmailUpdateRequestLanguageEnum["PtMz"] = "pt-mz";
    EmailUpdateRequestLanguageEnum["PtPt"] = "pt-pt";
    EmailUpdateRequestLanguageEnum["PtSt"] = "pt-st";
    EmailUpdateRequestLanguageEnum["PtTl"] = "pt-tl";
    EmailUpdateRequestLanguageEnum["Qu"] = "qu";
    EmailUpdateRequestLanguageEnum["QuBo"] = "qu-bo";
    EmailUpdateRequestLanguageEnum["QuEc"] = "qu-ec";
    EmailUpdateRequestLanguageEnum["QuPe"] = "qu-pe";
    EmailUpdateRequestLanguageEnum["Raj"] = "raj";
    EmailUpdateRequestLanguageEnum["RajIn"] = "raj-in";
    EmailUpdateRequestLanguageEnum["Rm"] = "rm";
    EmailUpdateRequestLanguageEnum["RmCh"] = "rm-ch";
    EmailUpdateRequestLanguageEnum["Rn"] = "rn";
    EmailUpdateRequestLanguageEnum["RnBi"] = "rn-bi";
    EmailUpdateRequestLanguageEnum["Ro"] = "ro";
    EmailUpdateRequestLanguageEnum["Rof"] = "rof";
    EmailUpdateRequestLanguageEnum["RofTz"] = "rof-tz";
    EmailUpdateRequestLanguageEnum["RoMd"] = "ro-md";
    EmailUpdateRequestLanguageEnum["RoRo"] = "ro-ro";
    EmailUpdateRequestLanguageEnum["Ru"] = "ru";
    EmailUpdateRequestLanguageEnum["RuBy"] = "ru-by";
    EmailUpdateRequestLanguageEnum["RuKg"] = "ru-kg";
    EmailUpdateRequestLanguageEnum["RuKz"] = "ru-kz";
    EmailUpdateRequestLanguageEnum["RuMd"] = "ru-md";
    EmailUpdateRequestLanguageEnum["RuRu"] = "ru-ru";
    EmailUpdateRequestLanguageEnum["RuUa"] = "ru-ua";
    EmailUpdateRequestLanguageEnum["Rw"] = "rw";
    EmailUpdateRequestLanguageEnum["Rwk"] = "rwk";
    EmailUpdateRequestLanguageEnum["RwkTz"] = "rwk-tz";
    EmailUpdateRequestLanguageEnum["RwRw"] = "rw-rw";
    EmailUpdateRequestLanguageEnum["Sa"] = "sa";
    EmailUpdateRequestLanguageEnum["Sah"] = "sah";
    EmailUpdateRequestLanguageEnum["SahRu"] = "sah-ru";
    EmailUpdateRequestLanguageEnum["Saq"] = "saq";
    EmailUpdateRequestLanguageEnum["SaqKe"] = "saq-ke";
    EmailUpdateRequestLanguageEnum["Sat"] = "sat";
    EmailUpdateRequestLanguageEnum["SatIn"] = "sat-in";
    EmailUpdateRequestLanguageEnum["SaIn"] = "sa-in";
    EmailUpdateRequestLanguageEnum["Sbp"] = "sbp";
    EmailUpdateRequestLanguageEnum["SbpTz"] = "sbp-tz";
    EmailUpdateRequestLanguageEnum["Sc"] = "sc";
    EmailUpdateRequestLanguageEnum["ScIt"] = "sc-it";
    EmailUpdateRequestLanguageEnum["Sd"] = "sd";
    EmailUpdateRequestLanguageEnum["SdIn"] = "sd-in";
    EmailUpdateRequestLanguageEnum["SdPk"] = "sd-pk";
    EmailUpdateRequestLanguageEnum["Se"] = "se";
    EmailUpdateRequestLanguageEnum["Seh"] = "seh";
    EmailUpdateRequestLanguageEnum["SehMz"] = "seh-mz";
    EmailUpdateRequestLanguageEnum["Ses"] = "ses";
    EmailUpdateRequestLanguageEnum["SesMl"] = "ses-ml";
    EmailUpdateRequestLanguageEnum["SeFi"] = "se-fi";
    EmailUpdateRequestLanguageEnum["SeNo"] = "se-no";
    EmailUpdateRequestLanguageEnum["SeSe"] = "se-se";
    EmailUpdateRequestLanguageEnum["Sg"] = "sg";
    EmailUpdateRequestLanguageEnum["SgCf"] = "sg-cf";
    EmailUpdateRequestLanguageEnum["Shi"] = "shi";
    EmailUpdateRequestLanguageEnum["ShiMa"] = "shi-ma";
    EmailUpdateRequestLanguageEnum["Si"] = "si";
    EmailUpdateRequestLanguageEnum["SiLk"] = "si-lk";
    EmailUpdateRequestLanguageEnum["Sk"] = "sk";
    EmailUpdateRequestLanguageEnum["SkSk"] = "sk-sk";
    EmailUpdateRequestLanguageEnum["Sl"] = "sl";
    EmailUpdateRequestLanguageEnum["SlSi"] = "sl-si";
    EmailUpdateRequestLanguageEnum["Smn"] = "smn";
    EmailUpdateRequestLanguageEnum["SmnFi"] = "smn-fi";
    EmailUpdateRequestLanguageEnum["Sms"] = "sms";
    EmailUpdateRequestLanguageEnum["SmsFi"] = "sms-fi";
    EmailUpdateRequestLanguageEnum["Sn"] = "sn";
    EmailUpdateRequestLanguageEnum["SnZw"] = "sn-zw";
    EmailUpdateRequestLanguageEnum["So"] = "so";
    EmailUpdateRequestLanguageEnum["SoDj"] = "so-dj";
    EmailUpdateRequestLanguageEnum["SoEt"] = "so-et";
    EmailUpdateRequestLanguageEnum["SoKe"] = "so-ke";
    EmailUpdateRequestLanguageEnum["SoSo"] = "so-so";
    EmailUpdateRequestLanguageEnum["Sq"] = "sq";
    EmailUpdateRequestLanguageEnum["SqAl"] = "sq-al";
    EmailUpdateRequestLanguageEnum["SqMk"] = "sq-mk";
    EmailUpdateRequestLanguageEnum["SqXk"] = "sq-xk";
    EmailUpdateRequestLanguageEnum["Sr"] = "sr";
    EmailUpdateRequestLanguageEnum["SrBa"] = "sr-ba";
    EmailUpdateRequestLanguageEnum["SrCs"] = "sr-cs";
    EmailUpdateRequestLanguageEnum["SrMe"] = "sr-me";
    EmailUpdateRequestLanguageEnum["SrRs"] = "sr-rs";
    EmailUpdateRequestLanguageEnum["SrXk"] = "sr-xk";
    EmailUpdateRequestLanguageEnum["Su"] = "su";
    EmailUpdateRequestLanguageEnum["SuId"] = "su-id";
    EmailUpdateRequestLanguageEnum["Sv"] = "sv";
    EmailUpdateRequestLanguageEnum["SvAx"] = "sv-ax";
    EmailUpdateRequestLanguageEnum["SvFi"] = "sv-fi";
    EmailUpdateRequestLanguageEnum["SvSe"] = "sv-se";
    EmailUpdateRequestLanguageEnum["Sw"] = "sw";
    EmailUpdateRequestLanguageEnum["SwCd"] = "sw-cd";
    EmailUpdateRequestLanguageEnum["SwKe"] = "sw-ke";
    EmailUpdateRequestLanguageEnum["SwTz"] = "sw-tz";
    EmailUpdateRequestLanguageEnum["SwUg"] = "sw-ug";
    EmailUpdateRequestLanguageEnum["Sy"] = "sy";
    EmailUpdateRequestLanguageEnum["Ta"] = "ta";
    EmailUpdateRequestLanguageEnum["TaIn"] = "ta-in";
    EmailUpdateRequestLanguageEnum["TaLk"] = "ta-lk";
    EmailUpdateRequestLanguageEnum["TaMy"] = "ta-my";
    EmailUpdateRequestLanguageEnum["TaSg"] = "ta-sg";
    EmailUpdateRequestLanguageEnum["Te"] = "te";
    EmailUpdateRequestLanguageEnum["Teo"] = "teo";
    EmailUpdateRequestLanguageEnum["TeoKe"] = "teo-ke";
    EmailUpdateRequestLanguageEnum["TeoUg"] = "teo-ug";
    EmailUpdateRequestLanguageEnum["TeIn"] = "te-in";
    EmailUpdateRequestLanguageEnum["Tg"] = "tg";
    EmailUpdateRequestLanguageEnum["TgTj"] = "tg-tj";
    EmailUpdateRequestLanguageEnum["Th"] = "th";
    EmailUpdateRequestLanguageEnum["ThTh"] = "th-th";
    EmailUpdateRequestLanguageEnum["Ti"] = "ti";
    EmailUpdateRequestLanguageEnum["TiEr"] = "ti-er";
    EmailUpdateRequestLanguageEnum["TiEt"] = "ti-et";
    EmailUpdateRequestLanguageEnum["Tk"] = "tk";
    EmailUpdateRequestLanguageEnum["TkTm"] = "tk-tm";
    EmailUpdateRequestLanguageEnum["Tl"] = "tl";
    EmailUpdateRequestLanguageEnum["To"] = "to";
    EmailUpdateRequestLanguageEnum["Tok"] = "tok";
    EmailUpdateRequestLanguageEnum["Tok001"] = "tok-001";
    EmailUpdateRequestLanguageEnum["ToTo"] = "to-to";
    EmailUpdateRequestLanguageEnum["Tr"] = "tr";
    EmailUpdateRequestLanguageEnum["TrCy"] = "tr-cy";
    EmailUpdateRequestLanguageEnum["TrTr"] = "tr-tr";
    EmailUpdateRequestLanguageEnum["Tt"] = "tt";
    EmailUpdateRequestLanguageEnum["TtRu"] = "tt-ru";
    EmailUpdateRequestLanguageEnum["Twq"] = "twq";
    EmailUpdateRequestLanguageEnum["TwqNe"] = "twq-ne";
    EmailUpdateRequestLanguageEnum["Tzm"] = "tzm";
    EmailUpdateRequestLanguageEnum["TzmMa"] = "tzm-ma";
    EmailUpdateRequestLanguageEnum["Ug"] = "ug";
    EmailUpdateRequestLanguageEnum["UgCn"] = "ug-cn";
    EmailUpdateRequestLanguageEnum["Uk"] = "uk";
    EmailUpdateRequestLanguageEnum["UkUa"] = "uk-ua";
    EmailUpdateRequestLanguageEnum["Ur"] = "ur";
    EmailUpdateRequestLanguageEnum["UrIn"] = "ur-in";
    EmailUpdateRequestLanguageEnum["UrPk"] = "ur-pk";
    EmailUpdateRequestLanguageEnum["Uz"] = "uz";
    EmailUpdateRequestLanguageEnum["UzAf"] = "uz-af";
    EmailUpdateRequestLanguageEnum["UzUz"] = "uz-uz";
    EmailUpdateRequestLanguageEnum["Vai"] = "vai";
    EmailUpdateRequestLanguageEnum["VaiLr"] = "vai-lr";
    EmailUpdateRequestLanguageEnum["Vi"] = "vi";
    EmailUpdateRequestLanguageEnum["ViVn"] = "vi-vn";
    EmailUpdateRequestLanguageEnum["Vo"] = "vo";
    EmailUpdateRequestLanguageEnum["Vo001"] = "vo-001";
    EmailUpdateRequestLanguageEnum["Vun"] = "vun";
    EmailUpdateRequestLanguageEnum["VunTz"] = "vun-tz";
    EmailUpdateRequestLanguageEnum["Wae"] = "wae";
    EmailUpdateRequestLanguageEnum["WaeCh"] = "wae-ch";
    EmailUpdateRequestLanguageEnum["Wo"] = "wo";
    EmailUpdateRequestLanguageEnum["WoSn"] = "wo-sn";
    EmailUpdateRequestLanguageEnum["Xh"] = "xh";
    EmailUpdateRequestLanguageEnum["XhZa"] = "xh-za";
    EmailUpdateRequestLanguageEnum["Xog"] = "xog";
    EmailUpdateRequestLanguageEnum["XogUg"] = "xog-ug";
    EmailUpdateRequestLanguageEnum["Yav"] = "yav";
    EmailUpdateRequestLanguageEnum["YavCm"] = "yav-cm";
    EmailUpdateRequestLanguageEnum["Yo"] = "yo";
    EmailUpdateRequestLanguageEnum["YoBj"] = "yo-bj";
    EmailUpdateRequestLanguageEnum["YoNg"] = "yo-ng";
    EmailUpdateRequestLanguageEnum["Yrl"] = "yrl";
    EmailUpdateRequestLanguageEnum["YrlBr"] = "yrl-br";
    EmailUpdateRequestLanguageEnum["YrlCo"] = "yrl-co";
    EmailUpdateRequestLanguageEnum["YrlVe"] = "yrl-ve";
    EmailUpdateRequestLanguageEnum["Yue"] = "yue";
    EmailUpdateRequestLanguageEnum["YueCn"] = "yue-cn";
    EmailUpdateRequestLanguageEnum["YueHk"] = "yue-hk";
    EmailUpdateRequestLanguageEnum["Zgh"] = "zgh";
    EmailUpdateRequestLanguageEnum["ZghMa"] = "zgh-ma";
    EmailUpdateRequestLanguageEnum["Zh"] = "zh";
    EmailUpdateRequestLanguageEnum["ZhCn"] = "zh-cn";
    EmailUpdateRequestLanguageEnum["ZhHans"] = "zh-hans";
    EmailUpdateRequestLanguageEnum["ZhHant"] = "zh-hant";
    EmailUpdateRequestLanguageEnum["ZhHk"] = "zh-hk";
    EmailUpdateRequestLanguageEnum["ZhMo"] = "zh-mo";
    EmailUpdateRequestLanguageEnum["ZhSg"] = "zh-sg";
    EmailUpdateRequestLanguageEnum["ZhTw"] = "zh-tw";
    EmailUpdateRequestLanguageEnum["Zu"] = "zu";
    EmailUpdateRequestLanguageEnum["ZuZa"] = "zu-za";
})(EmailUpdateRequestLanguageEnum = exports.EmailUpdateRequestLanguageEnum || (exports.EmailUpdateRequestLanguageEnum = {}));
var EmailUpdateRequestStateEnum;
(function (EmailUpdateRequestStateEnum) {
    EmailUpdateRequestStateEnum["Automated"] = "AUTOMATED";
    EmailUpdateRequestStateEnum["AutomatedDraft"] = "AUTOMATED_DRAFT";
    EmailUpdateRequestStateEnum["AutomatedSending"] = "AUTOMATED_SENDING";
    EmailUpdateRequestStateEnum["AutomatedForForm"] = "AUTOMATED_FOR_FORM";
    EmailUpdateRequestStateEnum["AutomatedForFormBuffer"] = "AUTOMATED_FOR_FORM_BUFFER";
    EmailUpdateRequestStateEnum["AutomatedForFormDraft"] = "AUTOMATED_FOR_FORM_DRAFT";
    EmailUpdateRequestStateEnum["AutomatedForFormLegacy"] = "AUTOMATED_FOR_FORM_LEGACY";
    EmailUpdateRequestStateEnum["BlogEmailDraft"] = "BLOG_EMAIL_DRAFT";
    EmailUpdateRequestStateEnum["BlogEmailPublished"] = "BLOG_EMAIL_PUBLISHED";
    EmailUpdateRequestStateEnum["Draft"] = "DRAFT";
    EmailUpdateRequestStateEnum["DraftAb"] = "DRAFT_AB";
    EmailUpdateRequestStateEnum["DraftAbVariant"] = "DRAFT_AB_VARIANT";
    EmailUpdateRequestStateEnum["Error"] = "ERROR";
    EmailUpdateRequestStateEnum["LoserAbVariant"] = "LOSER_AB_VARIANT";
    EmailUpdateRequestStateEnum["PageStub"] = "PAGE_STUB";
    EmailUpdateRequestStateEnum["PreProcessing"] = "PRE_PROCESSING";
    EmailUpdateRequestStateEnum["Processing"] = "PROCESSING";
    EmailUpdateRequestStateEnum["Published"] = "PUBLISHED";
    EmailUpdateRequestStateEnum["PublishedAb"] = "PUBLISHED_AB";
    EmailUpdateRequestStateEnum["PublishedAbVariant"] = "PUBLISHED_AB_VARIANT";
    EmailUpdateRequestStateEnum["PublishedOrScheduled"] = "PUBLISHED_OR_SCHEDULED";
    EmailUpdateRequestStateEnum["RssToEmailDraft"] = "RSS_TO_EMAIL_DRAFT";
    EmailUpdateRequestStateEnum["RssToEmailPublished"] = "RSS_TO_EMAIL_PUBLISHED";
    EmailUpdateRequestStateEnum["Scheduled"] = "SCHEDULED";
    EmailUpdateRequestStateEnum["ScheduledAb"] = "SCHEDULED_AB";
    EmailUpdateRequestStateEnum["ScheduledOrPublished"] = "SCHEDULED_OR_PUBLISHED";
    EmailUpdateRequestStateEnum["AutomatedAb"] = "AUTOMATED_AB";
    EmailUpdateRequestStateEnum["AutomatedAbVariant"] = "AUTOMATED_AB_VARIANT";
    EmailUpdateRequestStateEnum["AutomatedDraftAb"] = "AUTOMATED_DRAFT_AB";
    EmailUpdateRequestStateEnum["AutomatedDraftAbvariant"] = "AUTOMATED_DRAFT_ABVARIANT";
    EmailUpdateRequestStateEnum["AutomatedLoserAbvariant"] = "AUTOMATED_LOSER_ABVARIANT";
})(EmailUpdateRequestStateEnum = exports.EmailUpdateRequestStateEnum || (exports.EmailUpdateRequestStateEnum = {}));
var EmailUpdateRequestSubcategoryEnum;
(function (EmailUpdateRequestSubcategoryEnum) {
    EmailUpdateRequestSubcategoryEnum["AbMaster"] = "ab_master";
    EmailUpdateRequestSubcategoryEnum["AbVariant"] = "ab_variant";
    EmailUpdateRequestSubcategoryEnum["AbLoserVariant"] = "ab_loser_variant";
    EmailUpdateRequestSubcategoryEnum["PageStub"] = "page_stub";
    EmailUpdateRequestSubcategoryEnum["LandingPage"] = "landing_page";
    EmailUpdateRequestSubcategoryEnum["SitePage"] = "site_page";
    EmailUpdateRequestSubcategoryEnum["LegacyPage"] = "legacy_page";
    EmailUpdateRequestSubcategoryEnum["AbMasterSitePage"] = "ab_master_site_page";
    EmailUpdateRequestSubcategoryEnum["AbVariantSitePage"] = "ab_variant_site_page";
    EmailUpdateRequestSubcategoryEnum["AbLoserVariantSitePage"] = "ab_loser_variant_site_page";
    EmailUpdateRequestSubcategoryEnum["PerformableLandingPage"] = "performable_landing_page";
    EmailUpdateRequestSubcategoryEnum["PerformableLandingPageCutover"] = "performable_landing_page_cutover";
    EmailUpdateRequestSubcategoryEnum["StagedPage"] = "staged_page";
    EmailUpdateRequestSubcategoryEnum["Automated"] = "automated";
    EmailUpdateRequestSubcategoryEnum["AutomatedForDeal"] = "automated_for_deal";
    EmailUpdateRequestSubcategoryEnum["AutomatedForForm"] = "automated_for_form";
    EmailUpdateRequestSubcategoryEnum["AutomatedForFormLegacy"] = "automated_for_form_legacy";
    EmailUpdateRequestSubcategoryEnum["AutomatedForFormBuffer"] = "automated_for_form_buffer";
    EmailUpdateRequestSubcategoryEnum["AutomatedForFormDraft"] = "automated_for_form_draft";
    EmailUpdateRequestSubcategoryEnum["RssToEmail"] = "rss_to_email";
    EmailUpdateRequestSubcategoryEnum["RssToEmailChild"] = "rss_to_email_child";
    EmailUpdateRequestSubcategoryEnum["BlogEmail"] = "blog_email";
    EmailUpdateRequestSubcategoryEnum["BlogEmailChild"] = "blog_email_child";
    EmailUpdateRequestSubcategoryEnum["OptinEmail"] = "optin_email";
    EmailUpdateRequestSubcategoryEnum["OptinFollowupEmail"] = "optin_followup_email";
    EmailUpdateRequestSubcategoryEnum["Batch"] = "batch";
    EmailUpdateRequestSubcategoryEnum["ResubscribeEmail"] = "resubscribe_email";
    EmailUpdateRequestSubcategoryEnum["UnsubscribeConfirmationEmail"] = "unsubscribe_confirmation_email";
    EmailUpdateRequestSubcategoryEnum["ResubscribeConfirmationEmail"] = "resubscribe_confirmation_email";
    EmailUpdateRequestSubcategoryEnum["SingleSendApi"] = "single_send_api";
    EmailUpdateRequestSubcategoryEnum["MarketingSingleSendApi"] = "marketing_single_send_api";
    EmailUpdateRequestSubcategoryEnum["SmtpToken"] = "smtp_token";
    EmailUpdateRequestSubcategoryEnum["Localtime"] = "localtime";
    EmailUpdateRequestSubcategoryEnum["AutomatedForTicket"] = "automated_for_ticket";
    EmailUpdateRequestSubcategoryEnum["AutomatedForLeadflow"] = "automated_for_leadflow";
    EmailUpdateRequestSubcategoryEnum["AutomatedForFeedbackCes"] = "automated_for_feedback_ces";
    EmailUpdateRequestSubcategoryEnum["AutomatedForFeedbackNps"] = "automated_for_feedback_nps";
    EmailUpdateRequestSubcategoryEnum["AutomatedForFeedbackCustom"] = "automated_for_feedback_custom";
    EmailUpdateRequestSubcategoryEnum["MembershipRegistration"] = "membership_registration";
    EmailUpdateRequestSubcategoryEnum["MembershipPasswordSaved"] = "membership_password_saved";
    EmailUpdateRequestSubcategoryEnum["MembershipPasswordReset"] = "membership_password_reset";
    EmailUpdateRequestSubcategoryEnum["MembershipOtpLogin"] = "membership_otp_login";
    EmailUpdateRequestSubcategoryEnum["MembershipPasswordlessAuth"] = "membership_passwordless_auth";
    EmailUpdateRequestSubcategoryEnum["MembershipEmailVerification"] = "membership_email_verification";
    EmailUpdateRequestSubcategoryEnum["MembershipRegistrationFollowUp"] = "membership_registration_follow_up";
    EmailUpdateRequestSubcategoryEnum["MembershipVerification"] = "membership_verification";
    EmailUpdateRequestSubcategoryEnum["MembershipFollowUp"] = "membership_follow_up";
    EmailUpdateRequestSubcategoryEnum["TicketClosedKickbackEmail"] = "ticket_closed_kickback_email";
    EmailUpdateRequestSubcategoryEnum["TicketOpenedKickbackEmail"] = "ticket_opened_kickback_email";
    EmailUpdateRequestSubcategoryEnum["AutomatedForCustomSurvey"] = "automated_for_custom_survey";
    EmailUpdateRequestSubcategoryEnum["DiscardableStub"] = "discardable_stub";
    EmailUpdateRequestSubcategoryEnum["NormalBlogPost"] = "normal_blog_post";
    EmailUpdateRequestSubcategoryEnum["LegacyBlogPost"] = "legacy_blog_post";
    EmailUpdateRequestSubcategoryEnum["ImportedBlogPost"] = "imported_blog_post";
    EmailUpdateRequestSubcategoryEnum["AutomatedAbMaster"] = "automated_ab_master";
    EmailUpdateRequestSubcategoryEnum["AutomatedAbVariant"] = "automated_ab_variant";
    EmailUpdateRequestSubcategoryEnum["WebInteractive"] = "web_interactive";
    EmailUpdateRequestSubcategoryEnum["PortalContent"] = "portal_content";
    EmailUpdateRequestSubcategoryEnum["PageInstanceLayout"] = "page_instance_layout";
    EmailUpdateRequestSubcategoryEnum["KbArticleInstanceLayout"] = "kb_article_instance_layout";
    EmailUpdateRequestSubcategoryEnum["KbListing"] = "kb_listing";
    EmailUpdateRequestSubcategoryEnum["KbSearchResults"] = "kb_search_results";
    EmailUpdateRequestSubcategoryEnum["KbSupportForm"] = "kb_support_form";
    EmailUpdateRequestSubcategoryEnum["Kb404Page"] = "kb_404_page";
    EmailUpdateRequestSubcategoryEnum["CaseStudy"] = "case_study";
    EmailUpdateRequestSubcategoryEnum["CaseStudyListing"] = "case_study_listing";
    EmailUpdateRequestSubcategoryEnum["CaseStudyInstanceLayout"] = "case_study_instance_layout";
    EmailUpdateRequestSubcategoryEnum["ScpStaticPage"] = "scp_static_page";
    EmailUpdateRequestSubcategoryEnum["ScpInstanceLayoutPage"] = "scp_instance_layout_page";
    EmailUpdateRequestSubcategoryEnum["PodcastInstanceLayout"] = "podcast_instance_layout";
    EmailUpdateRequestSubcategoryEnum["PodcastListing"] = "podcast_listing";
    EmailUpdateRequestSubcategoryEnum["BlogArticleInstanceLayout"] = "blog_article_instance_layout";
    EmailUpdateRequestSubcategoryEnum["BlogArticleListing"] = "blog_article_listing";
    EmailUpdateRequestSubcategoryEnum["BlogAuthorDetail"] = "blog_author_detail";
    EmailUpdateRequestSubcategoryEnum["Unknown"] = "UNKNOWN";
})(EmailUpdateRequestSubcategoryEnum = exports.EmailUpdateRequestSubcategoryEnum || (exports.EmailUpdateRequestSubcategoryEnum = {}));
