// @hubspot/api-client/lib/codegen/marketing/emails/models/PublicEmail.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicEmailStateEnum = exports.PublicEmailTypeEnum = exports.PublicEmailLanguageEnum = exports.PublicEmail = void 0;
class PublicEmail {
    static getAttributeTypeMap() {
        return PublicEmail.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicEmail = PublicEmail;
PublicEmail.discriminator = undefined;
PublicEmail.mapping = undefined;
PublicEmail.attributeTypeMap = [
    {
        "name": "feedbackSurveyId",
        "baseName": "feedbackSurveyId",
        "type": "string",
        "format": ""
    },
    {
        "name": "subject",
        "baseName": "subject",
        "type": "string",
        "format": ""
    },
    {
        "name": "publishDate",
        "baseName": "publishDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "isTransactional",
        "baseName": "isTransactional",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "PublicEmailLanguageEnum",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PublicEmailTypeEnum",
        "format": ""
    },
    {
        "name": "content",
        "baseName": "content",
        "type": "PublicEmailContent",
        "format": ""
    },
    {
        "name": "businessUnitId",
        "baseName": "businessUnitId",
        "type": "string",
        "format": ""
    },
    {
        "name": "webversion",
        "baseName": "webversion",
        "type": "PublicWebversionDetails",
        "format": ""
    },
    {
        "name": "workflowNames",
        "baseName": "workflowNames",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "archived",
        "baseName": "archived",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "stats",
        "baseName": "stats",
        "type": "EmailStatisticsData",
        "format": ""
    },
    {
        "name": "jitterSendTime",
        "baseName": "jitterSendTime",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "_from",
        "baseName": "from",
        "type": "PublicEmailFromDetails",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "state",
        "baseName": "state",
        "type": "PublicEmailStateEnum",
        "format": ""
    },
    {
        "name": "createdById",
        "baseName": "createdById",
        "type": "string",
        "format": ""
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "clonedFrom",
        "baseName": "clonedFrom",
        "type": "string",
        "format": ""
    },
    {
        "name": "rssData",
        "baseName": "rssData",
        "type": "PublicRssEmailDetails",
        "format": ""
    },
    {
        "name": "publishedAt",
        "baseName": "publishedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "publishedById",
        "baseName": "publishedById",
        "type": "string",
        "format": ""
    },
    {
        "name": "isPublished",
        "baseName": "isPublished",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "testing",
        "baseName": "testing",
        "type": "PublicEmailTestingDetails",
        "format": ""
    },
    {
        "name": "updatedById",
        "baseName": "updatedById",
        "type": "string",
        "format": ""
    },
    {
        "name": "folderId",
        "baseName": "folderId",
        "type": "number",
        "format": "int64"
    },
    {
        "name": "subscriptionDetails",
        "baseName": "subscriptionDetails",
        "type": "PublicEmailSubscriptionDetails",
        "format": ""
    },
    {
        "name": "deletedAt",
        "baseName": "deletedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "activeDomain",
        "baseName": "activeDomain",
        "type": "string",
        "format": ""
    },
    {
        "name": "campaign",
        "baseName": "campaign",
        "type": "string",
        "format": ""
    },
    {
        "name": "to",
        "baseName": "to",
        "type": "PublicEmailToDetails",
        "format": ""
    },
    {
        "name": "subcategory",
        "baseName": "subcategory",
        "type": "string",
        "format": ""
    },
    {
        "name": "campaignName",
        "baseName": "campaignName",
        "type": "string",
        "format": ""
    },
    {
        "name": "sendOnPublish",
        "baseName": "sendOnPublish",
        "type": "boolean",
        "format": ""
    }
];
var PublicEmailLanguageEnum;
(function (PublicEmailLanguageEnum) {
    PublicEmailLanguageEnum["Af"] = "af";
    PublicEmailLanguageEnum["AfNa"] = "af-na";
    PublicEmailLanguageEnum["AfZa"] = "af-za";
    PublicEmailLanguageEnum["Agq"] = "agq";
    PublicEmailLanguageEnum["AgqCm"] = "agq-cm";
    PublicEmailLanguageEnum["Ak"] = "ak";
    PublicEmailLanguageEnum["AkGh"] = "ak-gh";
    PublicEmailLanguageEnum["Am"] = "am";
    PublicEmailLanguageEnum["AmEt"] = "am-et";
    PublicEmailLanguageEnum["Ann"] = "ann";
    PublicEmailLanguageEnum["AnnNg"] = "ann-ng";
    PublicEmailLanguageEnum["Ar"] = "ar";
    PublicEmailLanguageEnum["Ar001"] = "ar-001";
    PublicEmailLanguageEnum["ArAe"] = "ar-ae";
    PublicEmailLanguageEnum["ArBh"] = "ar-bh";
    PublicEmailLanguageEnum["ArDj"] = "ar-dj";
    PublicEmailLanguageEnum["ArDz"] = "ar-dz";
    PublicEmailLanguageEnum["ArEg"] = "ar-eg";
    PublicEmailLanguageEnum["ArEh"] = "ar-eh";
    PublicEmailLanguageEnum["ArEr"] = "ar-er";
    PublicEmailLanguageEnum["ArIl"] = "ar-il";
    PublicEmailLanguageEnum["ArIq"] = "ar-iq";
    PublicEmailLanguageEnum["ArJo"] = "ar-jo";
    PublicEmailLanguageEnum["ArKm"] = "ar-km";
    PublicEmailLanguageEnum["ArKw"] = "ar-kw";
    PublicEmailLanguageEnum["ArLb"] = "ar-lb";
    PublicEmailLanguageEnum["ArLy"] = "ar-ly";
    PublicEmailLanguageEnum["ArMa"] = "ar-ma";
    PublicEmailLanguageEnum["ArMr"] = "ar-mr";
    PublicEmailLanguageEnum["ArOm"] = "ar-om";
    PublicEmailLanguageEnum["ArPs"] = "ar-ps";
    PublicEmailLanguageEnum["ArQa"] = "ar-qa";
    PublicEmailLanguageEnum["ArSa"] = "ar-sa";
    PublicEmailLanguageEnum["ArSd"] = "ar-sd";
    PublicEmailLanguageEnum["ArSo"] = "ar-so";
    PublicEmailLanguageEnum["ArSs"] = "ar-ss";
    PublicEmailLanguageEnum["ArSy"] = "ar-sy";
    PublicEmailLanguageEnum["ArTd"] = "ar-td";
    PublicEmailLanguageEnum["ArTn"] = "ar-tn";
    PublicEmailLanguageEnum["ArYe"] = "ar-ye";
    PublicEmailLanguageEnum["As"] = "as";
    PublicEmailLanguageEnum["Asa"] = "asa";
    PublicEmailLanguageEnum["AsaTz"] = "asa-tz";
    PublicEmailLanguageEnum["Ast"] = "ast";
    PublicEmailLanguageEnum["AstEs"] = "ast-es";
    PublicEmailLanguageEnum["AsIn"] = "as-in";
    PublicEmailLanguageEnum["Az"] = "az";
    PublicEmailLanguageEnum["AzAz"] = "az-az";
    PublicEmailLanguageEnum["Bas"] = "bas";
    PublicEmailLanguageEnum["BasCm"] = "bas-cm";
    PublicEmailLanguageEnum["Be"] = "be";
    PublicEmailLanguageEnum["Bem"] = "bem";
    PublicEmailLanguageEnum["BemZm"] = "bem-zm";
    PublicEmailLanguageEnum["Bez"] = "bez";
    PublicEmailLanguageEnum["BezTz"] = "bez-tz";
    PublicEmailLanguageEnum["BeBy"] = "be-by";
    PublicEmailLanguageEnum["Bg"] = "bg";
    PublicEmailLanguageEnum["Bgc"] = "bgc";
    PublicEmailLanguageEnum["BgcIn"] = "bgc-in";
    PublicEmailLanguageEnum["BgBg"] = "bg-bg";
    PublicEmailLanguageEnum["Bho"] = "bho";
    PublicEmailLanguageEnum["BhoIn"] = "bho-in";
    PublicEmailLanguageEnum["Bm"] = "bm";
    PublicEmailLanguageEnum["BmMl"] = "bm-ml";
    PublicEmailLanguageEnum["Bn"] = "bn";
    PublicEmailLanguageEnum["BnBd"] = "bn-bd";
    PublicEmailLanguageEnum["BnIn"] = "bn-in";
    PublicEmailLanguageEnum["Bo"] = "bo";
    PublicEmailLanguageEnum["BoCn"] = "bo-cn";
    PublicEmailLanguageEnum["BoIn"] = "bo-in";
    PublicEmailLanguageEnum["Br"] = "br";
    PublicEmailLanguageEnum["Brx"] = "brx";
    PublicEmailLanguageEnum["BrxIn"] = "brx-in";
    PublicEmailLanguageEnum["BrFr"] = "br-fr";
    PublicEmailLanguageEnum["Bs"] = "bs";
    PublicEmailLanguageEnum["BsBa"] = "bs-ba";
    PublicEmailLanguageEnum["Ca"] = "ca";
    PublicEmailLanguageEnum["CaAd"] = "ca-ad";
    PublicEmailLanguageEnum["CaEs"] = "ca-es";
    PublicEmailLanguageEnum["CaFr"] = "ca-fr";
    PublicEmailLanguageEnum["CaIt"] = "ca-it";
    PublicEmailLanguageEnum["Ccp"] = "ccp";
    PublicEmailLanguageEnum["CcpBd"] = "ccp-bd";
    PublicEmailLanguageEnum["CcpIn"] = "ccp-in";
    PublicEmailLanguageEnum["Ce"] = "ce";
    PublicEmailLanguageEnum["Ceb"] = "ceb";
    PublicEmailLanguageEnum["CebPh"] = "ceb-ph";
    PublicEmailLanguageEnum["CeRu"] = "ce-ru";
    PublicEmailLanguageEnum["Cgg"] = "cgg";
    PublicEmailLanguageEnum["CggUg"] = "cgg-ug";
    PublicEmailLanguageEnum["Chr"] = "chr";
    PublicEmailLanguageEnum["ChrUs"] = "chr-us";
    PublicEmailLanguageEnum["Ckb"] = "ckb";
    PublicEmailLanguageEnum["CkbIq"] = "ckb-iq";
    PublicEmailLanguageEnum["CkbIr"] = "ckb-ir";
    PublicEmailLanguageEnum["Cs"] = "cs";
    PublicEmailLanguageEnum["CsCz"] = "cs-cz";
    PublicEmailLanguageEnum["Cu"] = "cu";
    PublicEmailLanguageEnum["CuRu"] = "cu-ru";
    PublicEmailLanguageEnum["Cv"] = "cv";
    PublicEmailLanguageEnum["CvRu"] = "cv-ru";
    PublicEmailLanguageEnum["Cy"] = "cy";
    PublicEmailLanguageEnum["CyGb"] = "cy-gb";
    PublicEmailLanguageEnum["Da"] = "da";
    PublicEmailLanguageEnum["Dav"] = "dav";
    PublicEmailLanguageEnum["DavKe"] = "dav-ke";
    PublicEmailLanguageEnum["DaDk"] = "da-dk";
    PublicEmailLanguageEnum["DaGl"] = "da-gl";
    PublicEmailLanguageEnum["De"] = "de";
    PublicEmailLanguageEnum["DeAt"] = "de-at";
    PublicEmailLanguageEnum["DeBe"] = "de-be";
    PublicEmailLanguageEnum["DeCh"] = "de-ch";
    PublicEmailLanguageEnum["DeDe"] = "de-de";
    PublicEmailLanguageEnum["DeGr"] = "de-gr";
    PublicEmailLanguageEnum["DeIt"] = "de-it";
    PublicEmailLanguageEnum["DeLi"] = "de-li";
    PublicEmailLanguageEnum["DeLu"] = "de-lu";
    PublicEmailLanguageEnum["Dje"] = "dje";
    PublicEmailLanguageEnum["DjeNe"] = "dje-ne";
    PublicEmailLanguageEnum["Doi"] = "doi";
    PublicEmailLanguageEnum["DoiIn"] = "doi-in";
    PublicEmailLanguageEnum["Dsb"] = "dsb";
    PublicEmailLanguageEnum["DsbDe"] = "dsb-de";
    PublicEmailLanguageEnum["Dua"] = "dua";
    PublicEmailLanguageEnum["DuaCm"] = "dua-cm";
    PublicEmailLanguageEnum["Dyo"] = "dyo";
    PublicEmailLanguageEnum["DyoSn"] = "dyo-sn";
    PublicEmailLanguageEnum["Dz"] = "dz";
    PublicEmailLanguageEnum["DzBt"] = "dz-bt";
    PublicEmailLanguageEnum["Ebu"] = "ebu";
    PublicEmailLanguageEnum["EbuKe"] = "ebu-ke";
    PublicEmailLanguageEnum["Ee"] = "ee";
    PublicEmailLanguageEnum["EeGh"] = "ee-gh";
    PublicEmailLanguageEnum["EeTg"] = "ee-tg";
    PublicEmailLanguageEnum["El"] = "el";
    PublicEmailLanguageEnum["ElCy"] = "el-cy";
    PublicEmailLanguageEnum["ElGr"] = "el-gr";
    PublicEmailLanguageEnum["En"] = "en";
    PublicEmailLanguageEnum["En001"] = "en-001";
    PublicEmailLanguageEnum["En150"] = "en-150";
    PublicEmailLanguageEnum["EnAe"] = "en-ae";
    PublicEmailLanguageEnum["EnAg"] = "en-ag";
    PublicEmailLanguageEnum["EnAi"] = "en-ai";
    PublicEmailLanguageEnum["EnAs"] = "en-as";
    PublicEmailLanguageEnum["EnAt"] = "en-at";
    PublicEmailLanguageEnum["EnAu"] = "en-au";
    PublicEmailLanguageEnum["EnBb"] = "en-bb";
    PublicEmailLanguageEnum["EnBe"] = "en-be";
    PublicEmailLanguageEnum["EnBi"] = "en-bi";
    PublicEmailLanguageEnum["EnBm"] = "en-bm";
    PublicEmailLanguageEnum["EnBs"] = "en-bs";
    PublicEmailLanguageEnum["EnBw"] = "en-bw";
    PublicEmailLanguageEnum["EnBz"] = "en-bz";
    PublicEmailLanguageEnum["EnCa"] = "en-ca";
    PublicEmailLanguageEnum["EnCc"] = "en-cc";
    PublicEmailLanguageEnum["EnCh"] = "en-ch";
    PublicEmailLanguageEnum["EnCk"] = "en-ck";
    PublicEmailLanguageEnum["EnCm"] = "en-cm";
    PublicEmailLanguageEnum["EnCn"] = "en-cn";
    PublicEmailLanguageEnum["EnCx"] = "en-cx";
    PublicEmailLanguageEnum["EnCy"] = "en-cy";
    PublicEmailLanguageEnum["EnDe"] = "en-de";
    PublicEmailLanguageEnum["EnDg"] = "en-dg";
    PublicEmailLanguageEnum["EnDk"] = "en-dk";
    PublicEmailLanguageEnum["EnDm"] = "en-dm";
    PublicEmailLanguageEnum["EnEe"] = "en-ee";
    PublicEmailLanguageEnum["EnEr"] = "en-er";
    PublicEmailLanguageEnum["EnEs"] = "en-es";
    PublicEmailLanguageEnum["EnFi"] = "en-fi";
    PublicEmailLanguageEnum["EnFj"] = "en-fj";
    PublicEmailLanguageEnum["EnFk"] = "en-fk";
    PublicEmailLanguageEnum["EnFm"] = "en-fm";
    PublicEmailLanguageEnum["EnFr"] = "en-fr";
    PublicEmailLanguageEnum["EnGb"] = "en-gb";
    PublicEmailLanguageEnum["EnGd"] = "en-gd";
    PublicEmailLanguageEnum["EnGg"] = "en-gg";
    PublicEmailLanguageEnum["EnGh"] = "en-gh";
    PublicEmailLanguageEnum["EnGi"] = "en-gi";
    PublicEmailLanguageEnum["EnGm"] = "en-gm";
    PublicEmailLanguageEnum["EnGu"] = "en-gu";
    PublicEmailLanguageEnum["EnGy"] = "en-gy";
    PublicEmailLanguageEnum["EnHk"] = "en-hk";
    PublicEmailLanguageEnum["EnIe"] = "en-ie";
    PublicEmailLanguageEnum["EnIl"] = "en-il";
    PublicEmailLanguageEnum["EnIm"] = "en-im";
    PublicEmailLanguageEnum["EnIn"] = "en-in";
    PublicEmailLanguageEnum["EnIo"] = "en-io";
    PublicEmailLanguageEnum["EnJe"] = "en-je";
    PublicEmailLanguageEnum["EnJm"] = "en-jm";
    PublicEmailLanguageEnum["EnKe"] = "en-ke";
    PublicEmailLanguageEnum["EnKi"] = "en-ki";
    PublicEmailLanguageEnum["EnKn"] = "en-kn";
    PublicEmailLanguageEnum["EnKy"] = "en-ky";
    PublicEmailLanguageEnum["EnLc"] = "en-lc";
    PublicEmailLanguageEnum["EnLr"] = "en-lr";
    PublicEmailLanguageEnum["EnLs"] = "en-ls";
    PublicEmailLanguageEnum["EnLu"] = "en-lu";
    PublicEmailLanguageEnum["EnMg"] = "en-mg";
    PublicEmailLanguageEnum["EnMh"] = "en-mh";
    PublicEmailLanguageEnum["EnMo"] = "en-mo";
    PublicEmailLanguageEnum["EnMp"] = "en-mp";
    PublicEmailLanguageEnum["EnMs"] = "en-ms";
    PublicEmailLanguageEnum["EnMt"] = "en-mt";
    PublicEmailLanguageEnum["EnMu"] = "en-mu";
    PublicEmailLanguageEnum["EnMv"] = "en-mv";
    PublicEmailLanguageEnum["EnMw"] = "en-mw";
    PublicEmailLanguageEnum["EnMx"] = "en-mx";
    PublicEmailLanguageEnum["EnMy"] = "en-my";
    PublicEmailLanguageEnum["EnNa"] = "en-na";
    PublicEmailLanguageEnum["EnNf"] = "en-nf";
    PublicEmailLanguageEnum["EnNg"] = "en-ng";
    PublicEmailLanguageEnum["EnNl"] = "en-nl";
    PublicEmailLanguageEnum["EnNr"] = "en-nr";
    PublicEmailLanguageEnum["EnNu"] = "en-nu";
    PublicEmailLanguageEnum["EnNz"] = "en-nz";
    PublicEmailLanguageEnum["EnPg"] = "en-pg";
    PublicEmailLanguageEnum["EnPh"] = "en-ph";
    PublicEmailLanguageEnum["EnPk"] = "en-pk";
    PublicEmailLanguageEnum["EnPn"] = "en-pn";
    PublicEmailLanguageEnum["EnPr"] = "en-pr";
    PublicEmailLanguageEnum["EnPw"] = "en-pw";
    PublicEmailLanguageEnum["EnRw"] = "en-rw";
    PublicEmailLanguageEnum["EnSb"] = "en-sb";
    PublicEmailLanguageEnum["EnSc"] = "en-sc";
    PublicEmailLanguageEnum["EnSd"] = "en-sd";
    PublicEmailLanguageEnum["EnSe"] = "en-se";
    PublicEmailLanguageEnum["EnSg"] = "en-sg";
    PublicEmailLanguageEnum["EnSh"] = "en-sh";
    PublicEmailLanguageEnum["EnSi"] = "en-si";
    PublicEmailLanguageEnum["EnSl"] = "en-sl";
    PublicEmailLanguageEnum["EnSs"] = "en-ss";
    PublicEmailLanguageEnum["EnSx"] = "en-sx";
    PublicEmailLanguageEnum["EnSz"] = "en-sz";
    PublicEmailLanguageEnum["EnTc"] = "en-tc";
    PublicEmailLanguageEnum["EnTk"] = "en-tk";
    PublicEmailLanguageEnum["EnTn"] = "en-tn";
    PublicEmailLanguageEnum["EnTo"] = "en-to";
    PublicEmailLanguageEnum["EnTt"] = "en-tt";
    PublicEmailLanguageEnum["EnTv"] = "en-tv";
    PublicEmailLanguageEnum["EnTz"] = "en-tz";
    PublicEmailLanguageEnum["EnUg"] = "en-ug";
    PublicEmailLanguageEnum["EnUm"] = "en-um";
    PublicEmailLanguageEnum["EnUs"] = "en-us";
    PublicEmailLanguageEnum["EnVc"] = "en-vc";
    PublicEmailLanguageEnum["EnVg"] = "en-vg";
    PublicEmailLanguageEnum["EnVi"] = "en-vi";
    PublicEmailLanguageEnum["EnVu"] = "en-vu";
    PublicEmailLanguageEnum["EnWs"] = "en-ws";
    PublicEmailLanguageEnum["EnZa"] = "en-za";
    PublicEmailLanguageEnum["EnZm"] = "en-zm";
    PublicEmailLanguageEnum["EnZw"] = "en-zw";
    PublicEmailLanguageEnum["Eo"] = "eo";
    PublicEmailLanguageEnum["Eo001"] = "eo-001";
    PublicEmailLanguageEnum["Es"] = "es";
    PublicEmailLanguageEnum["Es419"] = "es-419";
    PublicEmailLanguageEnum["EsAr"] = "es-ar";
    PublicEmailLanguageEnum["EsBo"] = "es-bo";
    PublicEmailLanguageEnum["EsBr"] = "es-br";
    PublicEmailLanguageEnum["EsBz"] = "es-bz";
    PublicEmailLanguageEnum["EsCl"] = "es-cl";
    PublicEmailLanguageEnum["EsCo"] = "es-co";
    PublicEmailLanguageEnum["EsCr"] = "es-cr";
    PublicEmailLanguageEnum["EsCu"] = "es-cu";
    PublicEmailLanguageEnum["EsDo"] = "es-do";
    PublicEmailLanguageEnum["EsEa"] = "es-ea";
    PublicEmailLanguageEnum["EsEc"] = "es-ec";
    PublicEmailLanguageEnum["EsEs"] = "es-es";
    PublicEmailLanguageEnum["EsGq"] = "es-gq";
    PublicEmailLanguageEnum["EsGt"] = "es-gt";
    PublicEmailLanguageEnum["EsHn"] = "es-hn";
    PublicEmailLanguageEnum["EsIc"] = "es-ic";
    PublicEmailLanguageEnum["EsMx"] = "es-mx";
    PublicEmailLanguageEnum["EsNi"] = "es-ni";
    PublicEmailLanguageEnum["EsPa"] = "es-pa";
    PublicEmailLanguageEnum["EsPe"] = "es-pe";
    PublicEmailLanguageEnum["EsPh"] = "es-ph";
    PublicEmailLanguageEnum["EsPr"] = "es-pr";
    PublicEmailLanguageEnum["EsPy"] = "es-py";
    PublicEmailLanguageEnum["EsSv"] = "es-sv";
    PublicEmailLanguageEnum["EsUs"] = "es-us";
    PublicEmailLanguageEnum["EsUy"] = "es-uy";
    PublicEmailLanguageEnum["EsVe"] = "es-ve";
    PublicEmailLanguageEnum["Et"] = "et";
    PublicEmailLanguageEnum["EtEe"] = "et-ee";
    PublicEmailLanguageEnum["Eu"] = "eu";
    PublicEmailLanguageEnum["EuEs"] = "eu-es";
    PublicEmailLanguageEnum["Ewo"] = "ewo";
    PublicEmailLanguageEnum["EwoCm"] = "ewo-cm";
    PublicEmailLanguageEnum["Fa"] = "fa";
    PublicEmailLanguageEnum["FaAf"] = "fa-af";
    PublicEmailLanguageEnum["FaIr"] = "fa-ir";
    PublicEmailLanguageEnum["Ff"] = "ff";
    PublicEmailLanguageEnum["FfBf"] = "ff-bf";
    PublicEmailLanguageEnum["FfCm"] = "ff-cm";
    PublicEmailLanguageEnum["FfGh"] = "ff-gh";
    PublicEmailLanguageEnum["FfGm"] = "ff-gm";
    PublicEmailLanguageEnum["FfGn"] = "ff-gn";
    PublicEmailLanguageEnum["FfGw"] = "ff-gw";
    PublicEmailLanguageEnum["FfLr"] = "ff-lr";
    PublicEmailLanguageEnum["FfMr"] = "ff-mr";
    PublicEmailLanguageEnum["FfNe"] = "ff-ne";
    PublicEmailLanguageEnum["FfNg"] = "ff-ng";
    PublicEmailLanguageEnum["FfSl"] = "ff-sl";
    PublicEmailLanguageEnum["FfSn"] = "ff-sn";
    PublicEmailLanguageEnum["Fi"] = "fi";
    PublicEmailLanguageEnum["Fil"] = "fil";
    PublicEmailLanguageEnum["FilPh"] = "fil-ph";
    PublicEmailLanguageEnum["FiFi"] = "fi-fi";
    PublicEmailLanguageEnum["Fo"] = "fo";
    PublicEmailLanguageEnum["FoDk"] = "fo-dk";
    PublicEmailLanguageEnum["FoFo"] = "fo-fo";
    PublicEmailLanguageEnum["Fr"] = "fr";
    PublicEmailLanguageEnum["Frr"] = "frr";
    PublicEmailLanguageEnum["FrrDe"] = "frr-de";
    PublicEmailLanguageEnum["FrBe"] = "fr-be";
    PublicEmailLanguageEnum["FrBf"] = "fr-bf";
    PublicEmailLanguageEnum["FrBi"] = "fr-bi";
    PublicEmailLanguageEnum["FrBj"] = "fr-bj";
    PublicEmailLanguageEnum["FrBl"] = "fr-bl";
    PublicEmailLanguageEnum["FrCa"] = "fr-ca";
    PublicEmailLanguageEnum["FrCd"] = "fr-cd";
    PublicEmailLanguageEnum["FrCf"] = "fr-cf";
    PublicEmailLanguageEnum["FrCg"] = "fr-cg";
    PublicEmailLanguageEnum["FrCh"] = "fr-ch";
    PublicEmailLanguageEnum["FrCi"] = "fr-ci";
    PublicEmailLanguageEnum["FrCm"] = "fr-cm";
    PublicEmailLanguageEnum["FrDj"] = "fr-dj";
    PublicEmailLanguageEnum["FrDz"] = "fr-dz";
    PublicEmailLanguageEnum["FrFr"] = "fr-fr";
    PublicEmailLanguageEnum["FrGa"] = "fr-ga";
    PublicEmailLanguageEnum["FrGf"] = "fr-gf";
    PublicEmailLanguageEnum["FrGn"] = "fr-gn";
    PublicEmailLanguageEnum["FrGp"] = "fr-gp";
    PublicEmailLanguageEnum["FrGq"] = "fr-gq";
    PublicEmailLanguageEnum["FrHt"] = "fr-ht";
    PublicEmailLanguageEnum["FrKm"] = "fr-km";
    PublicEmailLanguageEnum["FrLu"] = "fr-lu";
    PublicEmailLanguageEnum["FrMa"] = "fr-ma";
    PublicEmailLanguageEnum["FrMc"] = "fr-mc";
    PublicEmailLanguageEnum["FrMf"] = "fr-mf";
    PublicEmailLanguageEnum["FrMg"] = "fr-mg";
    PublicEmailLanguageEnum["FrMl"] = "fr-ml";
    PublicEmailLanguageEnum["FrMq"] = "fr-mq";
    PublicEmailLanguageEnum["FrMr"] = "fr-mr";
    PublicEmailLanguageEnum["FrMu"] = "fr-mu";
    PublicEmailLanguageEnum["FrNc"] = "fr-nc";
    PublicEmailLanguageEnum["FrNe"] = "fr-ne";
    PublicEmailLanguageEnum["FrPf"] = "fr-pf";
    PublicEmailLanguageEnum["FrPm"] = "fr-pm";
    PublicEmailLanguageEnum["FrRe"] = "fr-re";
    PublicEmailLanguageEnum["FrRw"] = "fr-rw";
    PublicEmailLanguageEnum["FrSc"] = "fr-sc";
    PublicEmailLanguageEnum["FrSn"] = "fr-sn";
    PublicEmailLanguageEnum["FrSy"] = "fr-sy";
    PublicEmailLanguageEnum["FrTd"] = "fr-td";
    PublicEmailLanguageEnum["FrTg"] = "fr-tg";
    PublicEmailLanguageEnum["FrTn"] = "fr-tn";
    PublicEmailLanguageEnum["FrVu"] = "fr-vu";
    PublicEmailLanguageEnum["FrWf"] = "fr-wf";
    PublicEmailLanguageEnum["FrYt"] = "fr-yt";
    PublicEmailLanguageEnum["Fur"] = "fur";
    PublicEmailLanguageEnum["FurIt"] = "fur-it";
    PublicEmailLanguageEnum["Fy"] = "fy";
    PublicEmailLanguageEnum["FyNl"] = "fy-nl";
    PublicEmailLanguageEnum["Ga"] = "ga";
    PublicEmailLanguageEnum["GaGb"] = "ga-gb";
    PublicEmailLanguageEnum["GaIe"] = "ga-ie";
    PublicEmailLanguageEnum["Gd"] = "gd";
    PublicEmailLanguageEnum["GdGb"] = "gd-gb";
    PublicEmailLanguageEnum["Gl"] = "gl";
    PublicEmailLanguageEnum["GlEs"] = "gl-es";
    PublicEmailLanguageEnum["Gsw"] = "gsw";
    PublicEmailLanguageEnum["GswCh"] = "gsw-ch";
    PublicEmailLanguageEnum["GswFr"] = "gsw-fr";
    PublicEmailLanguageEnum["GswLi"] = "gsw-li";
    PublicEmailLanguageEnum["Gu"] = "gu";
    PublicEmailLanguageEnum["Guz"] = "guz";
    PublicEmailLanguageEnum["GuzKe"] = "guz-ke";
    PublicEmailLanguageEnum["GuIn"] = "gu-in";
    PublicEmailLanguageEnum["Gv"] = "gv";
    PublicEmailLanguageEnum["GvIm"] = "gv-im";
    PublicEmailLanguageEnum["Ha"] = "ha";
    PublicEmailLanguageEnum["Haw"] = "haw";
    PublicEmailLanguageEnum["HawUs"] = "haw-us";
    PublicEmailLanguageEnum["HaGh"] = "ha-gh";
    PublicEmailLanguageEnum["HaNe"] = "ha-ne";
    PublicEmailLanguageEnum["HaNg"] = "ha-ng";
    PublicEmailLanguageEnum["He"] = "he";
    PublicEmailLanguageEnum["HeIl"] = "he-il";
    PublicEmailLanguageEnum["Hi"] = "hi";
    PublicEmailLanguageEnum["HiIn"] = "hi-in";
    PublicEmailLanguageEnum["Hr"] = "hr";
    PublicEmailLanguageEnum["HrBa"] = "hr-ba";
    PublicEmailLanguageEnum["HrHr"] = "hr-hr";
    PublicEmailLanguageEnum["Hsb"] = "hsb";
    PublicEmailLanguageEnum["HsbDe"] = "hsb-de";
    PublicEmailLanguageEnum["Hu"] = "hu";
    PublicEmailLanguageEnum["HuHu"] = "hu-hu";
    PublicEmailLanguageEnum["Hy"] = "hy";
    PublicEmailLanguageEnum["HyAm"] = "hy-am";
    PublicEmailLanguageEnum["Ia"] = "ia";
    PublicEmailLanguageEnum["Ia001"] = "ia-001";
    PublicEmailLanguageEnum["Id"] = "id";
    PublicEmailLanguageEnum["Ig"] = "ig";
    PublicEmailLanguageEnum["IgNg"] = "ig-ng";
    PublicEmailLanguageEnum["Ii"] = "ii";
    PublicEmailLanguageEnum["IiCn"] = "ii-cn";
    PublicEmailLanguageEnum["IdId"] = "id-id";
    PublicEmailLanguageEnum["Is"] = "is";
    PublicEmailLanguageEnum["IsIs"] = "is-is";
    PublicEmailLanguageEnum["It"] = "it";
    PublicEmailLanguageEnum["ItCh"] = "it-ch";
    PublicEmailLanguageEnum["ItIt"] = "it-it";
    PublicEmailLanguageEnum["ItSm"] = "it-sm";
    PublicEmailLanguageEnum["ItVa"] = "it-va";
    PublicEmailLanguageEnum["Ja"] = "ja";
    PublicEmailLanguageEnum["JaJp"] = "ja-jp";
    PublicEmailLanguageEnum["Jgo"] = "jgo";
    PublicEmailLanguageEnum["JgoCm"] = "jgo-cm";
    PublicEmailLanguageEnum["Yi"] = "yi";
    PublicEmailLanguageEnum["Yi001"] = "yi-001";
    PublicEmailLanguageEnum["Jmc"] = "jmc";
    PublicEmailLanguageEnum["JmcTz"] = "jmc-tz";
    PublicEmailLanguageEnum["Jv"] = "jv";
    PublicEmailLanguageEnum["JvId"] = "jv-id";
    PublicEmailLanguageEnum["Ka"] = "ka";
    PublicEmailLanguageEnum["Kab"] = "kab";
    PublicEmailLanguageEnum["KabDz"] = "kab-dz";
    PublicEmailLanguageEnum["Kam"] = "kam";
    PublicEmailLanguageEnum["KamKe"] = "kam-ke";
    PublicEmailLanguageEnum["KaGe"] = "ka-ge";
    PublicEmailLanguageEnum["Kde"] = "kde";
    PublicEmailLanguageEnum["KdeTz"] = "kde-tz";
    PublicEmailLanguageEnum["Kea"] = "kea";
    PublicEmailLanguageEnum["KeaCv"] = "kea-cv";
    PublicEmailLanguageEnum["Kgp"] = "kgp";
    PublicEmailLanguageEnum["KgpBr"] = "kgp-br";
    PublicEmailLanguageEnum["Khq"] = "khq";
    PublicEmailLanguageEnum["KhqMl"] = "khq-ml";
    PublicEmailLanguageEnum["Ki"] = "ki";
    PublicEmailLanguageEnum["KiKe"] = "ki-ke";
    PublicEmailLanguageEnum["Kk"] = "kk";
    PublicEmailLanguageEnum["Kkj"] = "kkj";
    PublicEmailLanguageEnum["KkjCm"] = "kkj-cm";
    PublicEmailLanguageEnum["KkKz"] = "kk-kz";
    PublicEmailLanguageEnum["Kl"] = "kl";
    PublicEmailLanguageEnum["Kln"] = "kln";
    PublicEmailLanguageEnum["KlnKe"] = "kln-ke";
    PublicEmailLanguageEnum["KlGl"] = "kl-gl";
    PublicEmailLanguageEnum["Km"] = "km";
    PublicEmailLanguageEnum["KmKh"] = "km-kh";
    PublicEmailLanguageEnum["Kn"] = "kn";
    PublicEmailLanguageEnum["KnIn"] = "kn-in";
    PublicEmailLanguageEnum["Ko"] = "ko";
    PublicEmailLanguageEnum["Kok"] = "kok";
    PublicEmailLanguageEnum["KokIn"] = "kok-in";
    PublicEmailLanguageEnum["KoKp"] = "ko-kp";
    PublicEmailLanguageEnum["KoKr"] = "ko-kr";
    PublicEmailLanguageEnum["Ks"] = "ks";
    PublicEmailLanguageEnum["Ksb"] = "ksb";
    PublicEmailLanguageEnum["KsbTz"] = "ksb-tz";
    PublicEmailLanguageEnum["Ksf"] = "ksf";
    PublicEmailLanguageEnum["KsfCm"] = "ksf-cm";
    PublicEmailLanguageEnum["Ksh"] = "ksh";
    PublicEmailLanguageEnum["KshDe"] = "ksh-de";
    PublicEmailLanguageEnum["KsIn"] = "ks-in";
    PublicEmailLanguageEnum["Ku"] = "ku";
    PublicEmailLanguageEnum["KuTr"] = "ku-tr";
    PublicEmailLanguageEnum["Kw"] = "kw";
    PublicEmailLanguageEnum["KwGb"] = "kw-gb";
    PublicEmailLanguageEnum["Ky"] = "ky";
    PublicEmailLanguageEnum["KyKg"] = "ky-kg";
    PublicEmailLanguageEnum["Lag"] = "lag";
    PublicEmailLanguageEnum["LagTz"] = "lag-tz";
    PublicEmailLanguageEnum["Lb"] = "lb";
    PublicEmailLanguageEnum["LbLu"] = "lb-lu";
    PublicEmailLanguageEnum["Lg"] = "lg";
    PublicEmailLanguageEnum["LgUg"] = "lg-ug";
    PublicEmailLanguageEnum["Lkt"] = "lkt";
    PublicEmailLanguageEnum["LktUs"] = "lkt-us";
    PublicEmailLanguageEnum["Ln"] = "ln";
    PublicEmailLanguageEnum["LnAo"] = "ln-ao";
    PublicEmailLanguageEnum["LnCd"] = "ln-cd";
    PublicEmailLanguageEnum["LnCf"] = "ln-cf";
    PublicEmailLanguageEnum["LnCg"] = "ln-cg";
    PublicEmailLanguageEnum["Lo"] = "lo";
    PublicEmailLanguageEnum["LoLa"] = "lo-la";
    PublicEmailLanguageEnum["Lrc"] = "lrc";
    PublicEmailLanguageEnum["LrcIq"] = "lrc-iq";
    PublicEmailLanguageEnum["LrcIr"] = "lrc-ir";
    PublicEmailLanguageEnum["Lt"] = "lt";
    PublicEmailLanguageEnum["LtLt"] = "lt-lt";
    PublicEmailLanguageEnum["Lu"] = "lu";
    PublicEmailLanguageEnum["Luo"] = "luo";
    PublicEmailLanguageEnum["LuoKe"] = "luo-ke";
    PublicEmailLanguageEnum["Luy"] = "luy";
    PublicEmailLanguageEnum["LuyKe"] = "luy-ke";
    PublicEmailLanguageEnum["LuCd"] = "lu-cd";
    PublicEmailLanguageEnum["Lv"] = "lv";
    PublicEmailLanguageEnum["LvLv"] = "lv-lv";
    PublicEmailLanguageEnum["Mai"] = "mai";
    PublicEmailLanguageEnum["MaiIn"] = "mai-in";
    PublicEmailLanguageEnum["Mas"] = "mas";
    PublicEmailLanguageEnum["MasKe"] = "mas-ke";
    PublicEmailLanguageEnum["MasTz"] = "mas-tz";
    PublicEmailLanguageEnum["Mdf"] = "mdf";
    PublicEmailLanguageEnum["MdfRu"] = "mdf-ru";
    PublicEmailLanguageEnum["Mer"] = "mer";
    PublicEmailLanguageEnum["MerKe"] = "mer-ke";
    PublicEmailLanguageEnum["Mfe"] = "mfe";
    PublicEmailLanguageEnum["MfeMu"] = "mfe-mu";
    PublicEmailLanguageEnum["Mg"] = "mg";
    PublicEmailLanguageEnum["Mgh"] = "mgh";
    PublicEmailLanguageEnum["MghMz"] = "mgh-mz";
    PublicEmailLanguageEnum["Mgo"] = "mgo";
    PublicEmailLanguageEnum["MgoCm"] = "mgo-cm";
    PublicEmailLanguageEnum["MgMg"] = "mg-mg";
    PublicEmailLanguageEnum["Mi"] = "mi";
    PublicEmailLanguageEnum["MiNz"] = "mi-nz";
    PublicEmailLanguageEnum["Mk"] = "mk";
    PublicEmailLanguageEnum["MkMk"] = "mk-mk";
    PublicEmailLanguageEnum["Ml"] = "ml";
    PublicEmailLanguageEnum["MlIn"] = "ml-in";
    PublicEmailLanguageEnum["Mn"] = "mn";
    PublicEmailLanguageEnum["Mni"] = "mni";
    PublicEmailLanguageEnum["MniIn"] = "mni-in";
    PublicEmailLanguageEnum["MnMn"] = "mn-mn";
    PublicEmailLanguageEnum["Mr"] = "mr";
    PublicEmailLanguageEnum["MrIn"] = "mr-in";
    PublicEmailLanguageEnum["Ms"] = "ms";
    PublicEmailLanguageEnum["MsBn"] = "ms-bn";
    PublicEmailLanguageEnum["MsId"] = "ms-id";
    PublicEmailLanguageEnum["MsMy"] = "ms-my";
    PublicEmailLanguageEnum["MsSg"] = "ms-sg";
    PublicEmailLanguageEnum["Mt"] = "mt";
    PublicEmailLanguageEnum["MtMt"] = "mt-mt";
    PublicEmailLanguageEnum["Mua"] = "mua";
    PublicEmailLanguageEnum["MuaCm"] = "mua-cm";
    PublicEmailLanguageEnum["My"] = "my";
    PublicEmailLanguageEnum["MyMm"] = "my-mm";
    PublicEmailLanguageEnum["Mzn"] = "mzn";
    PublicEmailLanguageEnum["MznIr"] = "mzn-ir";
    PublicEmailLanguageEnum["Naq"] = "naq";
    PublicEmailLanguageEnum["NaqNa"] = "naq-na";
    PublicEmailLanguageEnum["Nb"] = "nb";
    PublicEmailLanguageEnum["NbNo"] = "nb-no";
    PublicEmailLanguageEnum["NbSj"] = "nb-sj";
    PublicEmailLanguageEnum["Nd"] = "nd";
    PublicEmailLanguageEnum["Nds"] = "nds";
    PublicEmailLanguageEnum["NdsDe"] = "nds-de";
    PublicEmailLanguageEnum["NdsNl"] = "nds-nl";
    PublicEmailLanguageEnum["NdZw"] = "nd-zw";
    PublicEmailLanguageEnum["Ne"] = "ne";
    PublicEmailLanguageEnum["NeIn"] = "ne-in";
    PublicEmailLanguageEnum["NeNp"] = "ne-np";
    PublicEmailLanguageEnum["Nl"] = "nl";
    PublicEmailLanguageEnum["NlAw"] = "nl-aw";
    PublicEmailLanguageEnum["NlBe"] = "nl-be";
    PublicEmailLanguageEnum["NlBq"] = "nl-bq";
    PublicEmailLanguageEnum["NlCh"] = "nl-ch";
    PublicEmailLanguageEnum["NlCw"] = "nl-cw";
    PublicEmailLanguageEnum["NlLu"] = "nl-lu";
    PublicEmailLanguageEnum["NlNl"] = "nl-nl";
    PublicEmailLanguageEnum["NlSr"] = "nl-sr";
    PublicEmailLanguageEnum["NlSx"] = "nl-sx";
    PublicEmailLanguageEnum["Nmg"] = "nmg";
    PublicEmailLanguageEnum["NmgCm"] = "nmg-cm";
    PublicEmailLanguageEnum["Nn"] = "nn";
    PublicEmailLanguageEnum["Nnh"] = "nnh";
    PublicEmailLanguageEnum["NnhCm"] = "nnh-cm";
    PublicEmailLanguageEnum["NnNo"] = "nn-no";
    PublicEmailLanguageEnum["No"] = "no";
    PublicEmailLanguageEnum["NoNo"] = "no-no";
    PublicEmailLanguageEnum["Nus"] = "nus";
    PublicEmailLanguageEnum["NusSs"] = "nus-ss";
    PublicEmailLanguageEnum["Nyn"] = "nyn";
    PublicEmailLanguageEnum["NynUg"] = "nyn-ug";
    PublicEmailLanguageEnum["Oc"] = "oc";
    PublicEmailLanguageEnum["OcEs"] = "oc-es";
    PublicEmailLanguageEnum["OcFr"] = "oc-fr";
    PublicEmailLanguageEnum["Om"] = "om";
    PublicEmailLanguageEnum["OmEt"] = "om-et";
    PublicEmailLanguageEnum["OmKe"] = "om-ke";
    PublicEmailLanguageEnum["Or"] = "or";
    PublicEmailLanguageEnum["OrIn"] = "or-in";
    PublicEmailLanguageEnum["Os"] = "os";
    PublicEmailLanguageEnum["OsGe"] = "os-ge";
    PublicEmailLanguageEnum["OsRu"] = "os-ru";
    PublicEmailLanguageEnum["Pa"] = "pa";
    PublicEmailLanguageEnum["PaIn"] = "pa-in";
    PublicEmailLanguageEnum["PaPk"] = "pa-pk";
    PublicEmailLanguageEnum["Pcm"] = "pcm";
    PublicEmailLanguageEnum["PcmNg"] = "pcm-ng";
    PublicEmailLanguageEnum["Pis"] = "pis";
    PublicEmailLanguageEnum["PisSb"] = "pis-sb";
    PublicEmailLanguageEnum["Pl"] = "pl";
    PublicEmailLanguageEnum["PlPl"] = "pl-pl";
    PublicEmailLanguageEnum["Prg"] = "prg";
    PublicEmailLanguageEnum["Prg001"] = "prg-001";
    PublicEmailLanguageEnum["Ps"] = "ps";
    PublicEmailLanguageEnum["PsAf"] = "ps-af";
    PublicEmailLanguageEnum["PsPk"] = "ps-pk";
    PublicEmailLanguageEnum["Pt"] = "pt";
    PublicEmailLanguageEnum["PtAo"] = "pt-ao";
    PublicEmailLanguageEnum["PtBr"] = "pt-br";
    PublicEmailLanguageEnum["PtCh"] = "pt-ch";
    PublicEmailLanguageEnum["PtCv"] = "pt-cv";
    PublicEmailLanguageEnum["PtGq"] = "pt-gq";
    PublicEmailLanguageEnum["PtGw"] = "pt-gw";
    PublicEmailLanguageEnum["PtLu"] = "pt-lu";
    PublicEmailLanguageEnum["PtMo"] = "pt-mo";
    PublicEmailLanguageEnum["PtMz"] = "pt-mz";
    PublicEmailLanguageEnum["PtPt"] = "pt-pt";
    PublicEmailLanguageEnum["PtSt"] = "pt-st";
    PublicEmailLanguageEnum["PtTl"] = "pt-tl";
    PublicEmailLanguageEnum["Qu"] = "qu";
    PublicEmailLanguageEnum["QuBo"] = "qu-bo";
    PublicEmailLanguageEnum["QuEc"] = "qu-ec";
    PublicEmailLanguageEnum["QuPe"] = "qu-pe";
    PublicEmailLanguageEnum["Raj"] = "raj";
    PublicEmailLanguageEnum["RajIn"] = "raj-in";
    PublicEmailLanguageEnum["Rm"] = "rm";
    PublicEmailLanguageEnum["RmCh"] = "rm-ch";
    PublicEmailLanguageEnum["Rn"] = "rn";
    PublicEmailLanguageEnum["RnBi"] = "rn-bi";
    PublicEmailLanguageEnum["Ro"] = "ro";
    PublicEmailLanguageEnum["Rof"] = "rof";
    PublicEmailLanguageEnum["RofTz"] = "rof-tz";
    PublicEmailLanguageEnum["RoMd"] = "ro-md";
    PublicEmailLanguageEnum["RoRo"] = "ro-ro";
    PublicEmailLanguageEnum["Ru"] = "ru";
    PublicEmailLanguageEnum["RuBy"] = "ru-by";
    PublicEmailLanguageEnum["RuKg"] = "ru-kg";
    PublicEmailLanguageEnum["RuKz"] = "ru-kz";
    PublicEmailLanguageEnum["RuMd"] = "ru-md";
    PublicEmailLanguageEnum["RuRu"] = "ru-ru";
    PublicEmailLanguageEnum["RuUa"] = "ru-ua";
    PublicEmailLanguageEnum["Rw"] = "rw";
    PublicEmailLanguageEnum["Rwk"] = "rwk";
    PublicEmailLanguageEnum["RwkTz"] = "rwk-tz";
    PublicEmailLanguageEnum["RwRw"] = "rw-rw";
    PublicEmailLanguageEnum["Sa"] = "sa";
    PublicEmailLanguageEnum["Sah"] = "sah";
    PublicEmailLanguageEnum["SahRu"] = "sah-ru";
    PublicEmailLanguageEnum["Saq"] = "saq";
    PublicEmailLanguageEnum["SaqKe"] = "saq-ke";
    PublicEmailLanguageEnum["Sat"] = "sat";
    PublicEmailLanguageEnum["SatIn"] = "sat-in";
    PublicEmailLanguageEnum["SaIn"] = "sa-in";
    PublicEmailLanguageEnum["Sbp"] = "sbp";
    PublicEmailLanguageEnum["SbpTz"] = "sbp-tz";
    PublicEmailLanguageEnum["Sc"] = "sc";
    PublicEmailLanguageEnum["ScIt"] = "sc-it";
    PublicEmailLanguageEnum["Sd"] = "sd";
    PublicEmailLanguageEnum["SdIn"] = "sd-in";
    PublicEmailLanguageEnum["SdPk"] = "sd-pk";
    PublicEmailLanguageEnum["Se"] = "se";
    PublicEmailLanguageEnum["Seh"] = "seh";
    PublicEmailLanguageEnum["SehMz"] = "seh-mz";
    PublicEmailLanguageEnum["Ses"] = "ses";
    PublicEmailLanguageEnum["SesMl"] = "ses-ml";
    PublicEmailLanguageEnum["SeFi"] = "se-fi";
    PublicEmailLanguageEnum["SeNo"] = "se-no";
    PublicEmailLanguageEnum["SeSe"] = "se-se";
    PublicEmailLanguageEnum["Sg"] = "sg";
    PublicEmailLanguageEnum["SgCf"] = "sg-cf";
    PublicEmailLanguageEnum["Shi"] = "shi";
    PublicEmailLanguageEnum["ShiMa"] = "shi-ma";
    PublicEmailLanguageEnum["Si"] = "si";
    PublicEmailLanguageEnum["SiLk"] = "si-lk";
    PublicEmailLanguageEnum["Sk"] = "sk";
    PublicEmailLanguageEnum["SkSk"] = "sk-sk";
    PublicEmailLanguageEnum["Sl"] = "sl";
    PublicEmailLanguageEnum["SlSi"] = "sl-si";
    PublicEmailLanguageEnum["Smn"] = "smn";
    PublicEmailLanguageEnum["SmnFi"] = "smn-fi";
    PublicEmailLanguageEnum["Sms"] = "sms";
    PublicEmailLanguageEnum["SmsFi"] = "sms-fi";
    PublicEmailLanguageEnum["Sn"] = "sn";
    PublicEmailLanguageEnum["SnZw"] = "sn-zw";
    PublicEmailLanguageEnum["So"] = "so";
    PublicEmailLanguageEnum["SoDj"] = "so-dj";
    PublicEmailLanguageEnum["SoEt"] = "so-et";
    PublicEmailLanguageEnum["SoKe"] = "so-ke";
    PublicEmailLanguageEnum["SoSo"] = "so-so";
    PublicEmailLanguageEnum["Sq"] = "sq";
    PublicEmailLanguageEnum["SqAl"] = "sq-al";
    PublicEmailLanguageEnum["SqMk"] = "sq-mk";
    PublicEmailLanguageEnum["SqXk"] = "sq-xk";
    PublicEmailLanguageEnum["Sr"] = "sr";
    PublicEmailLanguageEnum["SrBa"] = "sr-ba";
    PublicEmailLanguageEnum["SrCs"] = "sr-cs";
    PublicEmailLanguageEnum["SrMe"] = "sr-me";
    PublicEmailLanguageEnum["SrRs"] = "sr-rs";
    PublicEmailLanguageEnum["SrXk"] = "sr-xk";
    PublicEmailLanguageEnum["Su"] = "su";
    PublicEmailLanguageEnum["SuId"] = "su-id";
    PublicEmailLanguageEnum["Sv"] = "sv";
    PublicEmailLanguageEnum["SvAx"] = "sv-ax";
    PublicEmailLanguageEnum["SvFi"] = "sv-fi";
    PublicEmailLanguageEnum["SvSe"] = "sv-se";
    PublicEmailLanguageEnum["Sw"] = "sw";
    PublicEmailLanguageEnum["SwCd"] = "sw-cd";
    PublicEmailLanguageEnum["SwKe"] = "sw-ke";
    PublicEmailLanguageEnum["SwTz"] = "sw-tz";
    PublicEmailLanguageEnum["SwUg"] = "sw-ug";
    PublicEmailLanguageEnum["Sy"] = "sy";
    PublicEmailLanguageEnum["Ta"] = "ta";
    PublicEmailLanguageEnum["TaIn"] = "ta-in";
    PublicEmailLanguageEnum["TaLk"] = "ta-lk";
    PublicEmailLanguageEnum["TaMy"] = "ta-my";
    PublicEmailLanguageEnum["TaSg"] = "ta-sg";
    PublicEmailLanguageEnum["Te"] = "te";
    PublicEmailLanguageEnum["Teo"] = "teo";
    PublicEmailLanguageEnum["TeoKe"] = "teo-ke";
    PublicEmailLanguageEnum["TeoUg"] = "teo-ug";
    PublicEmailLanguageEnum["TeIn"] = "te-in";
    PublicEmailLanguageEnum["Tg"] = "tg";
    PublicEmailLanguageEnum["TgTj"] = "tg-tj";
    PublicEmailLanguageEnum["Th"] = "th";
    PublicEmailLanguageEnum["ThTh"] = "th-th";
    PublicEmailLanguageEnum["Ti"] = "ti";
    PublicEmailLanguageEnum["TiEr"] = "ti-er";
    PublicEmailLanguageEnum["TiEt"] = "ti-et";
    PublicEmailLanguageEnum["Tk"] = "tk";
    PublicEmailLanguageEnum["TkTm"] = "tk-tm";
    PublicEmailLanguageEnum["Tl"] = "tl";
    PublicEmailLanguageEnum["To"] = "to";
    PublicEmailLanguageEnum["Tok"] = "tok";
    PublicEmailLanguageEnum["Tok001"] = "tok-001";
    PublicEmailLanguageEnum["ToTo"] = "to-to";
    PublicEmailLanguageEnum["Tr"] = "tr";
    PublicEmailLanguageEnum["TrCy"] = "tr-cy";
    PublicEmailLanguageEnum["TrTr"] = "tr-tr";
    PublicEmailLanguageEnum["Tt"] = "tt";
    PublicEmailLanguageEnum["TtRu"] = "tt-ru";
    PublicEmailLanguageEnum["Twq"] = "twq";
    PublicEmailLanguageEnum["TwqNe"] = "twq-ne";
    PublicEmailLanguageEnum["Tzm"] = "tzm";
    PublicEmailLanguageEnum["TzmMa"] = "tzm-ma";
    PublicEmailLanguageEnum["Ug"] = "ug";
    PublicEmailLanguageEnum["UgCn"] = "ug-cn";
    PublicEmailLanguageEnum["Uk"] = "uk";
    PublicEmailLanguageEnum["UkUa"] = "uk-ua";
    PublicEmailLanguageEnum["Ur"] = "ur";
    PublicEmailLanguageEnum["UrIn"] = "ur-in";
    PublicEmailLanguageEnum["UrPk"] = "ur-pk";
    PublicEmailLanguageEnum["Uz"] = "uz";
    PublicEmailLanguageEnum["UzAf"] = "uz-af";
    PublicEmailLanguageEnum["UzUz"] = "uz-uz";
    PublicEmailLanguageEnum["Vai"] = "vai";
    PublicEmailLanguageEnum["VaiLr"] = "vai-lr";
    PublicEmailLanguageEnum["Vi"] = "vi";
    PublicEmailLanguageEnum["ViVn"] = "vi-vn";
    PublicEmailLanguageEnum["Vo"] = "vo";
    PublicEmailLanguageEnum["Vo001"] = "vo-001";
    PublicEmailLanguageEnum["Vun"] = "vun";
    PublicEmailLanguageEnum["VunTz"] = "vun-tz";
    PublicEmailLanguageEnum["Wae"] = "wae";
    PublicEmailLanguageEnum["WaeCh"] = "wae-ch";
    PublicEmailLanguageEnum["Wo"] = "wo";
    PublicEmailLanguageEnum["WoSn"] = "wo-sn";
    PublicEmailLanguageEnum["Xh"] = "xh";
    PublicEmailLanguageEnum["XhZa"] = "xh-za";
    PublicEmailLanguageEnum["Xog"] = "xog";
    PublicEmailLanguageEnum["XogUg"] = "xog-ug";
    PublicEmailLanguageEnum["Yav"] = "yav";
    PublicEmailLanguageEnum["YavCm"] = "yav-cm";
    PublicEmailLanguageEnum["Yo"] = "yo";
    PublicEmailLanguageEnum["YoBj"] = "yo-bj";
    PublicEmailLanguageEnum["YoNg"] = "yo-ng";
    PublicEmailLanguageEnum["Yrl"] = "yrl";
    PublicEmailLanguageEnum["YrlBr"] = "yrl-br";
    PublicEmailLanguageEnum["YrlCo"] = "yrl-co";
    PublicEmailLanguageEnum["YrlVe"] = "yrl-ve";
    PublicEmailLanguageEnum["Yue"] = "yue";
    PublicEmailLanguageEnum["YueCn"] = "yue-cn";
    PublicEmailLanguageEnum["YueHk"] = "yue-hk";
    PublicEmailLanguageEnum["Zgh"] = "zgh";
    PublicEmailLanguageEnum["ZghMa"] = "zgh-ma";
    PublicEmailLanguageEnum["Zh"] = "zh";
    PublicEmailLanguageEnum["ZhCn"] = "zh-cn";
    PublicEmailLanguageEnum["ZhHans"] = "zh-hans";
    PublicEmailLanguageEnum["ZhHant"] = "zh-hant";
    PublicEmailLanguageEnum["ZhHk"] = "zh-hk";
    PublicEmailLanguageEnum["ZhMo"] = "zh-mo";
    PublicEmailLanguageEnum["ZhSg"] = "zh-sg";
    PublicEmailLanguageEnum["ZhTw"] = "zh-tw";
    PublicEmailLanguageEnum["Zu"] = "zu";
    PublicEmailLanguageEnum["ZuZa"] = "zu-za";
})(PublicEmailLanguageEnum = exports.PublicEmailLanguageEnum || (exports.PublicEmailLanguageEnum = {}));
var PublicEmailTypeEnum;
(function (PublicEmailTypeEnum) {
    PublicEmailTypeEnum["AbEmail"] = "AB_EMAIL";
    PublicEmailTypeEnum["BatchEmail"] = "BATCH_EMAIL";
    PublicEmailTypeEnum["LocaltimeEmail"] = "LOCALTIME_EMAIL";
    PublicEmailTypeEnum["AutomatedAbEmail"] = "AUTOMATED_AB_EMAIL";
    PublicEmailTypeEnum["BlogEmail"] = "BLOG_EMAIL";
    PublicEmailTypeEnum["BlogEmailChild"] = "BLOG_EMAIL_CHILD";
    PublicEmailTypeEnum["RssEmail"] = "RSS_EMAIL";
    PublicEmailTypeEnum["RssEmailChild"] = "RSS_EMAIL_CHILD";
    PublicEmailTypeEnum["ResubscribeEmail"] = "RESUBSCRIBE_EMAIL";
    PublicEmailTypeEnum["OptinEmail"] = "OPTIN_EMAIL";
    PublicEmailTypeEnum["OptinFollowupEmail"] = "OPTIN_FOLLOWUP_EMAIL";
    PublicEmailTypeEnum["AutomatedEmail"] = "AUTOMATED_EMAIL";
    PublicEmailTypeEnum["FeedbackCesEmail"] = "FEEDBACK_CES_EMAIL";
    PublicEmailTypeEnum["FeedbackCustomEmail"] = "FEEDBACK_CUSTOM_EMAIL";
    PublicEmailTypeEnum["FeedbackCustomSurveyEmail"] = "FEEDBACK_CUSTOM_SURVEY_EMAIL";
    PublicEmailTypeEnum["FeedbackNpsEmail"] = "FEEDBACK_NPS_EMAIL";
    PublicEmailTypeEnum["FollowupEmail"] = "FOLLOWUP_EMAIL";
    PublicEmailTypeEnum["LeadflowEmail"] = "LEADFLOW_EMAIL";
    PublicEmailTypeEnum["SingleSendApi"] = "SINGLE_SEND_API";
    PublicEmailTypeEnum["MarketingSingleSendApi"] = "MARKETING_SINGLE_SEND_API";
    PublicEmailTypeEnum["SmtpToken"] = "SMTP_TOKEN";
    PublicEmailTypeEnum["TicketEmail"] = "TICKET_EMAIL";
    PublicEmailTypeEnum["MembershipRegistrationEmail"] = "MEMBERSHIP_REGISTRATION_EMAIL";
    PublicEmailTypeEnum["MembershipPasswordSavedEmail"] = "MEMBERSHIP_PASSWORD_SAVED_EMAIL";
    PublicEmailTypeEnum["MembershipPasswordResetEmail"] = "MEMBERSHIP_PASSWORD_RESET_EMAIL";
    PublicEmailTypeEnum["MembershipEmailVerificationEmail"] = "MEMBERSHIP_EMAIL_VERIFICATION_EMAIL";
    PublicEmailTypeEnum["MembershipPasswordlessAuthEmail"] = "MEMBERSHIP_PASSWORDLESS_AUTH_EMAIL";
    PublicEmailTypeEnum["MembershipRegistrationFollowUpEmail"] = "MEMBERSHIP_REGISTRATION_FOLLOW_UP_EMAIL";
    PublicEmailTypeEnum["MembershipOtpLoginEmail"] = "MEMBERSHIP_OTP_LOGIN_EMAIL";
    PublicEmailTypeEnum["MembershipFollowUpEmail"] = "MEMBERSHIP_FOLLOW_UP_EMAIL";
    PublicEmailTypeEnum["MembershipVerificationEmail"] = "MEMBERSHIP_VERIFICATION_EMAIL";
})(PublicEmailTypeEnum = exports.PublicEmailTypeEnum || (exports.PublicEmailTypeEnum = {}));
var PublicEmailStateEnum;
(function (PublicEmailStateEnum) {
    PublicEmailStateEnum["Automated"] = "AUTOMATED";
    PublicEmailStateEnum["AutomatedDraft"] = "AUTOMATED_DRAFT";
    PublicEmailStateEnum["AutomatedSending"] = "AUTOMATED_SENDING";
    PublicEmailStateEnum["AutomatedForForm"] = "AUTOMATED_FOR_FORM";
    PublicEmailStateEnum["AutomatedForFormBuffer"] = "AUTOMATED_FOR_FORM_BUFFER";
    PublicEmailStateEnum["AutomatedForFormDraft"] = "AUTOMATED_FOR_FORM_DRAFT";
    PublicEmailStateEnum["AutomatedForFormLegacy"] = "AUTOMATED_FOR_FORM_LEGACY";
    PublicEmailStateEnum["BlogEmailDraft"] = "BLOG_EMAIL_DRAFT";
    PublicEmailStateEnum["BlogEmailPublished"] = "BLOG_EMAIL_PUBLISHED";
    PublicEmailStateEnum["Draft"] = "DRAFT";
    PublicEmailStateEnum["DraftAb"] = "DRAFT_AB";
    PublicEmailStateEnum["DraftAbVariant"] = "DRAFT_AB_VARIANT";
    PublicEmailStateEnum["Error"] = "ERROR";
    PublicEmailStateEnum["LoserAbVariant"] = "LOSER_AB_VARIANT";
    PublicEmailStateEnum["PageStub"] = "PAGE_STUB";
    PublicEmailStateEnum["PreProcessing"] = "PRE_PROCESSING";
    PublicEmailStateEnum["Processing"] = "PROCESSING";
    PublicEmailStateEnum["Published"] = "PUBLISHED";
    PublicEmailStateEnum["PublishedAb"] = "PUBLISHED_AB";
    PublicEmailStateEnum["PublishedAbVariant"] = "PUBLISHED_AB_VARIANT";
    PublicEmailStateEnum["PublishedOrScheduled"] = "PUBLISHED_OR_SCHEDULED";
    PublicEmailStateEnum["RssToEmailDraft"] = "RSS_TO_EMAIL_DRAFT";
    PublicEmailStateEnum["RssToEmailPublished"] = "RSS_TO_EMAIL_PUBLISHED";
    PublicEmailStateEnum["Scheduled"] = "SCHEDULED";
    PublicEmailStateEnum["ScheduledAb"] = "SCHEDULED_AB";
    PublicEmailStateEnum["ScheduledOrPublished"] = "SCHEDULED_OR_PUBLISHED";
    PublicEmailStateEnum["AutomatedAb"] = "AUTOMATED_AB";
    PublicEmailStateEnum["AutomatedAbVariant"] = "AUTOMATED_AB_VARIANT";
    PublicEmailStateEnum["AutomatedDraftAb"] = "AUTOMATED_DRAFT_AB";
    PublicEmailStateEnum["AutomatedDraftAbvariant"] = "AUTOMATED_DRAFT_ABVARIANT";
    PublicEmailStateEnum["AutomatedLoserAbvariant"] = "AUTOMATED_LOSER_ABVARIANT";
})(PublicEmailStateEnum = exports.PublicEmailStateEnum || (exports.PublicEmailStateEnum = {}));
