// @hubspot/api-client/lib/codegen/marketing/emails/models/PublicEmailTestingDetails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicEmailTestingDetailsAbSuccessMetricEnum = exports.PublicEmailTestingDetailsAbStatusEnum = exports.PublicEmailTestingDetailsAbSampleSizeDefaultEnum = exports.PublicEmailTestingDetailsAbSamplingDefaultEnum = exports.PublicEmailTestingDetails = void 0;
class PublicEmailTestingDetails {
    static getAttributeTypeMap() {
        return PublicEmailTestingDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicEmailTestingDetails = PublicEmailTestingDetails;
PublicEmailTestingDetails.discriminator = undefined;
PublicEmailTestingDetails.mapping = undefined;
PublicEmailTestingDetails.attributeTypeMap = [
    {
        "name": "abSamplingDefault",
        "baseName": "abSamplingDefault",
        "type": "PublicEmailTestingDetailsAbSamplingDefaultEnum",
        "format": ""
    },
    {
        "name": "abSampleSizeDefault",
        "baseName": "abSampleSizeDefault",
        "type": "PublicEmailTestingDetailsAbSampleSizeDefaultEnum",
        "format": ""
    },
    {
        "name": "abStatus",
        "baseName": "abStatus",
        "type": "PublicEmailTestingDetailsAbStatusEnum",
        "format": ""
    },
    {
        "name": "abTestPercentage",
        "baseName": "abTestPercentage",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "hoursToWait",
        "baseName": "hoursToWait",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "testId",
        "baseName": "testId",
        "type": "string",
        "format": ""
    },
    {
        "name": "abSuccessMetric",
        "baseName": "abSuccessMetric",
        "type": "PublicEmailTestingDetailsAbSuccessMetricEnum",
        "format": ""
    }
];
var PublicEmailTestingDetailsAbSamplingDefaultEnum;
(function (PublicEmailTestingDetailsAbSamplingDefaultEnum) {
    PublicEmailTestingDetailsAbSamplingDefaultEnum["Master"] = "master";
    PublicEmailTestingDetailsAbSamplingDefaultEnum["Variant"] = "variant";
    PublicEmailTestingDetailsAbSamplingDefaultEnum["LoserVariant"] = "loser_variant";
    PublicEmailTestingDetailsAbSamplingDefaultEnum["MabMaster"] = "mab_master";
    PublicEmailTestingDetailsAbSamplingDefaultEnum["MabVariant"] = "mab_variant";
    PublicEmailTestingDetailsAbSamplingDefaultEnum["AutomatedMaster"] = "automated_master";
    PublicEmailTestingDetailsAbSamplingDefaultEnum["AutomatedVariant"] = "automated_variant";
    PublicEmailTestingDetailsAbSamplingDefaultEnum["AutomatedLoserVariant"] = "automated_loser_variant";
})(PublicEmailTestingDetailsAbSamplingDefaultEnum = exports.PublicEmailTestingDetailsAbSamplingDefaultEnum || (exports.PublicEmailTestingDetailsAbSamplingDefaultEnum = {}));
var PublicEmailTestingDetailsAbSampleSizeDefaultEnum;
(function (PublicEmailTestingDetailsAbSampleSizeDefaultEnum) {
    PublicEmailTestingDetailsAbSampleSizeDefaultEnum["Master"] = "master";
    PublicEmailTestingDetailsAbSampleSizeDefaultEnum["Variant"] = "variant";
    PublicEmailTestingDetailsAbSampleSizeDefaultEnum["LoserVariant"] = "loser_variant";
    PublicEmailTestingDetailsAbSampleSizeDefaultEnum["MabMaster"] = "mab_master";
    PublicEmailTestingDetailsAbSampleSizeDefaultEnum["MabVariant"] = "mab_variant";
    PublicEmailTestingDetailsAbSampleSizeDefaultEnum["AutomatedMaster"] = "automated_master";
    PublicEmailTestingDetailsAbSampleSizeDefaultEnum["AutomatedVariant"] = "automated_variant";
    PublicEmailTestingDetailsAbSampleSizeDefaultEnum["AutomatedLoserVariant"] = "automated_loser_variant";
})(PublicEmailTestingDetailsAbSampleSizeDefaultEnum = exports.PublicEmailTestingDetailsAbSampleSizeDefaultEnum || (exports.PublicEmailTestingDetailsAbSampleSizeDefaultEnum = {}));
var PublicEmailTestingDetailsAbStatusEnum;
(function (PublicEmailTestingDetailsAbStatusEnum) {
    PublicEmailTestingDetailsAbStatusEnum["Master"] = "master";
    PublicEmailTestingDetailsAbStatusEnum["Variant"] = "variant";
    PublicEmailTestingDetailsAbStatusEnum["LoserVariant"] = "loser_variant";
    PublicEmailTestingDetailsAbStatusEnum["MabMaster"] = "mab_master";
    PublicEmailTestingDetailsAbStatusEnum["MabVariant"] = "mab_variant";
    PublicEmailTestingDetailsAbStatusEnum["AutomatedMaster"] = "automated_master";
    PublicEmailTestingDetailsAbStatusEnum["AutomatedVariant"] = "automated_variant";
    PublicEmailTestingDetailsAbStatusEnum["AutomatedLoserVariant"] = "automated_loser_variant";
})(PublicEmailTestingDetailsAbStatusEnum = exports.PublicEmailTestingDetailsAbStatusEnum || (exports.PublicEmailTestingDetailsAbStatusEnum = {}));
var PublicEmailTestingDetailsAbSuccessMetricEnum;
(function (PublicEmailTestingDetailsAbSuccessMetricEnum) {
    PublicEmailTestingDetailsAbSuccessMetricEnum["ClicksByOpens"] = "CLICKS_BY_OPENS";
    PublicEmailTestingDetailsAbSuccessMetricEnum["ClicksByDelivered"] = "CLICKS_BY_DELIVERED";
    PublicEmailTestingDetailsAbSuccessMetricEnum["OpensByDelivered"] = "OPENS_BY_DELIVERED";
})(PublicEmailTestingDetailsAbSuccessMetricEnum = exports.PublicEmailTestingDetailsAbSuccessMetricEnum || (exports.PublicEmailTestingDetailsAbSuccessMetricEnum = {}));
