// @hubspot/api-client/lib/codegen/crm/associations/schema/types/PromiseAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromiseTypesApi = void 0;
const middleware_1 = require("../middleware");
const ObservableAPI_1 = require("./ObservableAPI");
class PromiseTypesApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableTypesApi(configuration, requestFactory, responseProcessor);
    }
    getAllWithHttpInfo(fromObjectType, toObjectType, _options) {
        var _a;
        let observableOptions;
        if (_options) {
            observableOptions = {
                baseServer: _options.baseServer,
                httpApi: _options.httpApi,
                middleware: (_a = _options.middleware) === null || _a === void 0 ? void 0 : _a.map(m => new middleware_1.PromiseMiddlewareWrapper(m)),
                middlewareMergeStrategy: _options.middlewareMergeStrategy,
                authMethods: _options.authMethods
            };
        }
        const result = this.api.getAllWithHttpInfo(fromObjectType, toObjectType, observableOptions);
        return result.toPromise();
    }
    getAll(fromObjectType, toObjectType, _options) {
        var _a;
        let observableOptions;
        if (_options) {
            observableOptions = {
                baseServer: _options.baseServer,
                httpApi: _options.httpApi,
                middleware: (_a = _options.middleware) === null || _a === void 0 ? void 0 : _a.map(m => new middleware_1.PromiseMiddlewareWrapper(m)),
                middlewareMergeStrategy: _options.middlewareMergeStrategy,
                authMethods: _options.authMethods
            };
        }
        const result = this.api.getAll(fromObjectType, toObjectType, observableOptions);
        return result.toPromise();
    }
}
exports.PromiseTypesApi = PromiseTypesApi;
